
***

# Annotation #

We load the EPIC annotation (hg19), which can be downloaded from https://zwdzwd.github.io/InfiniumAnnotation

```{r 207anno}
anno <- read_tsv("./data/EPIC.hg19.manifest.tsv.gz")

anno <- anno %>% 
  dplyr::select(
    cpg = probeID,
    chr = CpG_chrm,
    start = CpG_beg,
    end = CpG_end,
    strand = probe_strand,
    gene_HGNC,
    MASK_general,
    probeType
  ) %>% 
  mutate(
    chr = substr(chr,4,5)
  )

str(anno)
```

***

# Filtering on Intensities #

We recommend performing probe-filtering by setting unreliably measured probes to NA. Unfortunately, Functional Normalization does not allow NA's in the `RGChannelSet`. Therefore, it is best to run normalization and probe-filtering separately and combine the results afterwards using our `reduce()` function. 

Some observations need to be removed from the dataset since they provide little to no information that could skew analyses. Probes which have zero intensity or are based on very few (default: <3) beads are removed, based on a specified cut off. 

```{r 401filter}
RGset <- probeFiltering(RGset)
```

There are differences between probe types that are used in `probeFiltering()`. Information on these can be extracted and stored using functions from [**minfi**](https://bioconductor.org/packages/release/bioc/html/minfi.html) (Feinberg, 2014).

```{r 402probeinfo}
locusNames <- getManifestInfo(RGset, "locusNames")
typeII <- getProbeInfo(RGset, type="II")
typeI <- getProbeInfo(RGset, type="I")

RGset
betas <- getBeta(RGset, type="Illumina")
```

***

# ENCODE Blacklist #

```{r}
load("./data/ENCODEBlacklist_CpGomit-EPIC.RData")
betas <- betas[!rownames(betas) %in% cpg_blacklist,]
dim(betas)
```

EPICv2 sent by Jazmin

***

# Zhou probes #

Several studies (Zhou, 2017) have characterized cross-reactive and polymorphic probes on both the 450k and EPIC methylation arrays and made suggestions for removal. We chose to remove probes identified by Zhou et al., since this set is actively maintained and supports both array sizes. Additionally, different sets are specified depending on the reference genome used.

In total, Zhou et al. suggest removal of probes for a variety of reasons, from ambiguous mapping and cross-reactivity to polymorphisms that interfere with extension. In general, around 60,000 probes are suggested for removal in 450k arrays, and 100,000 for EPIC. 

We developed the `probeMasking()` function, which removes specified probes from either beta- or M- value matrices. In this instance, 59,780 CpG rows are removed. 

```{r}
maskProbes <- anno[anno$MASK_general == TRUE,]$cpg
length(maskProbes)
betas <- betas[!rownames(betas) %in% maskProbes,]
dim(betas)
```

***

# Sex chromosomes #

For most EWAS we filter out sex chromosomes.

```{r}
xy_cpgs <- (anno %>% 
              dplyr::filter(chr %in% c("X", "Y")))$cpg
betas <- betas[!rownames(betas) %in% xy_cpgs,]
dim(betas)
```

***

# Outlier removal #

Any DNAm values more than 3 IQR from the nearest quartile are removed.

NO. dont remove 3 IQR. present a few options for going into normalization with pros and cons

* beta values - biological interpretation, suffer from heteroskedasticity, outliers

* 3IQR - no outliers, arbitrary, can still be outliers

* M values - no heteroskedasticity

* RIN transform - no outliers, nice dist, not easy to interpret

```{r}
xtabs(~is.na(betas))

iqr_dnam <- apply(betas, 1, function(x){
  iqr <- IQR(x, na.rm = TRUE)
  q1 <- quantile(x, na.rm=TRUE)[[2]]
  q3 <- quantile(x, na.rm=TRUE)[[4]]
  x <- ifelse((x <= q1 - (3*iqr) | x >= q3 + (3*iqr)), NA, x)
})

xtabs(~is.na(iqr_dnam))

dim(iqr_dnam)

betas <- t(iqr_dnam)
```

***

# Missingness #

Keep only the CpGs where less than 5% of sample values are missing

This is all CpGs, representing high quality data

```{r}
perc_na <- rowSums(is.na(iqr_dnam))*100/ncol(iqr_dnam)
summary(perc_na)

betas <- betas[,perc_na <= 95]
dim(betas)
```

Keep only the samples where less than 5% of CpGs are missing

This is all samples, representing high quality data

```{r}
perc_na <- colSums(is.na(iqr_dnam))*100/nrow(iqr_dnam)
summary(perc_na)
betas <- betas[perc_na <= 95,]
dim(betas)
```

***

# SummarizedExperiment #

Now that we have our annotations, we can combine them with our data using the [**SummarizedExperiment**](https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html) package. After ensuring that any column subsetting is accounted for, we extract the `colData()` from our `RGset`. Finally, the beta values, probe annotations, and sample information are combined into a `RangedSummarizedExperiment` object using the `SummarizedExperiment()` function.

Annotation data

```{r}
anno <- anno %>% 
  dplyr::filter(cpg %in% rownames(betas))

dim(anno)
```

Keep annotated probes

```{r}
betas <- betas[anno$cpg, ]
```


Sample information

```{r}
targets <- targets[colnames(betas), ]

dim(targets)
dim(betas)
```

Make summarized experiment object

```{r}
methData <- SummarizedExperiment(
  assays=SimpleList(beta=betas), 
  rowData=anno, 
  colData=targets)

methData
```

You can read more about [**SummarizedExperiment**](http://bioconductor.org/packages/SummarizedExperiment) from the following resources:
<ul>
<li>[Package vignette](http://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html)</li>
<li>[Course material](http://bioconductor.org/help/course-materials/2012/BiocEurope2012/SummarizedExperiment.pdf)</li>
<li>[BioConductor nature paper](http://www.nature.com/nmeth/journal/v12/n2/full/nmeth.3252.html)</li>
</ul>

***

# Save #

Save RGset

```{r}
RGset <- RGset[, colnames(RGset) %in% colnames(methData)]
RGset
```

Save clean data

```{r}
save(targets, file = "./data/targets.Rdata")
save(methData, file="./data/methData.Rdata")
save(RGset, file="./data/RGset.Rdata")
```

***

# Saving #

We want to save our data as a `SummarizedExperiment` object for the rest of the pipeline.

***





