
```{r, child="_setup.Rmd"}
```

```{r 601start, echo=FALSE, message=FALSE, results='hide'}
load("C:/Users/ljsinke/Documents/git/GitHub/test_workflow/data/targets.RData")
targets$Basename <- substring(targets$supplementary_file,68,95)
library(minfi)
RGset <- read.metharray.exp(base="C:/Users/ljsinke/Documents/r/cd/GSE113018/IDATs", targets, extended=TRUE)
GRset <- preprocessFunnorm.DNAmArray(RGset, nPCs=4, keepCN=FALSE)
RGset <- probeFiltering(RGset)
betas <- reduce(GRset, RGset, what="beta")
betas <- invisible(probeMasking(betas, array="450", genome="hg19"))
```

***

# The `RangedSummarizedExperiment` class #

Moving further into the pipeline, it will be necessary to have our methylation data stored in a [`RangedSummarizedExperiment`](http://bioconductor.org/packages/SummarizedExperiment) class object. Designed to handle microarray data, this class stores observations from multiple samples alongside relevant meta-data, and ensures that both features and phenotypes are kept in-sync when subsetting.

This matrix-like container is organised with rows representing features, such as genes or exons, which can be accessed using `rowRanges()`. This function calls a `GRanges` object describing features and their attributes. The columns of the `RangedSummarizedExperiment` store information about each sample, and this can be accessed using `colData()`. 

Lastly, each assay is represented in the third dimension of this matrix-like object and can be shown using the `assays()` function. Meta-data is then linked to this combination of data frames, and can be accessed with the `metadata()` command. 

### Creating a `RangedSummarizedExperiment` ###

The [**FDb.InfiniumMethylation.hg19**](https://bioconductor.org/packages/release/data/annotation/html/FDb.InfiniumMethylation.hg19.html) package contains probe annotations, which can be extracted using the `getPlatform()` function. To ensure that we apply the previous probe filtering, we use `match()` to keep only the relevant information.

```{r 602probeannotate}
library(FDb.InfiniumMethylation.hg19)
rowRanges <- getPlatform(platform="HM450", genome="hg19")
rowRanges[1:10, 1:2]
rows <- match(rownames(betas), names(rowRanges))
```

Now that we have our annotations, we can combine them with our data using the [**SummarizedExperiment**](https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html) package. After ensuring that any column subsetting is accounted for, we extract the `colData()` from our `RGset`. Finally, the beta values, probe annotations, and sample information are combined into a `RangedSummarizedExperiment` object using the `SummarizedExperiment()` function.

```{r 603summexp}
library(SummarizedExperiment)

cols <- match(colnames(betas), colnames(RGset))
colData <- colData(RGset[, cols])@listData

methData <- SummarizedExperiment(assays=SimpleList(beta=betas),
                                 rowRanges=rowRanges[rows], 
                                 colData = DataFrame(colData))
methData
```

You can read more about [**SummarizedExperiment**](http://bioconductor.org/packages/SummarizedExperiment) from the following resources:
- 1. [package vignette](http://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html)
- 2. [course material](http://bioconductor.org/help/course-materials/2012/BiocEurope2012/SummarizedExperiment.pdf)
- 3. [BioConductor nature paper](http://www.nature.com/nmeth/journal/v12/n2/full/nmeth.3252.html)

***

# Phenotype Prediction #

It can be useful to predict phenotypes in your data, either for comparison to assumed categories or to help impute missing values. We are in the process of training reference-based predictors using the large BIOS data, which includes samples from 6 dutch cohorts and over 4000 individuals. These are described in more detail below, and will be included in the package for use on your own datasets. However, please do be mindful that if the BIOS population vastly differs from your own, the predictive power and accuracy of these will suffer.

### Predict White Blood Cell Composition ###

White blood cell count can be predicted using Houseman's method [@Houseman2012] implemented in [**minfi**](http://bioconductor.org/packages/minfi) [@Feinberg2014] by the `estimateCellCounts()` function. Using this partial least squares method, we trained a cell count predictor on the BIOS data.

### Predict Sex ###

The [**DNAmArray**](https://github.com/molepi/DNAmArray) package contains a `getSex.DNAmArray()` function, which predicts the sex of your observations by inspecting intensities of X-chromosome signals. We use the `getBeta()` function from [**minfi**](https://bioconductor.org/packages/release/bioc/html/minfi.html) as before to get the beta-value matrix and then use this to predict sex. Since some filtered probes may be used by the sex prediction function, we recalculate the beta matrix before prediction.

```{r 604sexpredict}
betas <- getBeta(RGset)
mid <- match(colnames(betas), targets$Basename)

predictedSex <- getSex.DNAmArray(betas)
assumedSex <- targets$`gender:ch1`[mid]
predictedSex <- as.character(factor(predictedSex, labels=c("Female", "Male")))
assumedSex <- as.character(factor(assumedSex, labels=c("Female", "Male")))

table(predictedSex, assumedSex)
```

As you can see, this is complete data, but the predicted and assumed sexes are identical. This means that we can feel increased confidence that no incorrect labelling or mix-ups are present in this variable.

### Predict chronological age ###

Age can be predicted using Horvath's or Hannum's coefficients with `agep()` from [**wateRmelon**](https://bioconductor.org/packages/release/bioc/html/wateRmelon.html). If you use Hannum's, you will need to call the coefficients using `data(hannumCoef)` before prediction. The predicted and assumed ages are then combined into a data frame that can be used to visualize the results.

```{r 610agep, message=FALSE, warning=FALSE}
library(wateRmelon)
predictedAge <- as.numeric(agep(betas, method="horvath"))
assumedAge <- as.numeric(targets$`age:ch1`[mid])

ages <- data.frame(Assumed=assumedAge, Predicted=predictedAge)
head(ages)
```

```{r 610theme, echo=FALSE}
th3 <- theme(text=element_text(size=12), 
             panel.border=element_rect(fill=NA, color="#cdcdcd", size=1, linetype=1), 
             plot.title=element_text(size=20, hjust=0.5, margin=margin(0,0,15,0)), 
             plot.margin=margin(20,25,25,25))
```

```{r 610plot}
ggplot(ages, aes(x=Assumed, y=Predicted)) + ggtitle("Assumed vs. Predicted Age") + th3 +
  geom_point(shape=16, color="#158cba") + geom_smooth(method=lm, color="#D67D87")
```

The above graph shows that this prediction follows the general trend of the assumed ages, however the residuals are quite large. It may be possible to develop an improved predictor [@Zhang2018], and we will attempt this using the BIOS data. 

***

# Checking Sample Relationships #

Beta-values of the probes containing SNPs can be extracted from the RGset using [**minfi**](http:/bioconductor.org/packages/minfi)'s `getSnpBeta` function. These can then be converted to alleles using k-means clustering using the `genki()` function from [**wateRmelon**](https://bioconductor.org/packages/release/bioc/html/wateRmelon.html). This function returns values of disperson for each of three genotype peaks (AA, AB, and BB). which can be used as a performance metric since technical variation would decrease the tightness of the genotypic peaks.

```{r 611getsnps, eval=FALSE}
library(wateRmelon)
snpBetas <- getSnpBeta(RGset)
snpBetasVar <- genki(snpBetas)
```

[**omicsPrint**](http://bioconductor.org/packages/release/bioc/html/omicsPrint.html) [@VanIterson2018] is a package we developed to detect data linkage errors through inspecting sample relations in multiple omics studies. This example uses data where individuals are unrelated, but if you have data on twins, for example, it would detect that and also show any mismatches.

Included with the package is the `hm450.manifest.pop.GoNL` data, which stores SNP probe information in a `GRanges` class object. This is then used to create a subset of the beta values for genotyping. 

```{r 612omicsprint, message=FALSE, warning=FALSE}
library(omicsPrint)
data(hm450.manifest.pop.GoNL)
cpgs <- names(hm450.manifest.pop.GoNL[mcols(hm450.manifest.pop.GoNL)$MASK.snp5.EAS])
cpgs <- na.omit(match(cpgs, rownames(betas)))
omicsBetas <- betas[cpgs,]
omicsBetas[1:10, 1:2]
```

The function `beta2genotype()` then genotypes the observations by measuring homozygous or heterozygous alleles at these SNP probes. Lastly `alleleSharing()` assesses the relationships between different individuals, which can be unrelated, twins, or identical. The results can then be visualised using the `inferRelations()` function.

```{r 613genotype}
dnamCalls <- beta2genotype(omicsBetas, assayName="exprs")
dim(dnamCalls)
dnamCalls[1:10, 1:2]

omicsData <- alleleSharing(dnamCalls, verbose=TRUE)
mismatches <- inferRelations(omicsData)
```

***

# Imputating NA Values #

Several algorithms do not allow NA's but beta-values and or M-values can contain NA's after preprocessing. Either because some probes/CpG were specifically set NA or because of transformation from intensities to beta or M-values. The [**impute**](https://bioconductor.org/packages/release/bioc/html/impute.html) package can be used to impute these values. Below, we use the knn approach implemented by `impute.knn()` from this package.

```{r 614imputation, results='hide'}
library(impute)
betas <- impute.knn(as.matrix(betas))$data
```

***

# Probe Annotation #

Our package also contains a `cpgInfo()` function, which can be used to map CpG probes to their nearest genes. This utilises the [**TxDb.Hsapiens.UCSC.hg19.knownGene**](http://bioconductor.statistik.tu-dortmund.de/packages/3.6/data/annotation/html/TxDb.Hsapiens.UCSC.hg19.knownGene.html) package, which accesses the USCS annotation databases.

```{r 615annotate}
cpgs <- rownames(betas)
cpgInfo <- cpgInfo(cpgs, TxDb="TxDb.Hsapiens.UCSC.hg19.knownGene")
cpgInfo[1:10, 1:7]
```

***

# References #
