
```{r, child="_setup.Rmd"}
```

```{r 701start, echo=FALSE, message=FALSE, results='hide'}
load("C:/Users/ljsinke/Documents/git/GitHub/test_workflow/data/targets.RData")
targets$Basename <- substring(targets$supplementary_file,68,95)
RGset <- read.metharray.exp(base="C:/Users/ljsinke/Documents/r/cd/GSE113018/IDATs", targets, extended=TRUE)

GRset <- preprocessFunnorm.DNAmArray(RGset, nPCs=4, keepCN=FALSE)

RGset <- probeFiltering(RGset)
betas <- reduce(GRset, RGset, what="beta")
betas <- probeMasking(betas, array="450", genome="hg19")

assays <- betas

rowRanges <- getPlatform(platform="HM450", genome="hg19")
rows <- match(rownames(betas), names(rowRanges))
rowRanges <- rowRanges[rows]

cols <- match(colnames(betas), colnames(RGset))
colData <- DataFrame(colData(RGset[, cols])@listData)

methData <- SummarizedExperiment(assays=SimpleList(assays=assays),
                                 rowRanges=rowRanges,
                                 colData=colData)

library(dplyr)

targets %>% 
  rename(
    gender.ch1 = `gender:ch1`,
    age.ch1 = `age:ch1`
    )
```

***

# Simple EWAS using **limma** #

Our approach for running an EWAS is similar to the approach of [**missMethyl**](http://bioconductor.org/packages/missMethyl/), except that we skip the empirical Bayes step, because the sample size in an EWAS is large enough and there is no added value in including this.

We assume the data has been normalized, filtered and NA's have been imputed. Furthermore, it is very convenient to construct a `SummarizedExperiment` class object as described in the previous section. In our data we have cases and controls, and our primary interest is differential methylation between the two. We want to adjust for likely confounders, such as age and sex.

```{r 702casen}
table(methData$disease.state.ch1)
```

It is important to ensure that all variables are of a type that facilitates model fitting, so age is converted to a continuous variables and all others are treated as factors. The model formula that you would like to run can be added into your data as shown, ensuring that the covariate of interest immediately follows the tilde. 

```{r 703formula}
ourData <- methData[, methData$disease.state.ch1 %in% c("FASD", "control")]

ourData$disease.state.ch1 <- factor(ourData$disease.state.ch1)
ourData$gender.ch1 <- factor(ourData$gender.ch1)
ourData$age.ch1 <- as.numeric(ourData$age.ch1)

metadata(ourData)$formula <- ~disease.state.ch1 + age.ch1 + gender.ch1 
```

By subsetting our `SummarizedExperiment` object, we can extract covariates with complete information. Sex chromosome probes are also filtered out here.

```{r 704complete}
covariates <- get_all_vars(metadata(ourData)$formula, data=colData(ourData))
nas <- apply(covariates, 1, anyNA)
ourData <- ourData[, !nas]

ourData <- ourData[!(seqnames(ourData) %in% c("chrX", "chrY")),]
```

The `design` object is a matrix holding coefficient multipliers for each observation. As you can see, it is very important to check this to be sure that variables are being handled correctly.

```{r 705model}
design <- model.matrix(metadata(ourData)$formula, data=colData(ourData))
design[1:10,1:3]
```

After extracting beta values from our data, we can use the `lmFit()` function from [**limma**](https://bioconductor.org/packages/release/bioc/html/limma.html) to fit the model. Returned p-values are adjusted for multiple testing using the Bonferroni correction, and the CpG sites bearing most significance are returned for inspection.

```{r 706ewas}
data <- assays(ourData)$assays

library(limma)
fit <- lmFit(data, design)

tstat <- fit$coef/fit$stdev.unscaled/fit$sigma

pval <- 2*pnorm(-abs(tstat[,2]))
padj <- p.adjust(sort(pval, decreasing=FALSE), method="bonf")
head(padj[padj < 0.05])
```

# Unobserved covariates using **cate** #

Often unobserved technical or biological covariates can be present. A
recent approach available from cran
[**cate**](https://cran.r-project.org/package=cate) is able to
estimate the unobserved covariates. There are other methods as well,
like, sva, ruv but we prefer this one...

```{r cate, eval=FALSE}
library(cate)
fa <- cate.fit(design[,2,drop=FALSE], design[,-2], t(data), r = 3, calibrate=FALSE, adj.method="naive")
```

Here we assumed the presence of three unobserved covariates but cate
has two methods implemented that can estimate the precise number of
unobserved covariates. Furthermore, notice that we use
`calibrate=FALSE` as we think the calibration of test-statistics and P
values is too conservative. Actually, we propose a different method to
control for test-statistic inflation and bias (see next section).

# Correct for bias and inflation #

We developed a package called
[**bacon**](http://bioconductor.org/packages/bacon) to estimate and
correct for bias and inflation for EWAS (and TWAS)[@Iterson2017] it
maximizes power while properly controlling the false positive
rate. [**bacon**](http://bioconductor.org/packages/bacon) is based on
the estimation of the empirical null distribution using Bayesian
statistics. The utility of the tool was illustrated through the
application of meta-analysis by performing epigenome- and
transcriptome-wide association studies (EWASs/TWASs) on age and
smoking which highlighted an overlap in differential methylation and
expression of associated genes. It can be used to remove inflation and
bias often observed in EWAS/TWAS see for more details[@Iterson2017].

For example, we can use it on the EWAS that we just run:

```{r bacon, eval=FALSE}
library(bacon)
tstats <- cbind(limma=tstat[,2], cate=as.vector(fa$beta.t))
library(BiocParallel)
register(MulticoreParam(2))
bc <- bacon(tstats)
bc
print(plot(bc, type="hist"))
print(plot(bc, type="qq"))
pvalsbc <- pval(bc)
head(pvalsbc[order(pvalsbc[,1]),])
pvals <- pval(bc, FALSE)
head(pvals[order(pvals[,1]),])
```

Strong bias is observed so running
[**bacon**](http://bioconductor.org/packages/bacon) should really be
considered. There is a small difference in the calculated P
values. Running [**cate**](https://cran.r-project.org/package=cate)
did not yielded additional control for bias and inflation in this
example.

# References #
