
```{r, child="_setup.Rmd"}
```

# Simple EWAS using **limma** #

Our approach for running an EWAS is similar to the approach of
[**missMethyl**](http://bioconductor.org/packages/missMethyl/). Expect
that we the empirical Bayes step, because the sample size in an EWAS
is large enough and there should no added value in running the
`eBayes`-step?

We assume the data has been normalized, filtered and NA's have been
imputed. Furthermore, it is very convenient to construct a
`SummarizedExperiment`-object as described in the previous section.

For example, the model that you would like to run can be added like
this with the first covariate the phenotype of interest.

```{r formula}
metadata(mvalues)$formula <- ~DNA_BloodSampling_Age + Sex + Sentrix_Position + Smoking + Mono_Perc + Baso_Perc + Eos_Perc + Neut_Perc
```

Next, we extract those samples having a complete set of
covariates. Notice that we subset the `SummarizedExperiment`-object!


```{r complete}
covariates <- get_all_vars(metadata(mvalues)$formula, data=colData(mvalues))
nas <- apply(covariates, 1, anyNA)
mvalues <- mvalues[, !nas]
```

Remove probes on the X and Y chromosomes and probes containing a SNPs
or does mapping to multiple locations. 

```{r, snpprobes}
mvalues <- mvalues[!(seqnames(mvalues) %in% c("chrX", "chrY")),]
data(hm450.manifest.pop.GoNL) ##From DNAmArray
hm450.manifest.pop.GoNL
hm450.manifest.pop.GoNL <- hm450.manifest.pop.GoNL[!is.na(hm450.manifest.pop.GoNL$MASK.general.GoNL) &
                                                   hm450.manifest.pop.GoNL$MASK.general.GoNL == TRUE, ]

mvalues <- mvalues[!(names(mvalues) %in% names(hm450.manifest.pop.GoNL)),]         
```

Now we can run the model like this and we do not have to bother on
sample administration!

```{r ewas}
library(limma)
design <- model.matrix(metadata(mvalues)$formula, data=colData(mvalues))
data <- assays(mvalues)$data
fit <- lmFit(data, design)
tstat <- fit$coef/fit$stdev.unscaled/fit$sigma

pval <- 2*pnorm(-abs(tstat[,2]))
padj <- p.adjust(sort(pval, decreasing=FALSE), method="bonf")
head(padj[padj < 0.05])
```

# Unobserved covariates using **cate** #

Often unobserved technical or biological covariates can be present. A
recent approach available from cran
[**cate**](https://cran.r-project.org/package=cate) is able to
estimate the unobserved covariates. There are other methods as well,
like, sva, ruv but we prefer this one...

```{r cate}
library(cate)
fa <- cate.fit(design[,2,drop=FALSE], design[,-2], t(data), r = 3, calibrate=FALSE, adj.method="naive")
```

Here we assumed the presence of three unobserved covariates but cate
has two methods implemented that can estimate the precise number of
unobserved covariates. Furthermore, notice that we use
`calibrate=FALSE` as we think the calibration of test-statistics and P
values is too conservative. Actually, we propose a different method to
control for test-statistic inflation and bias (see next section).

# Correct for bias and inflation #

We developed a package called
[**bacon**](http://bioconductor.org/packages/bacon) to estimate and
correct for bias and inflation for EWAS (and TWAS)[@Iterson2017] it
maximizes power while properly controlling the false positive
rate. [**bacon**](http://bioconductor.org/packages/bacon) is based on
the estimation of the empirical null distribution using Bayesian
statistics. The utility of the tool was illustrated through the
application of meta-analysis by performing epigenome- and
transcriptome-wide association studies (EWASs/TWASs) on age and
smoking which highlighted an overlap in differential methylation and
expression of associated genes. It can be used to remove inflation and
bias often observed in EWAS/TWAS see for more details[@Iterson2017].

For example, we can use it on the EWAS that we just run:

```{r bacon}
library(bacon)
tstats <- cbind(limma=tstat[,2], cate=as.vector(fa$beta.t))
library(BiocParallel)
register(MulticoreParam(2))
bc <- bacon(tstats)
bc
print(plot(bc, type="hist"))
print(plot(bc, type="qq"))
pvalsbc <- pval(bc)
head(pvalsbc[order(pvalsbc[,1]),])
pvals <- pval(bc, FALSE)
head(pvals[order(pvals[,1]),])
```

Strong bias is observed so running
[**bacon**](http://bioconductor.org/packages/bacon) should really be
considered. There is a small difference in the calculated P
values. Running [**cate**](https://cran.r-project.org/package=cate)
did not yielded additional control for bias and inflation in this
example.

# References #
