---
title:  | 
  | Streamlined workflow for the quality control, normalization, and analysis of Illumina methylation array data
  | 
  
author:  | 
  | Lucy Sinke, Maarten van Iterson, Davy Cats, BIOS Consortium, Roderick Slieker, and Bas Heijmans
  | <font size=2 color=#6b6b6b>Molecular Epidemiology, Department of Biomedical Data Sciences, Leiden University Medical Center, Leiden, The Netherlands</font>
  
email: l.j.sinke@lumc.nl
---

```{r, child="_setup.Rmd"}
```
<div class="DOI">
  [![DOI](/exports/molepi/users/ljsinke/DNAmArray/DNAmArray_workflow/data/images/zenodo_3355292.svg)](http://doi.org/10.5281/zenodo.3355292)
</div>

***

# Introduction #

Here we provide a complete workflow for the preprocessing and analysis of DNA methylation array data. The workflow combines best practices in the field with in-house developed methodology, and is geared towards large-scale studies, including Epigenome-wide Association Studies (EWAS). Its development was informed by our research analysing [BIOS consortium](https://www.bbmri.nl/samples-images-data) data, which contains multiomics measures from 6 Dutch biobanks comprising ~4000 individuals (Dekkers, 2016; Slieker, 2016; Bonder, 2017; Luijk, 2018).

The [**DNAmArray**](https://github.com/molepi/DNAmArray)-package contains a series of convenient functions for the quality control, normalization, and analysis of methylation array data. The workflow has been thoroughly tested for the Illumina 450k and EPIC array but is similarly applicable to the newer EPICv2 array.

It is worth noting that this workflow makes extensive use of other [BioConductor](https://www.bioconductor.org) packages. For example, the [**DNAmArray**](https://github.com/molepi/DNAmArray) function `read.metharray.exp.par()` converts IDAT files to an `RGset` by harnessing functions from [**minfi**](https://bioconductor.org/packages/release/bioc/html/minfi.html) (Feinberg, 2014) and combining them with [**BiocParallel**](https://bioconductor.org/packages/release/bioc/html/BiocParallel.html). Usually the required packages are installed automatically, but otherwise please refer to the relevant package's documentation.

Furthermore, we have also developed packages used by this workflow.  [**MethylAid**](http:/bioconductor.org/packages/MethylAid) (Luijk, 2014) provides a web application to assist in performing interactive sample quality control, and [**bacon**](http:/bioconductor.org/packages/bacon) (Van Iterson, 2017) corrects for bias and inflation in ome-wide association studies, such as EWAS.

***

# Example Data #

The [example data](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE116339) (Curtis et al., 2019) used in this workflow is available from the [NCBI Gene Expression Omnibus](https://www.ncbi.nlm.nih.gov/geo/) (GEO), a public repository of microarray data. It contains genome-wide DNA methylation data from whole blood obtained using the Illumina EPIC microarray. The participants consist of 679 children exposed to polybrominated biphenyl (PBB), an endocrine-disrupting compound, which was accidently added to the food supply in Michigan in the 1970â€™s.

***


