---
title: "Streamlined workflow for the quality control, normalization, and bias-free analysis of Illumina methylation array data - The Leiden approach"
author: Maarten van Iterson, Elmar Tobi, Roderick Slieker, Wouter den Hollander, Koen Dekkers, Rene Luijk, Lucy Sinke, Eline Slagboom, and Bas Heijmans; Department of Molecular Epidemiology, Leiden University Medical Center, Leiden, The Netherlands
email: l.j.sinke@lumc.nl
---

```{r, child="_setup.Rmd"}
```

[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.158908.svg)](https://doi.org/10.5281/zenodo.158908)

***

# Introduction #

Here we provide a complete workflow for the preprocessing and analysis
of DNA methylation array data. The workflow combines best practices in
the field with in-house developed methodology, and is geared towards
large-scale studies, including Epigenome-wide Association Studies
(EWAS). Specifically, it is designed for analysis of data obtained from the [BIOS consortium](https://www.bbmri.nl/samples-images-data), which contains multiomics measures from 6 Dutch biobanks comprising ~4000 individuals.

The [**DNAmArray**](https://github.com/molepi/DNAmArray)-package
contains a series of convenient functions for the quality control,
normalization, and analysis of methylation array data. The workflow has
been thoroughly tested for the Illumina 450k array but will similarly be
applicable to the newer 850k EPIC array pending updates of
[BioConductor](https://www.bioconductor.org) packages from other
groups.

It is worth noting that this workflow makes extensive use of other [BioConductor](https://www.bioconductor.org) packages. For example, [**DNAmArray**](https://github.com/molepi/DNAmArray) converts IDAT files to an `RGset` by harnessing functions from [**minfi**](https://bioconductor.org/packages/release/bioc/html/minfi.html) [@Aryee2014] and combining them with [BiocParallel](https://bioconductor.org/packages/release/bioc/html/BiocParallel.html). Usually these are installed
automatically, but otherwise please refer to the relevant package's documentation.

Furthermore, we have also developed packages used by this workflow.  [**MethylAid**](http:/bioconductor.org/packages/MethylAid)
[@Iterson2014] provides a web application to assist in performing interactive sample quality control, and  [**bacon**](http:/bioconductor.org/packages/bacon)
[@Iterson2017] corrects for bias and inflation in ome-wide association studies, such as EWAS.

***

# Example Data #

The [example data](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE48472) [@Slieker2013] used in this workflow is available from the [NCBI Gene Expression Omnibus](https://www.ncbi.nlm.nih.gov/geo/) (GEO), a public repository of microarray data. It contains genome-wide DNA methylation data from multiple tissues obtained using the Illumina 450k microarray. 

Internal tissue samples (blood, liver, muscle, pancreas, and spleen) were obtained from six cadavers (mean age: 65.5 years), and internal tissue samples (blood, saliva, hair follicle, and buccal swab) were collected from five healthy volunteers (mean age: 28 years).

***

# Installation #
  
The [**DNAmArray**](https://github.com/molepi/DNAmArray)-package can
be installed in several ways, and has been successfully for >= R-3.2.0 on various Linux-builds and for >= R-3.5.3 on Windows.

### Install using the **devtools**-package ###

First install the package
[**devtools**](https://github.com/hadley/devtools). Next use:

```{r devtools, eval=FALSE}
library(devtools)
install_github("molepi/DNAmArray")
```

### Install from source using `git/R` ###

Using [git](https://git-scm.com/), you can `git clone`, build,
and then install the package from source:

```{r git, engine='bash', eval=FALSE}
git clone git@github.com/molepi/DNAmArray.git
R CMD build DNAmArray
R CMD INSTALL DNAmArray_x.y.z.tar.gz
```
Change `_x.y.z.` to the proper version you downloaded!

***

# References #


