---
title: "Streamlined workflow for the quality control, normalization and bias-free analysis of Illumina methylation array data - The Leiden approach"
author: Maarten van Iterson, Elmar Tobi, Roderick Slieker, Wouter den Hollander, Koen Dekkers, Rene Luijk, Eline Slagboom and Bas Heijmans Department of Molecular Epidemiology, Leiden University Medical Center, Leiden, The Netherlands
email: m.van\_iterson@lumc.nl
---

```{r, child="_setup.Rmd"}
```

[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.158908.svg)](https://doi.org/10.5281/zenodo.158908)

# Introduction #

Here we provide a complete workflow for the preprocessing and analysis
of DNA methylation array data. The workflow combines best practices in
the field with in-house developed methodology and is geared towards
large-scale studies, including Epigenomewide Association Studies
(EWAS). A specific aim of the work flow is to support such studies
within the BIOS consortium encompassing multiple omics data on 4000
individuals from 6 Dutch biobanks operating within the frame of
[BBMRI-NL](https://www.bbmri.nl/on_offer/bios/)

The [**DNAmArray**](https://github.com/molepi/DNAmArray)-package
contains a series of convenient functions for the quality control,
normalization and analysis of methylation array data. The workflow has
been thoroughly tested on 450k array data but will similarly be
applicable to the newer 850k EPIC array pending updates of
[BioConductor](https://www.bioconductor.org) packages from other
groups.

The workflow and package extensively uses functionality of other
[BioConductor](https://www.bioconductor.org) packages, e.g., for
transforming idats to so-called `RGset` we use
[**minfi**](http:/bioconductor.org/packages/minfi) but
[**DNAmArray**](https://github.com/molepi/DNAmArray) provides
functionality to do this in parallel. Furthermore, we have developed
two other packages
[**MethylAid**](http:/bioconductor.org/packages/MethylAid)
[@Iterson2014] and [**bacon**](http:/bioconductor.org/packages/bacon)
[@Iterson2017], reps., performing interactive sample quality control
and bias- and inflation-correction in epigenomewide association
studies, each have been deposited at BioConductor (of note:
[**bacon**](http:/bioconductor.org/packages/bacon) also applies to
transcriptome- and any other ome-wide association studies) .

# Leiden Longevity Study #

The example DNA methylation data that is used in this workflow is from
the Leiden Longevity Study. The Leiden Longevity Study has been
described in detail previously [@Westendorp2009]. It is a family based
study consists of 1671 offspring of 421 nonagenarians sibling pairs of
Dutch descent, and their 744 partners. In the current EWAS initiative
793 individuals with available DNA methylation data and phenotypic
data (mean age = 60 years) were included.

# Installation #
  
The [**DNAmArray**](https://github.com/molepi/DNAmArray)-package can
be installed in several ways which are described below. The package
has been installed successfully for >= R-3.2.0 on different
linux-builds.

The package depends on many other packages from
[BioConductor](https://www.bioconductor.org) or
[cran](https://cran.r-project.org/). Usually these are installed
automatically, otherwise, we refer to
[BioConductor](https://www.bioconductor.org/install/) or
[cran](https://cran.r-project.org/doc/manuals/r-release/R-admin.html#Installing-packages)
documentation for the installation of these packages.

> THIS WILL BE REPLACE BY ANNOTATION FROM [@Zhou2016]:

**The installed package takes around ~7Mb of memory a large bit of it
comes from an `Rdata`-object containing the information on the
cross-hybridizing probes[@Chen2013].**


## Install using the **devtools**-package ##

First install the package
[**devtools**](https://github.com/hadley/devtools). Next use:

```{r devtools, eval=FALSE}
library(devtools)
install_github("molepi/DNAmArray.git")
```

## Install from source using `git/R` ##

Using [git](https://git-scm.com/), e.g, use `git clone` and then build
and install the package from source:

```{r git, engine='bash', eval=FALSE}
git clone git@github.com/molepi/DNAmArray.git
R CMD build DNAmArray
R CMD INSTALL DNAmArray_x.y.z.tar.gz
```
Change `_x.y.z.` to the proper version you downloaded!
    
# References #


