<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>01_GettingStarted.knit</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span. { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DNAmArray</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="01_GettingStarted.html">Getting Started</a>
</li>
<li>
  <a href="02_SampleQualityControl.html">Sample QC</a>
</li>
<li>
  <a href="03_Predict.html">Predict</a>
</li>
<li>
  <a href="04_Normalization.html">Normalize</a>
</li>
<li>
  <a href="05_ProbeQC.html">Probe QC</a>
</li>
<li>
  <a href="06_EWAS.html">Analyse</a>
</li>
<li>
  <a href="07_Interpret.html">Interpret</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">About</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">




</div>


<hr />
<div id="installation" class="section level1">
<h1>Installation</h1>
<p>The <a href="https://github.com/molepi/DNAmArray"><strong>DNAmArray</strong></a> package can be installed in several ways, and has been tested for &gt;= R-4.4.3 on various Linux-builds and Windows.</p>
<hr />
<div id="install-using-devtools" class="section level3">
<h3>Install using <strong>devtools</strong></h3>
<p>First, install <a href="https://github.com/hadley/devtools"><strong>devtools</strong></a>, and then use the <code>install_github()</code> function to fetch the <a href="https://github.com/molepi/DNAmArray"><strong>DNAmArray</strong></a> package.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(devtools)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">install_github</span>(<span class="st">&quot;molepi/DNAmArray&quot;</span>)</a></code></pre></div>
<hr />
</div>
<div id="install-from-source-using-gitr" class="section level3">
<h3>Install from source using <code>git/R</code></h3>
<p>Using <a href="https://git-scm.com/">git</a>, you can <code>git clone</code> our repository and then install the package, changing <code>_x.y.z.</code> to the relevant version.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">git clone git<span class="op">@</span>github.com<span class="op">/</span>molepi<span class="op">/</span>DNAmArray.git</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">R CMD build DNAmArray</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">R CMD INSTALL DNAmArray_x.y.z.tar.gz</a></code></pre></div>
<hr />
</div>
</div>
<div id="loading-packages" class="section level1">
<h1>Loading packages</h1>
<p>First, load the packages that are required for this workflow:</p>
<ul>
<li><a href="https://github.com/molepi/DNAmArray"><strong>DNAmArray</strong></a> - the main package, containing in-house build functions for the preprocessing of DNAm data</li>
<li><a href="https://www.bioconductor.org/packages/release/bioc/html/MethylAid.html"><strong>MethylAid</strong></a><sup>10</sup> - for sample-level quality control</li>
<li><a href="https://bioconductor.org/packages/release/bioc/html/omicsPrint.html"><strong>omicsPrint</strong></a><sup>12</sup> - in-house tool use to detect sample linkage errors and resolve them</li>
<li><a href="http://bioconductor.org/packages/bacon"><strong>bacon</strong></a><sup>11</sup> - in-house tool for reducing bias and inflation in EWAS test statistics</li>
<li><a href="https://bioconductor.org/packages/release/bioc/html/GEOquery.html"><strong>GEOquery</strong></a><sup>14</sup> - bridges the gap between <a href="https://www.bioconductor.org">BioConductor</a> tools and <a href="https://www.ncbi.nlm.nih.gov/geo/">GEO</a></li>
<li><a href="https://www.tidyverse.org/"><strong>tidyverse</strong></a> - for data wrangling</li>
<li><a href="https://cran.r-project.org/web/packages/reshape2/index.html"><strong>reshape2</strong></a> - for data wrangling</li>
<li><a href="https://ggrepel.slowkow.com/"><strong>ggrepel</strong></a> - for data visualization</li>
<li><a href="https://www.bioconductor.org/packages/release/bioc/html/ComplexHeatmap.html"><strong>ComplexHeatmap</strong></a> - for creating heatmaps</li>
<li><a href="https://cran.r-project.org/web/packages/circlize/index.html"><strong>circlize</strong></a> - designed to allow circular plots, but also used to create custom colour palettes</li>
<li><a href="https://www.bioconductor.org/packages/release/bioc/html/BiocParallel.html"><strong>BiocParallel</strong></a> - to parallelize processing of genomic data</li>
<li><a href="https://www.bioconductor.org/packages/release/data/annotation/html/IlluminaHumanMethylationEPICmanifest.html"><strong>IlluminaHumanMethylationEPICmanifest</strong></a> - containing EPIC probe annotation for the example data (other packages are available for 450k or EPICv2)</li>
<li><a href="https://bioconductor.org/packages/release/bioc/html/minfi.html"><strong>minfi</strong></a><sup>9</sup> - functions to preprocess DNAm data</li>
<li><a href="https://www.bioconductor.org/packages/devel/bioc/html/wateRmelon.html"><strong>wateRmelon</strong></a><sup>30</sup> - functions to preprocess DNAm data</li>
<li><a href="https://cran.r-project.org/web/packages/snow/index.html"><strong>snow</strong></a> - required by MethyLImp2 for parallelization</li>
<li><a href="https://www.bioconductor.org/packages/release/bioc/html/limma.html"><strong>limma</strong></a><sup>27</sup> - for EWAS analysis</li>
<li><a href="https://www.bioconductor.org/packages/release/bioc/html/EpiDISH.html"><strong>EpiDISH</strong></a><sup>21</sup> - for cell count predictions</li>
<li><a href="https://bioconductor.org/packages/release/bioc/html/sva.html"><strong>sva</strong></a><sup>23-25</sup> - for estimating latent factors</li>
<li><a href="https://bioconductor.org/packages/release/bioc/html/methyLImp2.html"><strong>methyLImp2</strong></a><sup>20</sup></li>
<li><a href="https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html"><strong>SummarizedExperiment</strong></a></li>
</ul>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw">library</span>(DNAmArray)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw">library</span>(MethylAid)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="kw">library</span>(omicsPrint)</a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="kw">library</span>(bacon)</a>
<a class="sourceLine" id="cb3-5" data-line-number="5"></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="kw">library</span>(GEOquery)</a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="kw">library</span>(tidyverse)</a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="kw">library</span>(ggrepel)</a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="kw">library</span>(reshape2)</a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="kw">library</span>(ComplexHeatmap)</a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="kw">library</span>(circlize)</a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="kw">library</span>(BiocParallel)</a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="kw">library</span>(IlluminaHumanMethylationEPICmanifest)</a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="kw">library</span>(minfi)</a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="kw">library</span>(wateRmelon)</a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="kw">library</span>(snow)</a>
<a class="sourceLine" id="cb3-17" data-line-number="17"></a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="kw">library</span>(limma)</a>
<a class="sourceLine" id="cb3-19" data-line-number="19"><span class="kw">library</span>(EpiDISH)</a>
<a class="sourceLine" id="cb3-20" data-line-number="20"><span class="kw">library</span>(sva)</a>
<a class="sourceLine" id="cb3-21" data-line-number="21"><span class="kw">library</span>(methyLImp2)</a>
<a class="sourceLine" id="cb3-22" data-line-number="22"><span class="kw">library</span>(SummarizedExperiment)</a></code></pre></div>
<hr />
</div>
<div id="importing-idats" class="section level1">
<h1>Importing IDATs</h1>
<p>The first step in analysing microarray data is importing raw intensity files into your software program. In this example, we show how to import raw IDAT files from GEO into R, but similar strategies can be employed for all Illumina DNAm array files when using R.</p>
<p>Using the <code>getGEOSuppFiles()</code> function, supplementary data is downloaded to the current working directory. These consist of the raw IDAT files alongside relevant documentation.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw">getGEOSuppFiles</span>(<span class="st">&quot;GSE116339&quot;</span>)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="kw">untar</span>(<span class="dt">tarfile=</span><span class="st">&quot;../GSE116339/GSE116339_RAW.tar&quot;</span>, <span class="dt">exdir=</span><span class="st">&quot;../GSE116339/IDATs&quot;</span>)</a></code></pre></div>
<p>The data can then be efficiently unpacked using the <code>gunzip()</code> function.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw">setwd</span>(<span class="st">&quot;../GSE116339/IDATs&quot;</span>)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw">sapply</span>(<span class="kw">list.files</span>(), gunzip)</a></code></pre></div>
<p><code>getGEO()</code> can then be used to import SOFT format microarray data into R as a large GSE-class list, and extract the metadata of interest.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">GSE116339 &lt;-<span class="st"> </span><span class="kw">getGEO</span>(<span class="dt">filename=</span><span class="st">&#39;./data/GSE116339_series_matrix.txt.gz&#39;</span>, <span class="dt">getGPL =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">targets &lt;-<span class="st"> </span><span class="kw">phenoData</span>(GSE116339)<span class="op">@</span>data</a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="kw">rm</span>(GSE116339)</a></code></pre></div>
<hr />
</div>
<div id="preparing-targets" class="section level1">
<h1>Preparing <code>targets</code></h1>
<p>Before progressing further, take some time to get familiar with the <code>targets</code> data frame, removing duplicate information and converting variables to relevant classes.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">targets &lt;-<span class="st"> </span>targets <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="dt">sample_ID =</span> <span class="st">`</span><span class="dt">sample_id:ch1</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">                              geo_accession,</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">                              <span class="dt">sex =</span> <span class="st">`</span><span class="dt">gender:ch1</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb7-4" data-line-number="4">                              <span class="dt">age =</span> <span class="st">`</span><span class="dt">age:ch1</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb7-5" data-line-number="5">                              <span class="dt">log_total_pbb =</span> <span class="st">`</span><span class="dt">ln(totalpbb):ch1</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb7-6" data-line-number="6">                              <span class="dt">pbb_153 =</span> <span class="st">`</span><span class="dt">pbb-153:ch1</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb7-7" data-line-number="7">                              <span class="dt">pbb_77 =</span> <span class="st">`</span><span class="dt">pbb-77:ch1</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb7-8" data-line-number="8">                              <span class="dt">pbb_101 =</span> <span class="st">`</span><span class="dt">pbb-101:ch1</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb7-9" data-line-number="9">                              <span class="dt">pbb_180 =</span> <span class="st">`</span><span class="dt">pbb-180:ch1</span><span class="st">`</span>,</a>
<a class="sourceLine" id="cb7-10" data-line-number="10">                              supplementary_file) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb7-11" data-line-number="11"><span class="st">  </span><span class="kw">separate</span>(supplementary_file, <span class="dt">sep=</span><span class="st">&quot;_&quot;</span>, <span class="dt">remove=</span>F, <span class="dt">into=</span><span class="kw">c</span>(<span class="ot">NA</span>, <span class="st">&quot;plate&quot;</span>, <span class="st">&quot;row&quot;</span>, <span class="ot">NA</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb7-12" data-line-number="12"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">age =</span> <span class="kw">as.numeric</span>(age),</a>
<a class="sourceLine" id="cb7-13" data-line-number="13">         <span class="dt">log_total_pbb =</span> <span class="kw">as.numeric</span>(log_total_pbb),</a>
<a class="sourceLine" id="cb7-14" data-line-number="14">         <span class="dt">pbb_153 =</span> <span class="kw">as.numeric</span>(pbb_<span class="dv">153</span>),</a>
<a class="sourceLine" id="cb7-15" data-line-number="15">         <span class="dt">pbb_77 =</span> <span class="kw">as.numeric</span>(pbb_<span class="dv">77</span>),</a>
<a class="sourceLine" id="cb7-16" data-line-number="16">         <span class="dt">pbb_101 =</span> <span class="kw">as.numeric</span>(pbb_<span class="dv">101</span>),</a>
<a class="sourceLine" id="cb7-17" data-line-number="17">         <span class="dt">pbb_180 =</span> <span class="kw">as.numeric</span>(pbb_<span class="dv">180</span>),</a>
<a class="sourceLine" id="cb7-18" data-line-number="18">         <span class="dt">row =</span> <span class="kw">as.numeric</span>(<span class="kw">substr</span>(row, <span class="dv">3</span>, <span class="dv">3</span>)))</a>
<a class="sourceLine" id="cb7-19" data-line-number="19"></a>
<a class="sourceLine" id="cb7-20" data-line-number="20"><span class="kw">str</span>(targets)</a></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    679 obs. of  12 variables:
##  $ sample_ID         : chr  &quot;3141584&quot; &quot;10120245&quot; &quot;99990005&quot; &quot;12130802&quot; ...
##  $ geo_accession     : chr  &quot;GSM3228562&quot; &quot;GSM3228563&quot; &quot;GSM3228564&quot; &quot;GSM3228565&quot; ...
##  $ sex               : chr  &quot;Female&quot; &quot;Female&quot; &quot;Male&quot; &quot;Male&quot; ...
##  $ age               : num  75 31.5 46.7 46.4 53.6 ...
##  $ log_total_pbb     : num  -2.542 -2.754 -0.371 0.115 -1.548 ...
##  $ pbb_153           : num  0.069 0.054 0.68 1.107 0.203 ...
##  $ pbb_77            : num  0 0 0 0 0 0 0 0 0 0 ...
##  $ pbb_101           : num  0 0 0 0.008 0 0 0 0 0 0 ...
##  $ pbb_180           : num  0 0 0 0 0 0 0 0 0 0 ...
##  $ supplementary_file: chr  &quot;ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3228nnn/GSM3228562/suppl/GSM3228562_200550980002_R01C01_Grn.idat.gz&quot; &quot;ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3228nnn/GSM3228563/suppl/GSM3228563_200550980002_R02C01_Grn.idat.gz&quot; &quot;ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3228nnn/GSM3228564/suppl/GSM3228564_200550980002_R03C01_Grn.idat.gz&quot; &quot;ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM3228nnn/GSM3228565/suppl/GSM3228565_200550980002_R04C01_Grn.idat.gz&quot; ...
##  $ plate             : chr  &quot;200550980002&quot; &quot;200550980002&quot; &quot;200550980002&quot; &quot;200550980002&quot; ...
##  $ row               : num  1 2 3 4 5 6 7 8 1 3 ...</code></pre>
<p>This data frame consists of 679 observations and phenotypic information for 51 variables is stored after cleaning. This includes:</p>
<ul>
<li><code>sample_ID</code> - the ID of the sample</li>
<li><code>geo_accession</code> - the GEO accession number of the sample</li>
<li><code>sex</code> - male or female</li>
<li><code>age</code> - age of the individual</li>
<li><code>log_total_pbb</code>, <code>pbb_153</code>, <code>pbb_77</code>, <code>pb_101</code>, <code>pbb_180</code> - exposure levels</li>
<li><code>supplementary_file</code> - location of IDAT file</li>
<li><code>plate</code> - EPIC array number</li>
<li><code>row</code> - row number on the array (continuous)</li>
</ul>
<hr />
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
