<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>02_SampleQualityControl.knit</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DNAmArray</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="01_GettingStarted.html">Getting Started</a>
</li>
<li>
  <a href="02_SampleQualityControl.html">Sample-level QC</a>
</li>
<li>
  <a href="03_Normalization.html">Normalization</a>
</li>
<li>
  <a href="04_ProbeFiltering.html">Probe-level QC</a>
</li>
<li>
  <a href="05_Predict.html">Predictors</a>
</li>
<li>
  <a href="06_Masking.html">Underperforming Probes</a>
</li>
<li>
  <a href="07_EWAS.html">Analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">About</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">




</div>


<hr />
<div id="using-methylaid" class="section level1">
<h1>Using MethylAid</h1>
<p>We now have the data we need to commence quality control, but some
reformatting is needed. The <a
href="http://bioconductor.org/packages/MethylAid"><strong>MethylAid</strong></a>
(Luijk, 2014) package that we developed requires the targets data frame
to store IDAT file root names in a <code>Basename</code> column.
Sometimes, data comes with a sample sheet to faciliate this, but in this
case the information needs to be extracted from the supplementary file
column instead.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>targets<span class="sc">$</span>Basename <span class="ot">&lt;-</span> <span class="fu">substring</span>(targets<span class="sc">$</span>supplementary_file,<span class="dv">68</span>,<span class="dv">106</span>)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">head</span>(targets<span class="sc">$</span>Basename)</span></code></pre></div>
<pre><code>## [1] &quot;GSM3228562_200550980002_R01C01_Grn.idat&quot; &quot;GSM3228563_200550980002_R02C01_Grn.idat&quot; &quot;GSM3228564_200550980002_R03C01_Grn.idat&quot;
## [4] &quot;GSM3228565_200550980002_R04C01_Grn.idat&quot; &quot;GSM3228566_200550980002_R05C01_Grn.idat&quot; &quot;GSM3228567_200550980002_R06C01_Grn.idat&quot;</code></pre>
<p>The following sample quality control steps require the <a
href="http://bioconductor.org/packages/MethylAid"><strong>MethylAid</strong></a>
and <a
href="https://bioconductor.org/packages/release/bioc/html/BiocParallel.html"><strong>BiocParallel</strong></a>
packages. Using parallel processing and/or batches will reduce both
memory load and run-times when extracting intensities from IDAT files.
Please see the <a
href="http://bioconductor.org/packages/release/bioc/vignettes/MethylAid/inst/doc/MethylAid.pdf">MethylAid
vignette</a> for more details.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">library</span>(MethylAid)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="fu">library</span>(BiocParallel)</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="fu">library</span>(IlluminaHumanMethylationEPICmanifest)</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>BPPARAM <span class="ot">&lt;-</span> <span class="fu">MulticoreParam</span>(<span class="dv">6</span>)</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>sData <span class="ot">&lt;-</span> MethylAid<span class="sc">::</span><span class="fu">summarize</span>(targets, <span class="at">batchSize=</span><span class="dv">50</span>, <span class="at">base=</span><span class="st">&quot;../GSE116339/IDATs&quot;</span>)</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="fu">save</span>(sData, <span class="at">file=</span><span class="st">&quot;./data/sData_example_small.Rdata&quot;</span>)</span></code></pre></div>
<p>After this <code>summarize()</code>, the Shiny web application can be
launched to <code>visualize()</code> the data and identify outliers. In
this instance, there are no apparent outliers, but if found they can be
removed from the dataset. To easier visualise if your data conforms to
typical patterns, you can utilise the <a
href="http://bioconductor.org/packages/release/data/experiment/html/MethylAidData.html"><strong>MethylAidData</strong></a>
package alongside the <code>background</code> option. Darker blue
colours indicate regions where many observations are found in the
example data, and can serve as a guide for expected peak areas.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">library</span>(MethylAidData)</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="fu">data</span>(exampleDataLarge)</span></code></pre></div>
<p>List of QC probes</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>qcProbes <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>  <span class="at">BSI =</span> <span class="st">&quot;^BISULFITE CONVERSION I$&quot;</span>,</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>  <span class="at">BSII =</span> <span class="st">&quot;^BISULFITE CONVERSION II$&quot;</span>,</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>  <span class="at">EC =</span> <span class="st">&quot;^EXTENSION$&quot;</span>,</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>  <span class="at">SPI =</span> <span class="st">&quot;^SPECIFICITY I$&quot;</span>,</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>  <span class="at">HYB =</span> <span class="st">&quot;^HYBRIDIZATION$&quot;</span>,</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>  <span class="at">NP =</span> <span class="st">&quot;^NON-POLYMORPHIC$&quot;</span>,</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>  <span class="at">SPII =</span> <span class="st">&quot;^SPECIFICITY II$&quot;</span>,</span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>  <span class="at">TR =</span> <span class="st">&quot;^TARGET REMOVAL$&quot;</span>,</span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>  <span class="at">SC =</span> <span class="st">&quot;^STAINING$&quot;</span>,</span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>  <span class="at">NC =</span> <span class="st">&quot;^NEGATIVE$&quot;</span></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a>)</span></code></pre></div>
<p>Read in function to make MethylAid plots</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./data/MethylAid_plots.R&quot;</span>)</span></code></pre></div>
<div id="plots" class="section level2">
<h2>Plots</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">plotMU</span>(<span class="at">object =</span> sData, <span class="at">threshold =</span> <span class="dv">10</span>, <span class="at">col=</span><span class="st">&quot;plate&quot;</span>)</span></code></pre></div>
<pre><code>## Warning: `aes_string()` was deprecated in ggplot2 3.0.0.
## ℹ Please use tidy evaluation idioms with `aes()`.
## ℹ See also `vignette(&quot;ggplot2-in-packages&quot;)` for more information.
## [90mThis warning is displayed once every 8 hours.[39m
## [90mCall `lifecycle::last_lifecycle_warnings()` to see where this warning was generated.[39m</code></pre>
<pre><code>## Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
## ℹ Please use `linewidth` instead.
## [90mThis warning is displayed once every 8 hours.[39m
## [90mCall `lifecycle::last_lifecycle_warnings()` to see where this warning was generated.[39m</code></pre>
<p><img src="02_SampleQualityControl_files/figure-html/207qcplots-1.png" width="672" /></p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="fu">plotOP</span>(<span class="at">object =</span> sData, <span class="at">threshold =</span> <span class="dv">12</span>, <span class="at">col=</span><span class="st">&quot;plate&quot;</span>)</span></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/207qcplots-2.png" width="672" /></p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="fu">plotBS</span>(<span class="at">object =</span> sData, <span class="at">threshold =</span> <span class="fl">11.75</span>, <span class="at">col=</span><span class="st">&quot;plate&quot;</span>)</span></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/207qcplots-3.png" width="672" /></p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="fu">plotHC</span>(<span class="at">object =</span> sData, <span class="at">threshold =</span> <span class="fl">12.75</span>, <span class="at">col=</span><span class="st">&quot;plate&quot;</span>)</span></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/207qcplots-4.png" width="672" /></p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="fu">plotDP</span>(<span class="at">object =</span> sData, <span class="at">threshold =</span> <span class="fl">0.95</span>, <span class="at">col=</span><span class="st">&quot;plate&quot;</span>)</span></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/207qcplots-5.png" width="672" /></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>thresholds <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">10</span>, <span class="dv">12</span>, <span class="fl">11.75</span>, <span class="fl">12.75</span>, <span class="fl">0.95</span>)</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>outliers_auto <span class="ot">&lt;-</span> <span class="fu">get_outliers</span>(<span class="at">object =</span> sData, <span class="at">thresholds =</span> thresholds)</span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a>outliers_auto</span></code></pre></div>
<pre><code>##                                   MU    OP    BS   HC    DP
## GSM3228562_200550980002_R01C01 FALSE FALSE FALSE TRUE FALSE
## GSM3228577_200550980025_R01C01 FALSE FALSE FALSE TRUE FALSE</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>outliers <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">rownames</span>(outliers_auto), <span class="dv">1</span>, <span class="dv">10</span>)</span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>outliers</span></code></pre></div>
<pre><code>## [1] &quot;GSM3228562&quot; &quot;GSM3228577&quot;</code></pre>
<p>Remove outliers</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>targets <span class="ot">&lt;-</span> targets <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>geo_accession <span class="sc">%in%</span> outliers)</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a><span class="fu">dim</span>(targets)</span></code></pre></div>
<pre><code>## [1] 62 13</code></pre>
<hr />
</div>
</div>
<div id="creating-an-rgset" class="section level1">
<h1>Creating an <code>RGset</code></h1>
<p>For the rest of the pipeline, our data will need to be available as
an <code>RGChannelSetExtended</code> object. Reading in large numbers of
IDAT files is memory-intensive and time-consuming. Therefore, our <a
href="https://github.com/molepi/DNAmArray"><strong>DNAmArray</strong></a>
package offers the <code>read.metharray.exp.par()</code> function, which
distributes the IDAT files to each of the workers registered using <a
href="https://bioconductor.org/packages/release/bioc/html/BiocParallel.html"><strong>BiocParallel</strong></a>.
It then passes them in batches to <code>read.metharray.exp()</code> from
<a
href="http://bioconductor.org/packages/minfi"><strong>minfi</strong></a>
(Feinberg, 2014) and combines the returned <code>RGset</code>
objects.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="fu">library</span>(DNAmArray)</span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a></span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a>.guessArrayTypes <span class="ot">&lt;-</span> <span class="cf">function</span>(nProbes) {</span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a>  arrayAnnotation <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">array =</span> <span class="st">&quot;IlluminaHumanMethylationEPIC&quot;</span>, <span class="at">annotation =</span> <span class="st">&quot;20a1.hg38&quot;</span>)</span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a>  arrayAnnotation</span>
<span id="cb20-6"><a href="#cb20-6" tabindex="-1"></a>}</span>
<span id="cb20-7"><a href="#cb20-7" tabindex="-1"></a></span>
<span id="cb20-8"><a href="#cb20-8" tabindex="-1"></a><span class="fu">environment</span>(.guessArrayTypes) <span class="ot">&lt;-</span> <span class="fu">asNamespace</span>(<span class="st">&#39;minfi&#39;</span>)</span>
<span id="cb20-9"><a href="#cb20-9" tabindex="-1"></a><span class="fu">assignInNamespace</span>(<span class="st">&quot;.guessArrayTypes&quot;</span>, .guessArrayTypes, <span class="at">ns =</span> <span class="st">&quot;minfi&quot;</span>)</span>
<span id="cb20-10"><a href="#cb20-10" tabindex="-1"></a><span class="fu">register</span>(<span class="fu">MulticoreParam</span>(<span class="dv">6</span>))</span>
<span id="cb20-11"><a href="#cb20-11" tabindex="-1"></a>RGset <span class="ot">&lt;-</span> <span class="fu">read.metharray.exp</span>(<span class="at">base=</span><span class="st">&quot;../GSE116339/IDATs/&quot;</span>, targets, <span class="at">verbose=</span><span class="cn">FALSE</span>, <span class="at">extended=</span><span class="cn">TRUE</span>)</span>
<span id="cb20-12"><a href="#cb20-12" tabindex="-1"></a></span>
<span id="cb20-13"><a href="#cb20-13" tabindex="-1"></a><span class="fu">library</span>(qs)</span>
<span id="cb20-14"><a href="#cb20-14" tabindex="-1"></a><span class="fu">qsave</span>(RGset, <span class="at">file =</span> <span class="st">&quot;./data/RGset_example_small.qs&quot;</span>, <span class="at">preset =</span> <span class="st">&quot;fast&quot;</span>)</span></code></pre></div>
<p>Reading data in parallel is subject to errors and debugging is often
difficult. Recently, <a
href="https://bioconductor.org/packages/release/bioc/html/BiocParallel.html"><strong>BiocParallel</strong></a>
has been extended with a comprehensive set of functions for debugging on
various parallel architectures. If problems arise, we recommend using
<code>BatchJobsParam()</code> with the <code>log=TRUE</code> option in
order to facilitate resolution.</p>
<p>Our data is now an <code>RGset</code> object that can used for
visualization. You can see below that inside this object the
<code>colData</code> holds the same information as <code>targets</code>,
and that there are 5 <code>assay</code> layers. The annotation
information tells us that the methylation was measured using a EPIC
array and that hg19 is the reference genome.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>RGset</span></code></pre></div>
<pre><code>## class: RGChannelSetExtended 
## dim: 1052641 62 
## metadata(0):
## assays(5): Green Red GreenSD RedSD NBeads
## rownames(1052641): 1600101 1600111 ... 99810990 99810992
## rowData names(0):
## colnames(62): GSM3228563_200550980002_R02C01 GSM3228564_200550980002_R03C01 ... GSM3228624_200590490052_R03C01
##   GSM3228625_200590490052_R04C01
## colData names(14): sample_ID geo_accession ... Basename filenames
## Annotation
##   array: IlluminaHumanMethylationEPIC
##   annotation: 20a1.hg38</code></pre>
<hr />
</div>
<div id="beta-values" class="section level1">
<h1>Beta values</h1>
<p>In order to further visualize the data, we store the beta values
using the <code>getBeta()</code> function from <a
href="https://bioconductor.org/packages/rel%20ease/bioc/html/minfi.html"><strong>minfi</strong></a>
(Feinberg, 2014). The <code>type="Illumina"</code> option adds 100 to
the denominator of the beta-value calculation, preventing NA values
being recorded when the methylated and unmethylated signal are both
0.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a>betas <span class="ot">&lt;-</span> <span class="fu">getBeta</span>(RGset, <span class="at">type=</span><span class="st">&quot;Illumina&quot;</span>)</span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a><span class="fu">dim</span>(betas)</span></code></pre></div>
<pre><code>## [1] 866836     62</code></pre>
<hr />
</div>
<div id="beta-density-plots" class="section level1">
<h1>Beta density plots</h1>
<p>Using <code>densityPlot()</code> from <a
href="https://bioconductor.org/packages/release/bioc/html/minfi.html"><strong>minfi</strong></a>
(Feinberg, 2014), we can visualize the per sample average beta-value
distribution. This gives us a global impression of the data and allows
us to identify possible anomalous samples. We expect this distribution
to be bimodal with the peaks representing methylated and unmethylated
signals. Any centre peaks should be further investigated for problems,
such as ambiguous mapping.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a><span class="fu">library</span>(minfi)</span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a><span class="fu">densityPlot</span>(RGset, </span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a>            <span class="at">xlab=</span><span class="st">&quot;Beta values&quot;</span>) </span></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/213betaplot-1.png" width="672" /></p>
<p>There is one sample with a strange distribution. We can identify and
remove it.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>beta_outlier <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">colSums</span>(betas <span class="sc">&gt;</span> <span class="fl">0.3</span> <span class="sc">&amp;</span> betas <span class="sc">&lt;</span> <span class="fl">0.5</span>) <span class="sc">&gt;</span> <span class="dv">120000</span>, T, F)</span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a>RGset<span class="sc">$</span>beta_outlier <span class="ot">&lt;-</span> beta_outlier</span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a></span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a><span class="fu">densityPlot</span>(RGset, </span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a>            <span class="at">main=</span><span class="st">&quot;Beta density plot&quot;</span>, </span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a>            <span class="at">xlab=</span><span class="st">&quot;Beta values&quot;</span>,</span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a>            <span class="at">sampGroups =</span> beta_outlier) </span></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a>RGset <span class="ot">&lt;-</span> RGset[,<span class="sc">!</span><span class="fu">colnames</span>(RGset) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;GSM3228582_200550980025_R06C01&#39;</span>)]</span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a>targets <span class="ot">&lt;-</span> targets <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>Basename <span class="sc">==</span> <span class="st">&#39;GSM3228582_200550980025_R06C01_Grn.idat&#39;</span>)</span></code></pre></div>
<p>For this data, the density plot is now clearly bimodal with no
obvious outliers.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a><span class="fu">densityPlot</span>(RGset, </span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a>            <span class="at">xlab=</span><span class="st">&quot;Beta values&quot;</span>) </span></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/215betaplot-1.png" width="672" /></p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a>betas <span class="ot">&lt;-</span> <span class="fu">getBeta</span>(RGset, <span class="at">type=</span><span class="st">&quot;Illumina&quot;</span>)</span></code></pre></div>
<hr />
</div>
<div id="principal-components-plot" class="section level1">
<h1>Principal components plot</h1>
<p>Using the <code>prcomp_irlba()</code> function from <a
href="https://cran.r-project.org/web/packages/irlba/index.html"><strong>irlba</strong></a>
we can calculate principal components. By assessing the amount of
variance explained by these and visualising them, we can better
interpret the data. The package <a
href="https://cran.r-project.org/web/packages/ggfortify/index.html"><strong>ggfortify</strong></a>
helps <a
href="https://ggplot2.tidyverse.org/"><strong>ggplot2</strong></a>
interpret PCA objects, allowing <code>prcomp</code> objects to be passed
to the <code>autoplot()</code> function.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a>tbetas <span class="ot">&lt;-</span> <span class="fu">t</span>(betas)</span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(tbetas, <span class="at">center =</span> <span class="cn">TRUE</span>, <span class="at">rank. =</span> <span class="dv">10</span>)</span>
<span id="cb30-3"><a href="#cb30-3" tabindex="-1"></a></span>
<span id="cb30-4"><a href="#cb30-4" tabindex="-1"></a><span class="fu">summary</span>(pca)</span></code></pre></div>
<pre><code>## Importance of first k=10 (out of 61) components:
##                            PC1      PC2     PC3    PC4     PC5     PC6    PC7     PC8     PC9    PC10
## Standard deviation     37.9256 13.99390 10.0197 8.1650 7.65359 7.26825 5.3510 5.13386 4.97126 4.83064
## Proportion of Variance  0.5373  0.07316  0.0375 0.0249 0.02188 0.01973 0.0107 0.00985 0.00923 0.00872
## Cumulative Proportion   0.5373  0.61049  0.6480 0.6729 0.69478 0.71452 0.7252 0.73506 0.74429 0.75301</code></pre>
<p>In this instance, our principal components explain over 50% of the
variance in the data and there is evidence of clustering in the plot. By
passing the original data to the <code>autoplot()</code> function using
the <code>data</code> option, we can investigate clustering by colouring
candidate variables. Our data has yet to undergo probe masking, which
removes sex chromosome data, so groups of principal components according
to whether an individual is male or female appear in the plot.</p>
<p>Calculate variance explained by each PC</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a>var_explained <span class="ot">=</span></span>
<span id="cb32-2"><a href="#cb32-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">PC =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(targets),</span>
<span id="cb32-3"><a href="#cb32-3" tabindex="-1"></a>             <span class="at">var_explained =</span> pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span> <span class="sc">/</span> <span class="fu">sum</span>(pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>))[<span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(pca<span class="sc">$</span>x),]</span>
<span id="cb32-4"><a href="#cb32-4" tabindex="-1"></a></span>
<span id="cb32-5"><a href="#cb32-5" tabindex="-1"></a>var_explained</span></code></pre></div>
<pre><code>##    PC var_explained
## 1   1   0.537331692
## 2   2   0.073156791
## 3   3   0.037504787
## 4   4   0.024904938
## 5   5   0.021883049
## 6   6   0.019734989
## 7   7   0.010696748
## 8   8   0.009846137
## 9   9   0.009232292
## 10 10   0.008717390</code></pre>
<div id="screeplot" class="section level2">
<h2>Screeplot</h2>
<p>Plot a screeplot to visualize variance explained</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a>var_explained <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>PC, <span class="at">y=</span>var_explained)) <span class="sc">+</span></span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb34-3"><a href="#cb34-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color=</span><span class="st">&#39;grey5&#39;</span>, <span class="at">fill=</span><span class="st">&#39;#6DACBC&#39;</span>, <span class="at">shape=</span><span class="dv">21</span>, <span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span> </span>
<span id="cb34-4"><a href="#cb34-4" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks=</span><span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(pca<span class="sc">$</span>x)) <span class="sc">+</span></span>
<span id="cb34-5"><a href="#cb34-5" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Principal Component&quot;</span>) <span class="sc">+</span> </span>
<span id="cb34-6"><a href="#cb34-6" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Proportion of variance explained&quot;</span>) <span class="sc">+</span></span>
<span id="cb34-7"><a href="#cb34-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>()</span></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
</div>
<div id="heatmap" class="section level2">
<h2>Heatmap</h2>
<p>Remove constant variables for heatmap</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a>plot_vars <span class="ot">&lt;-</span> <span class="fu">apply</span>(targets, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">sd</span>(<span class="fu">as.numeric</span>(<span class="fu">factor</span>(x)), <span class="at">na.rm=</span>T))</span>
<span id="cb35-2"><a href="#cb35-2" tabindex="-1"></a></span>
<span id="cb35-3"><a href="#cb35-3" tabindex="-1"></a>plot_vars <span class="ot">&lt;-</span> <span class="fu">names</span>(plot_vars[<span class="sc">!</span>plot_vars <span class="sc">%in%</span> <span class="fu">c</span>(<span class="cn">NA</span>, <span class="dv">0</span>)])</span></code></pre></div>
<p>Make heatmap data frame</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a>heatmap_df <span class="ot">&lt;-</span> targets <span class="sc">%&gt;%</span> </span>
<span id="cb36-2"><a href="#cb36-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">any_of</span>(plot_vars))</span></code></pre></div>
<p>Convert all variables to numeric and calculate correlations between
PCs and variables</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a>heatmap_df <span class="ot">&lt;-</span> <span class="fu">apply</span>(heatmap_df, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">as.numeric</span>(<span class="fu">factor</span>(x)))</span>
<span id="cb37-2"><a href="#cb37-2" tabindex="-1"></a></span>
<span id="cb37-3"><a href="#cb37-3" tabindex="-1"></a>cxy <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">cor</span>(pca<span class="sc">$</span>x, <span class="fu">scale</span>(heatmap_df), <span class="at">use=</span><span class="st">&quot;pairwise.complete.obs&quot;</span>),<span class="dv">2</span>) </span></code></pre></div>
<p>Make a heatmap of these correlations.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a>col_fun <span class="ot">&lt;-</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">1</span>), <span class="fu">c</span>(<span class="st">&quot;#458797&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;#ce67a6&quot;</span>))</span>
<span id="cb38-2"><a href="#cb38-2" tabindex="-1"></a></span>
<span id="cb38-3"><a href="#cb38-3" tabindex="-1"></a><span class="fu">Heatmap</span>(</span>
<span id="cb38-4"><a href="#cb38-4" tabindex="-1"></a>  <span class="fu">t</span>(cxy),                              </span>
<span id="cb38-5"><a href="#cb38-5" tabindex="-1"></a>  <span class="at">col =</span> col_fun,  </span>
<span id="cb38-6"><a href="#cb38-6" tabindex="-1"></a>  <span class="at">border =</span> <span class="st">&#39;grey5&#39;</span>,</span>
<span id="cb38-7"><a href="#cb38-7" tabindex="-1"></a>  <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,            </span>
<span id="cb38-8"><a href="#cb38-8" tabindex="-1"></a>  <span class="at">show_row_dend =</span> <span class="cn">TRUE</span>,             </span>
<span id="cb38-9"><a href="#cb38-9" tabindex="-1"></a>  <span class="at">show_column_dend =</span> <span class="cn">FALSE</span>,      </span>
<span id="cb38-10"><a href="#cb38-10" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&quot;Value&quot;</span>,                 </span>
<span id="cb38-11"><a href="#cb38-11" tabindex="-1"></a>  <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">8</span>), </span>
<span id="cb38-12"><a href="#cb38-12" tabindex="-1"></a>  <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">8</span>), </span>
<span id="cb38-13"><a href="#cb38-13" tabindex="-1"></a>  <span class="at">cell_fun =</span> <span class="cf">function</span>(j, i, x, y, width, height, fill) {</span>
<span id="cb38-14"><a href="#cb38-14" tabindex="-1"></a>    <span class="fu">grid.rect</span>(x, y, width, height, </span>
<span id="cb38-15"><a href="#cb38-15" tabindex="-1"></a>              <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">col =</span> <span class="st">&quot;white&quot;</span>, <span class="at">lwd =</span> <span class="dv">1</span>, <span class="at">fill =</span> <span class="cn">NA</span>))</span>
<span id="cb38-16"><a href="#cb38-16" tabindex="-1"></a>    <span class="fu">grid.text</span>(<span class="fu">ifelse</span>(<span class="fu">abs</span>(<span class="fu">t</span>(cxy)[i,j]) <span class="sc">&gt;</span> <span class="fl">0.4</span>,</span>
<span id="cb38-17"><a href="#cb38-17" tabindex="-1"></a>                     <span class="fu">sprintf</span>(<span class="st">&quot;%.2f&quot;</span>, <span class="fu">round</span>(<span class="fu">t</span>(cxy)[i, j], <span class="dv">2</span>)),</span>
<span id="cb38-18"><a href="#cb38-18" tabindex="-1"></a>                     <span class="st">&quot;&quot;</span>), </span>
<span id="cb38-19"><a href="#cb38-19" tabindex="-1"></a>              x, y, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">6</span>, <span class="at">col =</span> <span class="st">&quot;black&quot;</span>))</span>
<span id="cb38-20"><a href="#cb38-20" tabindex="-1"></a>  }</span>
<span id="cb38-21"><a href="#cb38-21" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
</div>
<div id="plot" class="section level2">
<h2>Plot</h2>
<p>Add PCs to experiment metadata</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a>pc_df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(targets, pca<span class="sc">$</span>x)</span></code></pre></div>
<p>Plot PC1 v. PC2 for variables of interest</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a>pc_df <span class="sc">%&gt;%</span> </span>
<span id="cb40-2"><a href="#cb40-2" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> plate)) <span class="sc">+</span></span>
<span id="cb40-3"><a href="#cb40-3" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb40-4"><a href="#cb40-4" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">paste0</span>(<span class="st">&quot;PC1 (&quot;</span>, <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span>var_explained[<span class="dv">1</span>,<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">&quot;%)&quot;</span>), </span>
<span id="cb40-5"><a href="#cb40-5" tabindex="-1"></a>         <span class="at">y =</span> <span class="fu">paste0</span>(<span class="st">&quot;PC2 (&quot;</span>, <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span>var_explained[<span class="dv">2</span>,<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">&quot;%)&quot;</span>), </span>
<span id="cb40-6"><a href="#cb40-6" tabindex="-1"></a>         <span class="at">color =</span> <span class="st">&quot;Array&quot;</span>) <span class="sc">+</span></span>
<span id="cb40-7"><a href="#cb40-7" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a>pc_df <span class="sc">%&gt;%</span> </span>
<span id="cb41-2"><a href="#cb41-2" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> sex)) <span class="sc">+</span></span>
<span id="cb41-3"><a href="#cb41-3" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb41-4"><a href="#cb41-4" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">paste0</span>(<span class="st">&quot;PC1 (&quot;</span>, <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span>var_explained[<span class="dv">1</span>,<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">&quot;%)&quot;</span>), </span>
<span id="cb41-5"><a href="#cb41-5" tabindex="-1"></a>         <span class="at">y =</span> <span class="fu">paste0</span>(<span class="st">&quot;PC2 (&quot;</span>, <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span>var_explained[<span class="dv">2</span>,<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">&quot;%)&quot;</span>), </span>
<span id="cb41-6"><a href="#cb41-6" tabindex="-1"></a>         <span class="at">color =</span> <span class="st">&quot;Sex&quot;</span>) <span class="sc">+</span></span>
<span id="cb41-7"><a href="#cb41-7" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<hr />
</div>
</div>
<div id="checking-sample-relationships" class="section level1">
<h1>Checking Sample Relationships</h1>
<p><a
href="http://bioconductor.org/packages/release/bioc/html/omicsPrint.html"><strong>omicsPrint</strong></a>
(Van Iterson, 2018) is a package we developed to detect data linkage
errors through inspecting sample relations in multiple omics studies.
Included with the package is the <code>hm450.manifest.pop.GoNL</code>
data, which stores SNP probe information in a <code>GRanges</code> class
object. This is then used to create a subset of the beta values for
genotyping.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a><span class="fu">library</span>(omicsPrint)</span></code></pre></div>
<div id="annotation" class="section level2">
<h2>Annotation</h2>
<p>We use the Zhou manifest, which was pulled recently (2022) from
Zhou’s github: <a href="https://zwdzwd.github.io/InfiniumAnnotation"
class="uri">https://zwdzwd.github.io/InfiniumAnnotation</a></p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a>snp_cpgs <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(</span>
<span id="cb43-2"><a href="#cb43-2" tabindex="-1"></a>  <span class="st">&quot;./data/EPIC.hg19.manifest.pop.tsv.gz&quot;</span>)</span></code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   .default = col_logical(),
##   CpG_chrm = col_character(),
##   CpG_beg = col_double(),
##   CpG_end = col_double(),
##   probeID = col_character()
## )</code></pre>
<pre><code>## See spec(...) for full column specifications.</code></pre>
<p>We are interested in the:</p>
<ul>
<li>cpg - ID of the probe for the CpG</li>
<li>chr - Chromosome where the CpG is located</li>
<li>start - Start position of the CpG</li>
<li>end - End position of the CpG</li>
</ul>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" tabindex="-1"></a>snp_cpgs <span class="ot">&lt;-</span> snp_cpgs <span class="sc">%&gt;%</span> </span>
<span id="cb46-2"><a href="#cb46-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb46-3"><a href="#cb46-3" tabindex="-1"></a>    <span class="at">cpg =</span> probeID,</span>
<span id="cb46-4"><a href="#cb46-4" tabindex="-1"></a>    <span class="at">chr =</span> CpG_chrm,</span>
<span id="cb46-5"><a href="#cb46-5" tabindex="-1"></a>    <span class="at">start =</span> CpG_beg,</span>
<span id="cb46-6"><a href="#cb46-6" tabindex="-1"></a>    <span class="at">end =</span> CpG_end,</span>
<span id="cb46-7"><a href="#cb46-7" tabindex="-1"></a>    MASK_snp5_EUR</span>
<span id="cb46-8"><a href="#cb46-8" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb46-9"><a href="#cb46-9" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb46-10"><a href="#cb46-10" tabindex="-1"></a>    <span class="at">chr =</span> <span class="fu">substr</span>(chr,<span class="dv">4</span>,<span class="dv">5</span>)</span>
<span id="cb46-11"><a href="#cb46-11" tabindex="-1"></a>  )</span>
<span id="cb46-12"><a href="#cb46-12" tabindex="-1"></a></span>
<span id="cb46-13"><a href="#cb46-13" tabindex="-1"></a>snp_cpgs <span class="ot">&lt;-</span> (snp_cpgs <span class="sc">%&gt;%</span> </span>
<span id="cb46-14"><a href="#cb46-14" tabindex="-1"></a>               dplyr<span class="sc">::</span><span class="fu">filter</span>(MASK_snp5_EUR <span class="sc">==</span> <span class="cn">TRUE</span>))<span class="sc">$</span>cpg</span>
<span id="cb46-15"><a href="#cb46-15" tabindex="-1"></a></span>
<span id="cb46-16"><a href="#cb46-16" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">&quot;There are &quot;</span>, <span class="fu">length</span>(snp_cpgs),</span>
<span id="cb46-17"><a href="#cb46-17" tabindex="-1"></a>             <span class="st">&quot; CpGs containing common European SNPs&quot;</span>))</span></code></pre></div>
<pre><code>## [1] &quot;There are 28152 CpGs containing common European SNPs&quot;</code></pre>
<p>Subset betas</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" tabindex="-1"></a>betas_snps <span class="ot">&lt;-</span> betas[snp_cpgs, ]</span>
<span id="cb48-2"><a href="#cb48-2" tabindex="-1"></a><span class="fu">dim</span>(betas_snps)</span></code></pre></div>
<pre><code>## [1] 28152    61</code></pre>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" tabindex="-1"></a><span class="fu">rownames</span>(targets) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;_Grn.idat&quot;</span>, <span class="st">&quot;&quot;</span>, targets<span class="sc">$</span>Basename)</span>
<span id="cb50-2"><a href="#cb50-2" tabindex="-1"></a></span>
<span id="cb50-3"><a href="#cb50-3" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">rownames</span>(targets) <span class="sc">==</span> <span class="fu">colnames</span>(betas_snps))</span></code></pre></div>
<pre><code>## 
## TRUE 
##   61</code></pre>
<div id="assumed-relations" class="section level3">
<h3>Assumed relations</h3>
<p>Create a data frame of all samples against each other e.g.:</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" tabindex="-1"></a>relations <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(</span>
<span id="cb52-2"><a href="#cb52-2" tabindex="-1"></a>  <span class="at">idx =</span> <span class="fu">colnames</span>(betas_snps), </span>
<span id="cb52-3"><a href="#cb52-3" tabindex="-1"></a>  <span class="at">idy =</span> <span class="fu">colnames</span>(betas_snps))</span>
<span id="cb52-4"><a href="#cb52-4" tabindex="-1"></a><span class="fu">head</span>(relations)</span></code></pre></div>
<pre><code>##                              idx                            idy
## 1 GSM3228563_200550980002_R02C01 GSM3228563_200550980002_R02C01
## 2 GSM3228564_200550980002_R03C01 GSM3228563_200550980002_R02C01
## 3 GSM3228565_200550980002_R04C01 GSM3228563_200550980002_R02C01
## 4 GSM3228566_200550980002_R05C01 GSM3228563_200550980002_R02C01
## 5 GSM3228567_200550980002_R06C01 GSM3228563_200550980002_R02C01
## 6 GSM3228568_200550980002_R07C01 GSM3228563_200550980002_R02C01</code></pre>
<p>Save the sample name, so we know which samples should match</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" tabindex="-1"></a>relations<span class="sc">$</span>sample_name_x <span class="ot">&lt;-</span> targets[relations<span class="sc">$</span>idx, <span class="st">&quot;sample_ID&quot;</span>] </span>
<span id="cb54-2"><a href="#cb54-2" tabindex="-1"></a>relations<span class="sc">$</span>sample_name_y <span class="ot">&lt;-</span> targets[relations<span class="sc">$</span>idy, <span class="st">&quot;sample_ID&quot;</span>]</span></code></pre></div>
<p>Create a relation_type variable and set it as Identical if the
samples come from the same individual</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" tabindex="-1"></a>relations<span class="sc">$</span>relation_type <span class="ot">&lt;-</span> <span class="st">&quot;unrelated&quot;</span></span>
<span id="cb55-2"><a href="#cb55-2" tabindex="-1"></a>relations<span class="sc">$</span>relation_type[relations<span class="sc">$</span>sample_name_x <span class="sc">==</span> relations<span class="sc">$</span>sample_name_y] <span class="ot">&lt;-</span> <span class="st">&quot;identical&quot;</span></span>
<span id="cb55-3"><a href="#cb55-3" tabindex="-1"></a></span>
<span id="cb55-4"><a href="#cb55-4" tabindex="-1"></a><span class="fu">table</span>(relations<span class="sc">$</span>relation_type)</span></code></pre></div>
<pre><code>## 
## identical unrelated 
##        61      3660</code></pre>
</div>
<div id="genotyping" class="section level3">
<h3>Genotyping</h3>
<p>Calculate genotype using omicsPrint</p>
<p>The function <code>beta2genotype()</code> then genotypes the
observations by measuring homozygous or heterozygous alleles at these
SNP probes. Lastly <code>alleleSharing()</code> assesses the
relationships between different individuals, which can be unrelated,
twins, or identical. The results can then be visualised using the
<code>inferRelations()</code> function.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" tabindex="-1"></a>genotype <span class="ot">&lt;-</span> <span class="fu">beta2genotype</span>(betas_snps, </span>
<span id="cb57-2"><a href="#cb57-2" tabindex="-1"></a>                          <span class="at">na.rm=</span><span class="cn">TRUE</span>,</span>
<span id="cb57-3"><a href="#cb57-3" tabindex="-1"></a>                          <span class="at">minSep =</span> <span class="fl">0.25</span>,</span>
<span id="cb57-4"><a href="#cb57-4" tabindex="-1"></a>                          <span class="at">minSize =</span> <span class="dv">5</span>,</span>
<span id="cb57-5"><a href="#cb57-5" tabindex="-1"></a>                          <span class="at">centers =</span> <span class="fu">c</span>(<span class="fl">0.2</span>, <span class="fl">0.5</span>, <span class="fl">0.8</span>),</span>
<span id="cb57-6"><a href="#cb57-6" tabindex="-1"></a>                          <span class="at">assayName=</span><span class="cn">NULL</span>)</span>
<span id="cb57-7"><a href="#cb57-7" tabindex="-1"></a><span class="fu">str</span>(genotype)</span></code></pre></div>
<pre><code>##  int [1:977, 1:61] 2 1 1 2 3 1 3 2 2 2 ...
##  - attr(*, &quot;dimnames&quot;)=List of 2
##   ..$ : chr [1:977] &quot;cg19405842&quot; &quot;cg01296877&quot; &quot;cg21783012&quot; &quot;cg10091792&quot; ...
##   ..$ : chr [1:61] &quot;GSM3228563_200550980002_R02C01&quot; &quot;GSM3228564_200550980002_R03C01&quot; &quot;GSM3228565_200550980002_R04C01&quot; &quot;GSM3228566_200550980002_R05C01&quot; ...</code></pre>
<p>Save output</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" tabindex="-1"></a>out <span class="ot">&lt;-</span> <span class="fu">alleleSharing</span>(genotype, <span class="at">relations =</span> relations)</span></code></pre></div>
<pre><code>## Hash relations</code></pre>
<pre><code>## Pruning 977 SNPs ...</code></pre>
<pre><code>## 0 SNPs removed because of low call rate!</code></pre>
<pre><code>## 0 samples removed because too few SNPs called!</code></pre>
<pre><code>## Using 977 polymorphic SNPs to determine allele sharing.</code></pre>
<pre><code>## Running `square` IBS algorithm!</code></pre>
<pre><code>## 62 of 1891 (3.28%) ...</code></pre>
</div>
<div id="identify-mismatches" class="section level3">
<h3>Identify mismatches</h3>
<p>Plot</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" tabindex="-1"></a>mismatches <span class="ot">&lt;-</span> <span class="fu">inferRelations</span>(out, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>##                   Assumed relation
## Predicted relation identical unrelated
##          identical        61         .
##          unrelated         .      1830</code></pre>
<p><img src="02_SampleQualityControl_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" tabindex="-1"></a><span class="fu">str</span>(mismatches)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    0 obs. of  6 variables:
##  $ mean      : num 
##  $ var       : num 
##  $ colnames.x: chr 
##  $ colnames.y: chr 
##  $ relation  : Factor w/ 2 levels &quot;identical&quot;,&quot;unrelated&quot;: 
##  $ predicted : Factor w/ 2 levels &quot;identical&quot;,&quot;unrelated&quot;:</code></pre>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" tabindex="-1"></a><span class="fu">table</span>(mismatches<span class="sc">$</span>relation, mismatches<span class="sc">$</span>predicted)</span></code></pre></div>
<pre><code>##            
##             identical unrelated
##   identical         0         0
##   unrelated         0         0</code></pre>
<p>Since there are no twins or relatives in our data, all observations
are shown as unrelated. In data with sample relationships, this would be
shown in the above graph as green or black clusters (Van Iterson, 2018).
It is important to carry out this type of visualization before
probe-filtering as otherwise the genotyping will be based on very few
SNPs.</p>
<hr />
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
