<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>02_SampleQualityControl.knit</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span. { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DNAmArray</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="01_GettingStarted.html">Getting Started</a>
</li>
<li>
  <a href="02_SampleQualityControl.html">Sample QC</a>
</li>
<li>
  <a href="03_Predict.html">Predict</a>
</li>
<li>
  <a href="04_Normalization.html">Normalize</a>
</li>
<li>
  <a href="05_ProbeQC.html">Probe QC</a>
</li>
<li>
  <a href="06_EWAS.html">Analyse</a>
</li>
<li>
  <a href="07_Interpret.html">Interpret</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">About</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">




</div>


<hr />
<div id="loading-dnam-data" class="section level1">
<h1>Loading DNAm data</h1>
<div id="summarizing-idats" class="section level2">
<h2>Summarizing IDATs</h2>
<p>The <a href="http://bioconductor.org/packages/MethylAid"><strong>MethylAid</strong></a><sup>10</sup> package, which will be used for quality control in this workflow, requires the targets data frame to contain a <code>Basename</code> variable that points to IDAT file names. Sometimes, data comes with a sample sheet to facilitate this, but in this example the information is extracted from the supplementary file column instead.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">targets<span class="op">$</span>Basename &lt;-<span class="st"> </span><span class="kw">substring</span>(targets<span class="op">$</span>supplementary_file,<span class="dv">68</span>,<span class="dv">106</span>)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">head</span>(targets<span class="op">$</span>Basename)</a></code></pre></div>
<pre><code>## [1] &quot;GSM3228562_200550980002_R01C01_Grn.idat&quot;
## [2] &quot;GSM3228563_200550980002_R02C01_Grn.idat&quot;
## [3] &quot;GSM3228564_200550980002_R03C01_Grn.idat&quot;
## [4] &quot;GSM3228565_200550980002_R04C01_Grn.idat&quot;
## [5] &quot;GSM3228566_200550980002_R05C01_Grn.idat&quot;
## [6] &quot;GSM3228567_200550980002_R06C01_Grn.idat&quot;</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw">rownames</span>(targets) &lt;-<span class="st"> </span><span class="kw">sub</span>(<span class="st">&quot;_Grn.idat&quot;</span>, <span class="st">&quot;&quot;</span>, targets<span class="op">$</span>Basename)</a></code></pre></div>
<p>The following sample quality control steps will also require the <a href="https://bioconductor.org/packages/release/bioc/html/BiocParallel.html"><strong>BiocParallel</strong></a> package. Using parallel processing and/or batches will reduce both memory load and run-times when extracting intensities from IDAT files. Please see the <a href="http://bioconductor.org/packages/release/bioc/vignettes/MethylAid/inst/doc/MethylAid.pdf">MethylAid vignette</a> for more details.</p>
<p>The available number of cores is set usnig <code>MulticoreParam</code>, and IDATs are summarized using the <code>summarize</code> function.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">BPPARAM &lt;-<span class="st"> </span><span class="kw">MulticoreParam</span>(<span class="dv">6</span>)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">sData &lt;-<span class="st"> </span>MethylAid<span class="op">::</span><span class="kw">summarize</span>(targets, <span class="dt">batchSize=</span><span class="dv">50</span>, <span class="dt">base=</span><span class="st">&quot;../GSE116339/IDATs&quot;</span>)</a></code></pre></div>
<hr />
</div>
</div>
<div id="creating-an-rgset" class="section level1">
<h1>Creating an <code>RGset</code></h1>
<p>In addition to the <code>sData</code> object, which is used by MethylAid, it is useful to also create an <code>RGChannelSetExtended</code> object, which contains more extensive information on the data. Here, we use <code>read.metharray.exp()</code> from <a href="http://bioconductor.org/packages/minfi"><strong>minfi</strong></a><sup>9</sup> but, since this has not been updated for the more recent arrays, we advise monkey patching the internal <code>.guessArrayTypes</code> function, with the appropriate array and annotation.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">.guessArrayTypes &lt;-<span class="st"> </span><span class="cf">function</span>(nProbes) {</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">  <span class="cf">if</span> (nProbes <span class="op">&gt;</span><span class="st"> </span><span class="dv">850000</span>) {</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">    <span class="co"># EPIC v2 array </span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4">    arrayAnnotation &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">array =</span> <span class="st">&quot;IlluminaHumanMethylationEPIC&quot;</span>, <span class="dt">annotation =</span> <span class="st">&quot;20a1.hg38&quot;</span>)</a>
<a class="sourceLine" id="cb5-5" data-line-number="5">  } <span class="cf">else</span> <span class="cf">if</span> (nProbes <span class="op">&gt;</span><span class="st"> </span><span class="dv">622000</span>) {</a>
<a class="sourceLine" id="cb5-6" data-line-number="6">    <span class="co"># EPIC (850k) array</span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7">    arrayAnnotation &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">array =</span> <span class="st">&quot;IlluminaHumanMethylationEPIC&quot;</span>, <span class="dt">annotation =</span> <span class="st">&quot;ilm10b4.hg19&quot;</span>)</a>
<a class="sourceLine" id="cb5-8" data-line-number="8">  } <span class="cf">else</span> <span class="cf">if</span> (nProbes <span class="op">&gt;</span><span class="st"> </span><span class="dv">480000</span>) {</a>
<a class="sourceLine" id="cb5-9" data-line-number="9">    <span class="co"># 450k array</span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10">    arrayAnnotation &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">array =</span> <span class="st">&quot;IlluminaHumanMethylation450k&quot;</span>, <span class="dt">annotation =</span> <span class="st">&quot;ilmn12.hg19&quot;</span>)</a>
<a class="sourceLine" id="cb5-11" data-line-number="11">  } <span class="cf">else</span> {</a>
<a class="sourceLine" id="cb5-12" data-line-number="12">    <span class="co"># 27k array</span></a>
<a class="sourceLine" id="cb5-13" data-line-number="13">    arrayAnnotation &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">array =</span> <span class="st">&quot;IlluminaHumanMethylation27k&quot;</span>, <span class="dt">annotation =</span> <span class="st">&quot;ilmn12.hg19&quot;</span>)</a>
<a class="sourceLine" id="cb5-14" data-line-number="14">  }</a>
<a class="sourceLine" id="cb5-15" data-line-number="15"></a>
<a class="sourceLine" id="cb5-16" data-line-number="16">  arrayAnnotation</a>
<a class="sourceLine" id="cb5-17" data-line-number="17">}</a>
<a class="sourceLine" id="cb5-18" data-line-number="18"></a>
<a class="sourceLine" id="cb5-19" data-line-number="19"><span class="kw">environment</span>(.guessArrayTypes) &lt;-<span class="st"> </span><span class="kw">asNamespace</span>(<span class="st">&quot;minfi&quot;</span>)</a>
<a class="sourceLine" id="cb5-20" data-line-number="20"><span class="kw">assignInNamespace</span>(<span class="st">&quot;.guessArrayTypes&quot;</span>, .guessArrayTypes, <span class="dt">ns =</span> <span class="st">&quot;minfi&quot;</span>)</a></code></pre></div>
<p>Then, proceed using <code>BiocParallel</code> and <code>minfi</code> to read in the IDAT files.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">RGset &lt;-<span class="st"> </span><span class="kw">read.metharray.exp</span>(<span class="dt">base=</span><span class="st">&quot;../GSE116339/IDATs/&quot;</span>, targets, <span class="dt">verbose=</span><span class="ot">FALSE</span>, <span class="dt">extended=</span><span class="ot">TRUE</span>)</a></code></pre></div>
<p>Reading data in parallel is subject to errors and debugging is often difficult. Recently, <a href="https://bioconductor.org/packages/release/bioc/html/BiocParallel.html"><strong>BiocParallel</strong></a> has been extended with a comprehensive set of functions for debugging on various parallel architectures. If problems arise, we recommend using <code>BatchJobsParam()</code> with the <code>log=TRUE</code> option in order to facilitate resolution.</p>
<p>Our data is now an <code>RGset</code> object that can used for visualization. You can see below that inside this object the <code>colData</code> holds the same information as <code>targets</code>, and that there are 5 <code>assay</code> layers. The annotation information tells us that the methylation was measured using a EPIC array and that hg19 is the reference genome.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">RGset</a></code></pre></div>
<pre><code>## class: RGChannelSetExtended 
## dim: 1052641 679 
## metadata(0):
## assays(5): Green Red GreenSD RedSD NBeads
## rownames(1052641): 1600101 1600111 ... 99810990 99810992
## rowData names(0):
## colnames(679): GSM3228562_200550980002_R01C01
##   GSM3228563_200550980002_R02C01 ... GSM3229239_200598350080_R03C01
##   GSM3229240_200594740075_R08C01
## colData names(14): sample_ID geo_accession ... Basename filenames
## Annotation
##   array: IlluminaHumanMethylationEPIC
##   annotation: 20a1.hg38</code></pre>
<hr />
</div>
<div id="beta-values" class="section level1">
<h1>Beta values</h1>
<p>In order to further visualize the data, we store the beta values using the <code>getBeta()</code> function from <a href="https://bioconductor.org/packages/release/bioc/html/minfi.html"><strong>minfi</strong></a>. The <code>type=&quot;Illumina&quot;</code> option adds 100 to the denominator of the beta-value calculation, preventing NA values being recorded when the methylated and unmethylated signal are both 0.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">betas &lt;-<span class="st"> </span><span class="kw">getBeta</span>(RGset, <span class="dt">type=</span><span class="st">&quot;Illumina&quot;</span>)</a></code></pre></div>
<p>On some environments, this function returns a list. In that case, the beta-values can be extracted using <code>betas$completeObs</code>.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="cf">if</span>(<span class="kw">is.list</span>(betas)){</a>
<a class="sourceLine" id="cb10-2" data-line-number="2">  betas &lt;-<span class="st"> </span>betas<span class="op">$</span>completeObs</a>
<a class="sourceLine" id="cb10-3" data-line-number="3">}</a></code></pre></div>
<p>It can also be advisable to round the beta values to 4 significant figures, which both reduces the data size and provides a realistic measure of the available precision.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">betas &lt;-<span class="st"> </span><span class="kw">round</span>(betas, <span class="dv">4</span>)</a></code></pre></div>
<p>Now, <code>betas</code> contains beta values for each measured probe and sample.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw">dim</span>(betas)</a></code></pre></div>
<pre><code>## [1] 866836    679</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">betas[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</a></code></pre></div>
<pre><code>##            GSM3228562_200550980002_R01C01 GSM3228563_200550980002_R02C01
## cg18478105                         0.0504                         0.0000
## cg09835024                         0.0745                         0.1027
## cg14361672                         0.6650                         0.6541
## cg01763666                         0.6506                         0.6182
## cg12950382                         0.8633                         0.8001
##            GSM3228564_200550980002_R03C01 GSM3228565_200550980002_R04C01
## cg18478105                         0.0432                         0.0396
## cg09835024                         0.0548                         0.0499
## cg14361672                         0.6505                         0.6050
## cg01763666                         0.6523                         0.6663
## cg12950382                         0.8153                         0.8313
##            GSM3228566_200550980002_R05C01
## cg18478105                         0.0454
## cg09835024                         0.0750
## cg14361672                         0.7440
## cg01763666                         0.6309
## cg12950382                         0.7643</code></pre>
<hr />
</div>
<div id="methylaid" class="section level1">
<h1>MethylAid</h1>
<div id="control-probes" class="section level2">
<h2>Control probes</h2>
<p><code>MethylAid</code> plots to visualize sample quality are included in the <code>DNAmArray</code> package and their use is outlined below.</p>
<p>They rely on the following <code>qcProbes</code> object that specifies the control probes.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">qcProbes =<span class="st"> </span><span class="kw">list</span>(</a>
<a class="sourceLine" id="cb16-2" data-line-number="2">  <span class="dt">BSI =</span> <span class="st">&quot;^BISULFITE CONVERSION I$&quot;</span>,</a>
<a class="sourceLine" id="cb16-3" data-line-number="3">  <span class="dt">BSII =</span> <span class="st">&quot;^BISULFITE CONVERSION II$&quot;</span>,</a>
<a class="sourceLine" id="cb16-4" data-line-number="4">  <span class="dt">EC =</span> <span class="st">&quot;^EXTENSION$&quot;</span>,</a>
<a class="sourceLine" id="cb16-5" data-line-number="5">  <span class="dt">SPI =</span> <span class="st">&quot;^SPECIFICITY I$&quot;</span>,</a>
<a class="sourceLine" id="cb16-6" data-line-number="6">  <span class="dt">HYB =</span> <span class="st">&quot;^HYBRIDIZATION$&quot;</span>,</a>
<a class="sourceLine" id="cb16-7" data-line-number="7">  <span class="dt">NP =</span> <span class="st">&quot;^NON-POLYMORPHIC$&quot;</span>,</a>
<a class="sourceLine" id="cb16-8" data-line-number="8">  <span class="dt">SPII =</span> <span class="st">&quot;^SPECIFICITY II$&quot;</span>,</a>
<a class="sourceLine" id="cb16-9" data-line-number="9">  <span class="dt">TR =</span> <span class="st">&quot;^TARGET REMOVAL$&quot;</span>,</a>
<a class="sourceLine" id="cb16-10" data-line-number="10">  <span class="dt">SC =</span> <span class="st">&quot;^STAINING$&quot;</span>,</a>
<a class="sourceLine" id="cb16-11" data-line-number="11">  <span class="dt">NC =</span> <span class="st">&quot;^NEGATIVE$&quot;</span></a>
<a class="sourceLine" id="cb16-12" data-line-number="12">)</a></code></pre></div>
<p>They also rely on array specific thresholds, which we advise to set on the basis of the following table:</p>
<table style="border-collapse: collapse; width: 100%; text-align: center;">
<thead>
<tr>
<th style="padding: 8px; border-bottom: 2px solid #ddd;">
Array
</th>
<th style="padding: 8px; border-bottom: 2px solid #ddd;">
MU
</th>
<th style="padding: 8px; border-bottom: 2px solid #ddd;">
OP
</th>
<th style="padding: 8px; border-bottom: 2px solid #ddd;">
BS
</th>
<th style="padding: 8px; border-bottom: 2px solid #ddd;">
HC
</th>
<th style="padding: 8px; border-bottom: 2px solid #ddd;">
DP
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="padding: 8px;">
450k
</td>
<td style="padding: 8px;">
10.5
</td>
<td style="padding: 8px;">
11.75
</td>
<td style="padding: 8px;">
12.75
</td>
<td style="padding: 8px;">
13.25
</td>
<td style="padding: 8px;">
0.95
</td>
</tr>
<tr>
<td style="padding: 8px;">
EPIC
</td>
<td style="padding: 8px;">
10
</td>
<td style="padding: 8px;">
12
</td>
<td style="padding: 8px;">
11.75
</td>
<td style="padding: 8px;">
12.75
</td>
<td style="padding: 8px;">
0.95
</td>
</tr>
<tr>
<td style="padding: 8px;">
EPIC v2
</td>
<td style="padding: 8px;">
10.5
</td>
<td style="padding: 8px;">
12.5
</td>
<td style="padding: 8px;">
12.75
</td>
<td style="padding: 8px;">
13.25
</td>
<td style="padding: 8px;">
0.95
</td>
</tr>
</tbody>
</table>
<p>The example data uses the EPIC array and the following thresholds</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">thresholds &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">10</span>, <span class="dv">12</span>, <span class="fl">11.75</span>, <span class="fl">12.75</span>, <span class="fl">0.95</span>)</a></code></pre></div>
<hr />
</div>
<div id="plots" class="section level2">
<h2>Plots</h2>
<p>All plots can be coloured by a variable in <code>targets</code>, specified by the <code>col</code> argument.</p>
<p><code>plotMU</code> plots a rotated MU plot. This shows the median methylated and unmethylated log2 intensities per sample, and flagging low-intensity outliers.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="kw">plotMU</span>(<span class="dt">object =</span> sData, <span class="dt">threshold =</span> thresholds[<span class="dv">1</span>], <span class="dt">col=</span><span class="st">&quot;plate&quot;</span>)</a></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/212plotMU-1.png" width="672" /></p>
<hr />
<p><code>plotOP</code> plots the sample-dependent overall quality control on the basis of non-polymorphic (NP) control probes.</p>
<p>NP control probes query each of the four nucleotides in a non-polymorphic region of the bisulfite genome. Signal intensity from these probes in the red (A and T) and green (C and G) channels can then be used to test overall performance of the assay, from amplification to detection. Intensity should be high in the red channel for NP probes querying A and T nucleotides, and high in the green channel for NP probes querying G and C nucleotides. The intensities for the relevant channel for NP probes are then combined and plotted per sample, where they should be above the specific threshold.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw">plotOP</span>(<span class="dt">object =</span> sData, <span class="dt">threshold =</span> thresholds[<span class="dv">2</span>], <span class="dt">col=</span><span class="st">&quot;plate&quot;</span>)</a></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/213plotOP-1.png" width="672" /></p>
<hr />
<p><code>plotBS</code> assesses bisulfite conversion efficiency using control probes, identifying poorly converted samples. Type I BC control probes monitor the efficiency of the bisulfite conversion. If the conversion reaction was successful, the ‘C’ (converted) probes will be extended, but if the sample has unconverted DNA, the ‘U’ (unconverted) probes are extended instead.</p>
<p>Performance of the BC control probes C1 and C2 is monitored in the green channel, and C3 and C4 are monitored in the red channel. The intensities for the relevant channels are then combined for all BC control probes and plotted per sample to ensure they are above the specified threshold.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="kw">plotBS</span>(<span class="dt">object =</span> sData, <span class="dt">threshold =</span> thresholds[<span class="dv">3</span>], <span class="dt">col=</span><span class="st">&quot;plate&quot;</span>)</a></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/214plotBS-1.png" width="672" /></p>
<hr />
<p><code>plotHC</code> uses hybridization controls to assess the hybridization step using synthetic targets, which complement the array perfectly. These synthetic targets are present in the hybridization buffer (RA1) at three concentration levels, and their performance is only monitored in the green channel. The difference in green intensity between the high (H) and low (L) concentration is combined and can be plotted to ensure it is above the specified threshold.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="kw">plotHC</span>(<span class="dt">object =</span> sData, <span class="dt">threshold =</span> thresholds[<span class="dv">4</span>], <span class="dt">col=</span><span class="st">&quot;plate&quot;</span>)</a></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/215plotHC-1.png" width="672" /></p>
<hr />
<p><code>plotDP</code> is the last MethylAid plot we provide. This function instead takes the <code>RGset</code> as input, to enable the user to specify both how visually distinct from the background they consider a detected signal (<code>detP</code> argument is the detection p-value) and what proportion of probes they want to be meeting this criterion (<code>threshold</code> is this proportion). The default we suggest for all arrays is 0.01 (detP) and 0.95 (threshold), but this can be adjusted.</p>
<p>This plot uses negative control probes, which are randomly permuted sequences that should not hybridize to the DNA template. The mean signal of these probes defines the background signal. This plot therefore show the fraction of probe visually distinct from the background signal in each sample, coloured by array number.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1"><span class="kw">plotDP</span>(<span class="dt">object =</span> RGset, <span class="dt">threshold =</span> <span class="fl">0.95</span>, <span class="dt">detP=</span><span class="fl">0.01</span>, <span class="dt">col=</span><span class="st">&quot;plate&quot;</span>)</a></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/216plotDP-1.png" width="672" /></p>
<hr />
</div>
<div id="outliers" class="section level2">
<h2>Outliers</h2>
<p>On the basis of the above thresholds, outliers can be extracted using the <code>get_outliers</code> function. The resulting table shows an indicator of which quality control step was failed.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">outliers_auto &lt;-<span class="st"> </span><span class="kw">get_outliers</span>(<span class="dt">object =</span> sData, <span class="dt">thresholds =</span> thresholds)</a>
<a class="sourceLine" id="cb23-2" data-line-number="2">outliers_auto</a></code></pre></div>
<pre><code>##                                   MU    OP    BS   HC    DP
## GSM3228562_200550980002_R01C01 FALSE FALSE FALSE TRUE FALSE
## GSM3228577_200550980025_R01C01 FALSE FALSE FALSE TRUE FALSE
## GSM3228637_200590490062_R01C01 FALSE FALSE FALSE TRUE FALSE
## GSM3228716_200594740039_R01C01 FALSE FALSE FALSE TRUE FALSE
## GSM3228903_200598350005_R01C01 FALSE FALSE FALSE TRUE FALSE</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1">outliers &lt;-<span class="st"> </span><span class="kw">substr</span>(<span class="kw">rownames</span>(outliers_auto), <span class="dv">1</span>, <span class="dv">10</span>)</a></code></pre></div>
<p>Outliers can then be removed from <code>targets</code> before proceding.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1">targets &lt;-<span class="st"> </span>targets <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="op">!</span>geo_accession <span class="op">%in%</span><span class="st"> </span>outliers)</a>
<a class="sourceLine" id="cb26-2" data-line-number="2"><span class="kw">dim</span>(targets)</a></code></pre></div>
<pre><code>## [1] 674  13</code></pre>
<hr />
</div>
</div>
<div id="beta-density-plots" class="section level1">
<h1>Beta density plots</h1>
<p>Using <code>densityPlot()</code> from <a href="https://bioconductor.org/packages/release/bioc/html/minfi.html"><strong>minfi</strong></a>, we can visualize the per sample average beta-value distribution. This gives us a global impression of the data and allows us to identify possible anomalous samples. We expect this distribution to be bimodal with the peaks representing methylated and unmethylated signals. Any centre peaks should be further investigated for problems, such as ambiguous mapping.</p>
<p>Centre peaks can be identified using a <code>beta_outlier</code> object, which flags any samples with high beta values in the central part of the distribution. Alternatively, a named boolean vector of the samples can be used to flag a specific sample.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1">beta_outlier &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">colSums</span>(betas <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.2</span> <span class="op">&amp;</span><span class="st"> </span>betas <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.5</span>) <span class="op">&gt;</span><span class="st"> </span><span class="dv">100000</span>, T, F)</a>
<a class="sourceLine" id="cb28-2" data-line-number="2">RGset<span class="op">$</span>beta_outlier &lt;-<span class="st"> </span>beta_outlier</a>
<a class="sourceLine" id="cb28-3" data-line-number="3"></a>
<a class="sourceLine" id="cb28-4" data-line-number="4">beta_outlier[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</a></code></pre></div>
<pre><code>## GSM3228562_200550980002_R01C01 GSM3228563_200550980002_R02C01 
##                           TRUE                           TRUE 
## GSM3228564_200550980002_R03C01 GSM3228565_200550980002_R04C01 
##                           TRUE                           TRUE 
## GSM3228566_200550980002_R05C01 
##                           TRUE</code></pre>
<p>These are then coloured in the beta values plot.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" data-line-number="1"><span class="kw">densityPlot</span>(RGset, </a>
<a class="sourceLine" id="cb30-2" data-line-number="2">            <span class="dt">main=</span><span class="st">&quot;Beta density plot&quot;</span>, </a>
<a class="sourceLine" id="cb30-3" data-line-number="3">            <span class="dt">xlab=</span><span class="st">&quot;Beta values&quot;</span>,</a>
<a class="sourceLine" id="cb30-4" data-line-number="4">            <span class="dt">sampGroups =</span> beta_outlier) </a></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/220betaplot-1.png" width="672" /></p>
<p>Any flagged samples can then be removed using the sample name.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1">flagged_samples &lt;-<span class="st"> </span><span class="kw">names</span>(beta_outlier[beta_outlier <span class="op">==</span><span class="st"> </span>T])</a>
<a class="sourceLine" id="cb31-2" data-line-number="2">flagged_samples</a></code></pre></div>
<pre><code>##   [1] &quot;GSM3228562_200550980002_R01C01&quot; &quot;GSM3228563_200550980002_R02C01&quot;
##   [3] &quot;GSM3228564_200550980002_R03C01&quot; &quot;GSM3228565_200550980002_R04C01&quot;
##   [5] &quot;GSM3228566_200550980002_R05C01&quot; &quot;GSM3228567_200550980002_R06C01&quot;
##   [7] &quot;GSM3228568_200550980002_R07C01&quot; &quot;GSM3228569_200550980002_R08C01&quot;
##   [9] &quot;GSM3228570_200550980010_R01C01&quot; &quot;GSM3228571_200550980010_R03C01&quot;
##  [11] &quot;GSM3228572_200550980010_R04C01&quot; &quot;GSM3228573_200550980010_R05C01&quot;
##  [13] &quot;GSM3228574_200550980010_R06C01&quot; &quot;GSM3228575_200550980010_R07C01&quot;
##  [15] &quot;GSM3228576_200550980010_R08C01&quot; &quot;GSM3228577_200550980025_R01C01&quot;
##  [17] &quot;GSM3228578_200550980025_R02C01&quot; &quot;GSM3228579_200550980025_R03C01&quot;
##  [19] &quot;GSM3228580_200550980025_R04C01&quot; &quot;GSM3228581_200550980025_R05C01&quot;
##  [21] &quot;GSM3228582_200550980025_R06C01&quot; &quot;GSM3228583_200550980025_R07C01&quot;
##  [23] &quot;GSM3228584_200550980025_R08C01&quot; &quot;GSM3228592_200550980035_R01C01&quot;
##  [25] &quot;GSM3228593_200550980035_R02C01&quot; &quot;GSM3228594_200550980035_R03C01&quot;
##  [27] &quot;GSM3228595_200550980035_R04C01&quot; &quot;GSM3228596_200550980035_R05C01&quot;
##  [29] &quot;GSM3228597_200550980035_R06C01&quot; &quot;GSM3228598_200550980035_R07C01&quot;
##  [31] &quot;GSM3228599_200550980035_R08C01&quot; &quot;GSM3228618_200590490021_R05C01&quot;
##  [33] &quot;GSM3228623_200590490052_R02C01&quot; &quot;GSM3228633_200590490056_R04C01&quot;
##  [35] &quot;GSM3228634_200590490056_R05C01&quot; &quot;GSM3228638_200590490062_R02C01&quot;
##  [37] &quot;GSM3228640_200590490062_R04C01&quot; &quot;GSM3228645_200590490074_R01C01&quot;
##  [39] &quot;GSM3228675_200594720075_R07C01&quot; &quot;GSM3228678_200594720089_R02C01&quot;
##  [41] &quot;GSM3228681_200594720089_R05C01&quot; &quot;GSM3228685_200594740031_R01C01&quot;
##  [43] &quot;GSM3228692_200594740033_R01C01&quot; &quot;GSM3228700_200594740034_R01C01&quot;
##  [45] &quot;GSM3228701_200594740034_R02C01&quot; &quot;GSM3228702_200594740034_R03C01&quot;
##  [47] &quot;GSM3228703_200594740034_R04C01&quot; &quot;GSM3228704_200594740034_R05C01&quot;
##  [49] &quot;GSM3228705_200594740034_R06C01&quot; &quot;GSM3228706_200594740034_R07C01&quot;
##  [51] &quot;GSM3228707_200594740034_R08C01&quot; &quot;GSM3228711_200594740035_R04C01&quot;
##  [53] &quot;GSM3228716_200594740039_R01C01&quot; &quot;GSM3228717_200594740039_R02C01&quot;
##  [55] &quot;GSM3228718_200594740039_R03C01&quot; &quot;GSM3228719_200594740039_R04C01&quot;
##  [57] &quot;GSM3228720_200594740039_R05C01&quot; &quot;GSM3228721_200594740039_R06C01&quot;
##  [59] &quot;GSM3228722_200594740039_R07C01&quot; &quot;GSM3228723_200594740039_R08C01&quot;
##  [61] &quot;GSM3228736_200594740047_R05C01&quot; &quot;GSM3228748_200594740052_R01C01&quot;
##  [63] &quot;GSM3228766_200594740069_R03C01&quot; &quot;GSM3228774_200594740070_R03C01&quot;
##  [65] &quot;GSM3228787_200594740073_R01C01&quot; &quot;GSM3228795_200594740075_R02C01&quot;
##  [67] &quot;GSM3228796_200594740075_R03C01&quot; &quot;GSM3228825_200594740084_R01C01&quot;
##  [69] &quot;GSM3228835_200594740085_R03C01&quot; &quot;GSM3228841_200594740086_R01C01&quot;
##  [71] &quot;GSM3228848_200594740087_R01C01&quot; &quot;GSM3228849_200594740087_R02C01&quot;
##  [73] &quot;GSM3228850_200594740087_R03C01&quot; &quot;GSM3228851_200594740087_R04C01&quot;
##  [75] &quot;GSM3228852_200594740087_R05C01&quot; &quot;GSM3228853_200594740087_R06C01&quot;
##  [77] &quot;GSM3228854_200594740087_R07C01&quot; &quot;GSM3228855_200594740087_R08C01&quot;
##  [79] &quot;GSM3228856_200594740089_R01C01&quot; &quot;GSM3228880_200598340037_R01C01&quot;
##  [81] &quot;GSM3228888_200598340040_R02C01&quot; &quot;GSM3228889_200598340040_R03C01&quot;
##  [83] &quot;GSM3228890_200598340040_R04C01&quot; &quot;GSM3228895_200598350004_R01C01&quot;
##  [85] &quot;GSM3228896_200598350004_R02C01&quot; &quot;GSM3228897_200598350004_R03C01&quot;
##  [87] &quot;GSM3228898_200598350004_R04C01&quot; &quot;GSM3228899_200598350004_R05C01&quot;
##  [89] &quot;GSM3228900_200598350004_R06C01&quot; &quot;GSM3228901_200598350004_R07C01&quot;
##  [91] &quot;GSM3228902_200598350004_R08C01&quot; &quot;GSM3228908_200598350006_R02C01&quot;
##  [93] &quot;GSM3228909_200598350006_R03C01&quot; &quot;GSM3228910_200598350006_R04C01&quot;
##  [95] &quot;GSM3228911_200598350006_R05C01&quot; &quot;GSM3228923_200598350012_R01C01&quot;
##  [97] &quot;GSM3228924_200598350012_R02C01&quot; &quot;GSM3228925_200598350012_R03C01&quot;
##  [99] &quot;GSM3228929_200598350012_R07C01&quot; &quot;GSM3228955_200598350022_R04C01&quot;
## [101] &quot;GSM3228961_200598350025_R02C01&quot; &quot;GSM3228970_200598360006_R02C01&quot;
## [103] &quot;GSM3228978_200598350029_R03C01&quot; &quot;GSM3228988_200598350041_R01C01&quot;
## [105] &quot;GSM3228990_200598350041_R03C01&quot; &quot;GSM3228997_200598350042_R02C01&quot;
## [107] &quot;GSM3229004_200598350043_R01C01&quot; &quot;GSM3229005_200598350043_R02C01&quot;
## [109] &quot;GSM3229006_200598350043_R03C01&quot; &quot;GSM3229007_200598350043_R04C01&quot;
## [111] &quot;GSM3229008_200598350043_R05C01&quot; &quot;GSM3229009_200598350043_R06C01&quot;
## [113] &quot;GSM3229012_200598350044_R01C01&quot; &quot;GSM3229013_200598350044_R02C01&quot;
## [115] &quot;GSM3229014_200598350044_R03C01&quot; &quot;GSM3229015_200598350044_R04C01&quot;
## [117] &quot;GSM3229016_200598350044_R06C01&quot; &quot;GSM3229017_200598350044_R07C01&quot;
## [119] &quot;GSM3229019_200598350045_R01C01&quot; &quot;GSM3229020_200598350045_R02C01&quot;
## [121] &quot;GSM3229021_200598350045_R03C01&quot; &quot;GSM3229022_200598350045_R04C01&quot;
## [123] &quot;GSM3229023_200598350045_R05C01&quot; &quot;GSM3229024_200598350045_R06C01&quot;
## [125] &quot;GSM3229025_200598350045_R07C01&quot; &quot;GSM3229026_200598350045_R08C01&quot;
## [127] &quot;GSM3229036_200598350053_R02C01&quot; &quot;GSM3229043_200598350054_R01C01&quot;
## [129] &quot;GSM3229044_200598350054_R02C01&quot; &quot;GSM3229045_200598350054_R03C01&quot;
## [131] &quot;GSM3229046_200598350054_R04C01&quot; &quot;GSM3229047_200598350054_R05C01&quot;
## [133] &quot;GSM3229048_200598350054_R06C01&quot; &quot;GSM3229049_200598350054_R07C01&quot;
## [135] &quot;GSM3229050_200598350054_R08C01&quot; &quot;GSM3229061_200598350065_R03C01&quot;
## [137] &quot;GSM3229069_200598350066_R03C01&quot; &quot;GSM3229098_200598350076_R01C01&quot;
## [139] &quot;GSM3229110_200598350078_R01C01&quot; &quot;GSM3229111_200598350078_R02C01&quot;
## [141] &quot;GSM3229112_200598350078_R03C01&quot; &quot;GSM3229153_200598360004_R02C01&quot;
## [143] &quot;GSM3229154_200598360004_R03C01&quot; &quot;GSM3229155_200598360004_R04C01&quot;
## [145] &quot;GSM3229167_200598360006_R01C01&quot; &quot;GSM3229169_200598360006_R03C01&quot;
## [147] &quot;GSM3229173_200598360006_R07C01&quot; &quot;GSM3229192_200598360014_R04C01&quot;
## [149] &quot;GSM3229220_200598360027_R08C01&quot; &quot;GSM3229229_200598350076_R02C01&quot;
## [151] &quot;GSM3229238_200598360007_R07C01&quot;</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" data-line-number="1">targets &lt;-<span class="st"> </span>targets[<span class="op">!</span><span class="kw">rownames</span>(targets) <span class="op">%in%</span><span class="st"> </span>flagged_samples, ]</a>
<a class="sourceLine" id="cb33-2" data-line-number="2"></a>
<a class="sourceLine" id="cb33-3" data-line-number="3">RGset &lt;-<span class="st"> </span>RGset[,<span class="kw">colnames</span>(RGset) <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(targets)]</a>
<a class="sourceLine" id="cb33-4" data-line-number="4">betas &lt;-<span class="st"> </span>betas[,<span class="kw">colnames</span>(betas) <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(targets)]</a></code></pre></div>
<hr />
</div>
<div id="principal-components-analysis" class="section level1">
<h1>Principal components analysis</h1>
<div id="calculate-pcs" class="section level2">
<h2>Calculate PCs</h2>
<p>Principal components (PCs) of the beta values can be calculated using <code>prcomp</code>.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1">tbetas &lt;-<span class="st"> </span><span class="kw">t</span>(betas)</a>
<a class="sourceLine" id="cb34-2" data-line-number="2">pca &lt;-<span class="st"> </span><span class="kw">prcomp</span>(tbetas, <span class="dt">center =</span> <span class="ot">TRUE</span>, <span class="dt">rank. =</span> <span class="dv">10</span>)</a>
<a class="sourceLine" id="cb34-3" data-line-number="3"></a>
<a class="sourceLine" id="cb34-4" data-line-number="4"><span class="kw">summary</span>(pca)</a></code></pre></div>
<pre><code>## Importance of first k=10 (out of 526) components:
##                            PC1     PC2     PC3     PC4     PC5     PC6     PC7
## Standard deviation     15.0245 12.8122 9.86803 7.62670 4.27133 3.81015 3.57736
## Proportion of Variance  0.1579  0.1148 0.06811 0.04068 0.01276 0.01015 0.00895
## Cumulative Proportion   0.1579  0.2727 0.34081 0.38149 0.39425 0.40440 0.41336
##                            PC8     PC9    PC10
## Standard deviation     3.37456 3.23810 2.92641
## Proportion of Variance 0.00796 0.00733 0.00599
## Cumulative Proportion  0.42132 0.42865 0.43464</code></pre>
<hr />
</div>
<div id="screeplot" class="section level2">
<h2>Screeplot</h2>
<p>Variance explained by each PC can be calculated.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" data-line-number="1">var_explained =</a>
<a class="sourceLine" id="cb36-2" data-line-number="2"><span class="st">  </span><span class="kw">data.frame</span>(<span class="dt">PC =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(betas),</a>
<a class="sourceLine" id="cb36-3" data-line-number="3">             <span class="dt">var_explained =</span> pca<span class="op">$</span>sdev<span class="op">^</span><span class="dv">2</span> <span class="op">/</span><span class="st"> </span><span class="kw">sum</span>(pca<span class="op">$</span>sdev<span class="op">^</span><span class="dv">2</span>))[<span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(pca<span class="op">$</span>x),]</a>
<a class="sourceLine" id="cb36-4" data-line-number="4"></a>
<a class="sourceLine" id="cb36-5" data-line-number="5">var_explained</a></code></pre></div>
<pre><code>##    PC var_explained
## 1   1   0.157885762
## 2   2   0.114812612
## 3   3   0.068108734
## 4   4   0.040683261
## 5   5   0.012760511
## 6   6   0.010153764
## 7   7   0.008950903
## 8   8   0.007964833
## 9   9   0.007333710
## 10 10   0.005989818</code></pre>
<p>This can be plotted in a screeplot to visualize the structure of the DNAm data.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb38-1" data-line-number="1">var_explained <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb38-2" data-line-number="2"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>PC, <span class="dt">y=</span>var_explained)) <span class="op">+</span></a>
<a class="sourceLine" id="cb38-3" data-line-number="3"><span class="st">  </span><span class="kw">geom_line</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb38-4" data-line-number="4"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">color=</span><span class="st">&#39;grey5&#39;</span>, <span class="dt">fill=</span><span class="st">&#39;#6DACBC&#39;</span>, <span class="dt">shape=</span><span class="dv">21</span>, <span class="dt">size=</span><span class="dv">3</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb38-5" data-line-number="5"><span class="st">  </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks=</span><span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(pca<span class="op">$</span>x)) <span class="op">+</span></a>
<a class="sourceLine" id="cb38-6" data-line-number="6"><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Principal Component&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb38-7" data-line-number="7"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Proportion of variance explained&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb38-8" data-line-number="8"><span class="st">  </span><span class="kw">theme_bw</span>()</a></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/224screeplot-1.png" width="672" /></p>
<hr />
</div>
<div id="heatmap" class="section level2">
<h2>Heatmap</h2>
<p>Any constant variables are removed from the heatmap, as they will not explain variance in the data.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb39-1" data-line-number="1">plot_vars &lt;-<span class="st"> </span><span class="kw">apply</span>(targets, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="kw">sd</span>(<span class="kw">as.numeric</span>(<span class="kw">factor</span>(x)), <span class="dt">na.rm=</span>T))</a>
<a class="sourceLine" id="cb39-2" data-line-number="2"></a>
<a class="sourceLine" id="cb39-3" data-line-number="3">plot_vars &lt;-<span class="st"> </span><span class="kw">names</span>(plot_vars[<span class="op">!</span>plot_vars <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="ot">NA</span>, <span class="dv">0</span>)])</a>
<a class="sourceLine" id="cb39-4" data-line-number="4">plot_vars</a></code></pre></div>
<pre><code>##  [1] &quot;sample_ID&quot;          &quot;geo_accession&quot;      &quot;sex&quot;               
##  [4] &quot;age&quot;                &quot;log_total_pbb&quot;      &quot;pbb_153&quot;           
##  [7] &quot;pbb_77&quot;             &quot;pbb_101&quot;            &quot;pbb_180&quot;           
## [10] &quot;supplementary_file&quot; &quot;plate&quot;              &quot;row&quot;               
## [13] &quot;Basename&quot;</code></pre>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb41-1" data-line-number="1">heatmap_df &lt;-<span class="st"> </span>targets <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb41-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="kw">any_of</span>(plot_vars))</a></code></pre></div>
<p>All variables are then converted to numeric and correlations between them and the PCs are calculated.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1">heatmap_df &lt;-<span class="st"> </span><span class="kw">apply</span>(heatmap_df, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="kw">as.numeric</span>(<span class="kw">factor</span>(x)))</a>
<a class="sourceLine" id="cb42-2" data-line-number="2"></a>
<a class="sourceLine" id="cb42-3" data-line-number="3">cxy &lt;-<span class="st"> </span><span class="kw">round</span>(<span class="kw">cor</span>(pca<span class="op">$</span>x, <span class="kw">scale</span>(heatmap_df), <span class="dt">use=</span><span class="st">&quot;pairwise.complete.obs&quot;</span>),<span class="dv">2</span>) </a></code></pre></div>
<p>A heatmap can then be used to visualize these correlations and uncover measured variables that explain a large proportion of DNAm variance.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb43-1" data-line-number="1">col_fun &lt;-<span class="st"> </span><span class="kw">colorRamp2</span>(<span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">1</span>), <span class="kw">c</span>(<span class="st">&quot;#000042&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;#800000&quot;</span>))</a>
<a class="sourceLine" id="cb43-2" data-line-number="2"></a>
<a class="sourceLine" id="cb43-3" data-line-number="3"><span class="kw">Heatmap</span>(</a>
<a class="sourceLine" id="cb43-4" data-line-number="4">  <span class="kw">t</span>(cxy),                              </a>
<a class="sourceLine" id="cb43-5" data-line-number="5">  <span class="dt">col =</span> col_fun,  </a>
<a class="sourceLine" id="cb43-6" data-line-number="6">  <span class="dt">border =</span> <span class="st">&#39;grey5&#39;</span>,</a>
<a class="sourceLine" id="cb43-7" data-line-number="7">  <span class="dt">cluster_columns =</span> <span class="ot">FALSE</span>,            </a>
<a class="sourceLine" id="cb43-8" data-line-number="8">  <span class="dt">show_row_dend =</span> <span class="ot">TRUE</span>,             </a>
<a class="sourceLine" id="cb43-9" data-line-number="9">  <span class="dt">show_column_dend =</span> <span class="ot">FALSE</span>,      </a>
<a class="sourceLine" id="cb43-10" data-line-number="10">  <span class="dt">name =</span> <span class="st">&quot;Value&quot;</span>,                 </a>
<a class="sourceLine" id="cb43-11" data-line-number="11">  <span class="dt">row_names_gp =</span> <span class="kw">gpar</span>(<span class="dt">fontsize =</span> <span class="dv">8</span>), </a>
<a class="sourceLine" id="cb43-12" data-line-number="12">  <span class="dt">column_names_gp =</span> <span class="kw">gpar</span>(<span class="dt">fontsize =</span> <span class="dv">8</span>), </a>
<a class="sourceLine" id="cb43-13" data-line-number="13">  <span class="dt">cell_fun =</span> <span class="cf">function</span>(j, i, x, y, width, height, fill) {</a>
<a class="sourceLine" id="cb43-14" data-line-number="14">    <span class="kw">grid.rect</span>(x, y, width, height, </a>
<a class="sourceLine" id="cb43-15" data-line-number="15">              <span class="dt">gp =</span> <span class="kw">gpar</span>(<span class="dt">col =</span> <span class="st">&quot;white&quot;</span>, <span class="dt">lwd =</span> <span class="dv">1</span>, <span class="dt">fill =</span> <span class="ot">NA</span>))</a>
<a class="sourceLine" id="cb43-16" data-line-number="16">    <span class="kw">grid.text</span>(<span class="kw">ifelse</span>(<span class="kw">abs</span>(<span class="kw">t</span>(cxy)[i,j]) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.4</span>,</a>
<a class="sourceLine" id="cb43-17" data-line-number="17">                     <span class="kw">sprintf</span>(<span class="st">&quot;%.2f&quot;</span>, <span class="kw">round</span>(<span class="kw">t</span>(cxy)[i, j], <span class="dv">2</span>)),</a>
<a class="sourceLine" id="cb43-18" data-line-number="18">                     <span class="st">&quot;&quot;</span>), </a>
<a class="sourceLine" id="cb43-19" data-line-number="19">              x, y, <span class="dt">gp =</span> <span class="kw">gpar</span>(<span class="dt">fontsize =</span> <span class="dv">10</span>, <span class="dt">col =</span> <span class="st">&quot;white&quot;</span>))</a>
<a class="sourceLine" id="cb43-20" data-line-number="20">  }</a>
<a class="sourceLine" id="cb43-21" data-line-number="21">)</a></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/227heatmap-1.png" width="672" /></p>
<hr />
</div>
<div id="plot" class="section level2">
<h2>Plot</h2>
<p>In order to further investigate flagged variables, we can add the PCs to the <code>targets</code> data frame.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" data-line-number="1">pc_df &lt;-<span class="st"> </span><span class="kw">cbind</span>(targets, pca<span class="op">$</span>x)</a></code></pre></div>
<p>PCs can then be plotted against each other, coloured by flagged variables.</p>
<p>The second PC is strongly linked to sex differences.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb45-1" data-line-number="1">pc_df <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb45-2" data-line-number="2"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> PC1, <span class="dt">y =</span> PC2, <span class="dt">color =</span> sex)) <span class="op">+</span></a>
<a class="sourceLine" id="cb45-3" data-line-number="3"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="fl">1.5</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb45-4" data-line-number="4"><span class="st">    </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="kw">paste0</span>(<span class="st">&quot;PC1 (&quot;</span>, <span class="kw">round</span>(<span class="dv">100</span><span class="op">*</span>var_explained[<span class="dv">1</span>,<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">&quot;%)&quot;</span>), </a>
<a class="sourceLine" id="cb45-5" data-line-number="5">         <span class="dt">y =</span> <span class="kw">paste0</span>(<span class="st">&quot;PC2 (&quot;</span>, <span class="kw">round</span>(<span class="dv">100</span><span class="op">*</span>var_explained[<span class="dv">2</span>,<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">&quot;%)&quot;</span>), </a>
<a class="sourceLine" id="cb45-6" data-line-number="6">         <span class="dt">color =</span> <span class="st">&quot;Sex&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb45-7" data-line-number="7"><span class="st">    </span><span class="kw">theme_bw</span>()</a></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/229pcplot2-1.png" width="672" /></p>
<hr />
</div>
</div>
<div id="checking-sample-relationships" class="section level1">
<h1>Checking Sample Relationships</h1>
<div id="annotations" class="section level2">
<h2>Annotations</h2>
<p><a href="http://bioconductor.org/packages/release/bioc/html/omicsPrint.html"><strong>omicsPrint</strong></a><sup>12</sup> is a package we developed to detect data linkage errors through inspecting sample relations in multiple omics studies.</p>
<p>Manifests can be downloaded from the <a href="https://zwdzwd.github.io/InfiniumAnnotation">Zhou lab GitHub page</a><sup>15</sup> for all commonly used arrays and reference genomes. For this example, we use SNP annotations downloaded for the EPIC array and the hg19 genome.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb46-1" data-line-number="1">snp_cpgs &lt;-<span class="st"> </span><span class="kw">read_tsv</span>(</a>
<a class="sourceLine" id="cb46-2" data-line-number="2">  <span class="st">&quot;./data/EPIC.hg19.manifest.pop.tsv.gz&quot;</span>)</a></code></pre></div>
<pre><code>## Rows: 865918 Columns: 66
## ── Column specification ────────────────────────────────────────────────────────
## Delimiter: &quot;\t&quot;
## chr  (2): CpG_chrm, probeID
## dbl  (2): CpG_beg, CpG_end
## lgl (62): MASK_general_AFR, MASK_snp5_AFR, MASK_general_EAS, MASK_snp5_EAS, ...
## 
## ℹ Use `spec()` to retrieve the full column specification for this data.
## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<p>Columns of interest include:</p>
<ul>
<li>cpg - ID of the probe for the CpG</li>
<li>chr - Chromosome where the CpG is located</li>
<li>start - Start position of the CpG</li>
<li>end - End position of the CpG</li>
</ul>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb48-1" data-line-number="1">snp_cpgs &lt;-<span class="st"> </span>snp_cpgs <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb48-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(</a>
<a class="sourceLine" id="cb48-3" data-line-number="3">    <span class="dt">cpg =</span> probeID,</a>
<a class="sourceLine" id="cb48-4" data-line-number="4">    <span class="dt">chr =</span> CpG_chrm,</a>
<a class="sourceLine" id="cb48-5" data-line-number="5">    <span class="dt">start =</span> CpG_beg,</a>
<a class="sourceLine" id="cb48-6" data-line-number="6">    <span class="dt">end =</span> CpG_end,</a>
<a class="sourceLine" id="cb48-7" data-line-number="7">    MASK_snp5_EUR</a>
<a class="sourceLine" id="cb48-8" data-line-number="8">  ) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb48-9" data-line-number="9"><span class="st">  </span><span class="kw">mutate</span>(</a>
<a class="sourceLine" id="cb48-10" data-line-number="10">    <span class="dt">chr =</span> <span class="kw">substr</span>(chr,<span class="dv">4</span>,<span class="dv">5</span>)</a>
<a class="sourceLine" id="cb48-11" data-line-number="11">  )</a>
<a class="sourceLine" id="cb48-12" data-line-number="12"></a>
<a class="sourceLine" id="cb48-13" data-line-number="13">snp_cpgs &lt;-<span class="st"> </span>(snp_cpgs <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb48-14" data-line-number="14"><span class="st">               </span>dplyr<span class="op">::</span><span class="kw">filter</span>(MASK_snp5_EUR <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span>))<span class="op">$</span>cpg</a>
<a class="sourceLine" id="cb48-15" data-line-number="15"></a>
<a class="sourceLine" id="cb48-16" data-line-number="16"><span class="kw">print</span>(<span class="kw">paste0</span>(<span class="st">&quot;There are &quot;</span>, <span class="kw">length</span>(snp_cpgs),</a>
<a class="sourceLine" id="cb48-17" data-line-number="17">             <span class="st">&quot; CpGs containing common European SNPs&quot;</span>))</a></code></pre></div>
<pre><code>## [1] &quot;There are 28152 CpGs containing common European SNPs&quot;</code></pre>
<p>Betas can then be subset to include only these CpG probes of interest.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb50-1" data-line-number="1">betas_snps &lt;-<span class="st"> </span>betas[snp_cpgs, ]</a>
<a class="sourceLine" id="cb50-2" data-line-number="2"><span class="kw">dim</span>(betas_snps)</a></code></pre></div>
<pre><code>## [1] 28152   526</code></pre>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb52-1" data-line-number="1"><span class="kw">table</span>(<span class="kw">rownames</span>(targets) <span class="op">==</span><span class="st"> </span><span class="kw">colnames</span>(betas_snps))</a></code></pre></div>
<pre><code>## 
## TRUE 
##  526</code></pre>
<hr />
<div id="assumed-relations" class="section level3">
<h3>Assumed relations</h3>
<p>Create a data frame of all pairwise sample comparisons.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb54-1" data-line-number="1">relations &lt;-<span class="st"> </span><span class="kw">expand.grid</span>(</a>
<a class="sourceLine" id="cb54-2" data-line-number="2">  <span class="dt">idx =</span> <span class="kw">colnames</span>(betas_snps), </a>
<a class="sourceLine" id="cb54-3" data-line-number="3">  <span class="dt">idy =</span> <span class="kw">colnames</span>(betas_snps))</a>
<a class="sourceLine" id="cb54-4" data-line-number="4"><span class="kw">head</span>(relations)</a></code></pre></div>
<pre><code>##                              idx                            idy
## 1 GSM3228585_200550980034_R01C01 GSM3228585_200550980034_R01C01
## 2 GSM3228586_200550980034_R02C01 GSM3228585_200550980034_R01C01
## 3 GSM3228587_200550980034_R03C01 GSM3228585_200550980034_R01C01
## 4 GSM3228588_200550980034_R04C01 GSM3228585_200550980034_R01C01
## 5 GSM3228589_200550980034_R05C01 GSM3228585_200550980034_R01C01
## 6 GSM3228590_200550980034_R06C01 GSM3228585_200550980034_R01C01</code></pre>
<p>Sample names are saved, as a reference for which pairs exist.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb56-1" data-line-number="1">relations<span class="op">$</span>sample_name_x &lt;-<span class="st"> </span>targets[relations<span class="op">$</span>idx, <span class="st">&quot;sample_ID&quot;</span>] </a>
<a class="sourceLine" id="cb56-2" data-line-number="2">relations<span class="op">$</span>sample_name_y &lt;-<span class="st"> </span>targets[relations<span class="op">$</span>idy, <span class="st">&quot;sample_ID&quot;</span>]</a></code></pre></div>
<p>A relation_type variable is then set as <code>identical</code> if the samples come from the same individual and <code>unrelated</code> otherwise. We see that the 59 pairs comparing the same sample are set as <code>identical</code>.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb57-1" data-line-number="1">relations<span class="op">$</span>relation_type &lt;-<span class="st"> &quot;unrelated&quot;</span></a>
<a class="sourceLine" id="cb57-2" data-line-number="2">relations<span class="op">$</span>relation_type[relations<span class="op">$</span>sample_name_x <span class="op">==</span><span class="st"> </span>relations<span class="op">$</span>sample_name_y] &lt;-<span class="st"> &quot;identical&quot;</span></a>
<a class="sourceLine" id="cb57-3" data-line-number="3"></a>
<a class="sourceLine" id="cb57-4" data-line-number="4"><span class="kw">table</span>(relations<span class="op">$</span>relation_type)</a></code></pre></div>
<pre><code>## 
## identical unrelated 
##       526    276150</code></pre>
<hr />
</div>
<div id="genotyping" class="section level3">
<h3>Genotyping</h3>
<p>The function <code>beta2genotype</code> from <code>omicsPrint</code> genotypes the observations by measuring homozygous or heterozygous alleles at these SNP probes. The arguments specified are:</p>
<ul>
<li><code>betas</code> - a beta matrix of probes possibly affected by SNPs</li>
<li><code>na.rm</code> - whether to drop CpGs for which no clustering is observed</li>
<li><code>minSep</code> - the minimal cluster separation</li>
<li><code>minSize</code> - the size of the smallest cluster as a percentage</li>
<li><code>centers</code> - centers of clusters</li>
<li><code>assayName</code> - the name of the assay to be used</li>
</ul>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb59-1" data-line-number="1">genotype &lt;-<span class="st"> </span><span class="kw">beta2genotype</span>(betas_snps, </a>
<a class="sourceLine" id="cb59-2" data-line-number="2">                          <span class="dt">na.rm=</span><span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb59-3" data-line-number="3">                          <span class="dt">minSep =</span> <span class="fl">0.25</span>,</a>
<a class="sourceLine" id="cb59-4" data-line-number="4">                          <span class="dt">minSize =</span> <span class="dv">5</span>,</a>
<a class="sourceLine" id="cb59-5" data-line-number="5">                          <span class="dt">centers =</span> <span class="kw">c</span>(<span class="fl">0.2</span>, <span class="fl">0.5</span>, <span class="fl">0.8</span>),</a>
<a class="sourceLine" id="cb59-6" data-line-number="6">                          <span class="dt">assayName=</span><span class="ot">NULL</span>)</a>
<a class="sourceLine" id="cb59-7" data-line-number="7"><span class="kw">str</span>(genotype)</a></code></pre></div>
<pre><code>##  int [1:1136, 1:526] 1 2 1 2 3 2 3 1 3 2 ...
##  - attr(*, &quot;dimnames&quot;)=List of 2
##   ..$ : chr [1:1136] &quot;cg12213037&quot; &quot;cg19405842&quot; &quot;cg24925741&quot; &quot;cg01296877&quot; ...
##   ..$ : chr [1:526] &quot;GSM3228585_200550980034_R01C01&quot; &quot;GSM3228586_200550980034_R02C01&quot; &quot;GSM3228587_200550980034_R03C01&quot; &quot;GSM3228588_200550980034_R04C01&quot; ...</code></pre>
<p><code>alleleSharing</code> can then assess the relationships between different individuals, which can be unrelated, twins, or identical. This uses identity by state (IBS), a genetic similarity measure that compares at a single locus the genotypes between two individuals and counts the number of alleles shared (0, 1 or 2).Essentially, if a sample is the same then the mean IBS should be 2 (alleles) and the variance of this should be low (close to 0).</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb61-1" data-line-number="1">out &lt;-<span class="st"> </span><span class="kw">alleleSharing</span>(genotype, <span class="dt">relations =</span> relations)</a></code></pre></div>
<pre><code>## Hash relations</code></pre>
<pre><code>## Pruning 1136 SNPs ...</code></pre>
<pre><code>## 0 SNPs removed because of low call rate!</code></pre>
<pre><code>## 0 samples removed because too few SNPs called!</code></pre>
<pre><code>## Using 1136 polymorphic SNPs to determine allele sharing.</code></pre>
<pre><code>## Running `square` IBS algorithm!</code></pre>
<pre><code>## 527 of 138601 (0.38%) ...</code></pre>
<pre><code>## 47651 of 138601 (34.38%) ...</code></pre>
<pre><code>## 85301 of 138601 (61.54%) ...</code></pre>
<pre><code>## 112951 of 138601 (81.49%) ...</code></pre>
<pre><code>## 130601 of 138601 (94.23%) ...</code></pre>
<pre><code>## 138251 of 138601 (99.75%) ...</code></pre>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb74-1" data-line-number="1"><span class="kw">head</span>(out)</a></code></pre></div>
<pre><code>##       mean       var                     colnames.x
## 1 2.000000 0.0000000 GSM3228585_200550980034_R01C01
## 2 1.318662 0.4446206 GSM3228586_200550980034_R02C01
## 3 1.289613 0.4138480 GSM3228587_200550980034_R03C01
## 4 1.320423 0.4170627 GSM3228588_200550980034_R04C01
## 5 1.350352 0.4093038 GSM3228589_200550980034_R05C01
## 6 1.322183 0.4194546 GSM3228590_200550980034_R06C01
##                       colnames.y  relation
## 1 GSM3228585_200550980034_R01C01 identical
## 2 GSM3228585_200550980034_R01C01 unrelated
## 3 GSM3228585_200550980034_R01C01 unrelated
## 4 GSM3228585_200550980034_R01C01 unrelated
## 5 GSM3228585_200550980034_R01C01 unrelated
## 6 GSM3228585_200550980034_R01C01 unrelated</code></pre>
<hr />
</div>
<div id="identify-mismatches" class="section level3">
<h3>Identify mismatches</h3>
<p>The mean and variance of IBS can then be visualised using the <code>inferRelations()</code> function. Genetically similar sample comparisons will be plotted closer to the bottom, right of the graph, and those that are genetically dissimilar are in the top, left. A table of mismatches gives an overview of any identified.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb76-1" data-line-number="1">mismatches &lt;-<span class="st"> </span><span class="kw">inferRelations</span>(out, <span class="dt">verbose=</span><span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>##                   Assumed relation
## Predicted relation identical unrelated
##          identical       526        15
##          unrelated         .    138060</code></pre>
<p><img src="02_SampleQualityControl_files/figure-html/238plot-1.png" width="672" /></p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb78-1" data-line-number="1"><span class="kw">table</span>(mismatches<span class="op">$</span>relation, mismatches<span class="op">$</span>predicted)</a></code></pre></div>
<pre><code>##            
##             identical unrelated
##   identical         0         0
##   unrelated        15         0</code></pre>
<p>There are 18 pairs of samples (n=36) that should be unrelated but are genetically identical on the basis of these beta-derived genotypes. To remove them from further analysis you can use the <code>mismatches</code> object:</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb80-1" data-line-number="1">removed_ids &lt;-<span class="st"> </span><span class="kw">unique</span>(<span class="kw">c</span>(mismatches<span class="op">$</span>colnames.x, mismatches<span class="op">$</span>colnames.y))</a>
<a class="sourceLine" id="cb80-2" data-line-number="2">removed_ids</a></code></pre></div>
<pre><code>##  [1] &quot;GSM3229240_200594740075_R08C01&quot; &quot;GSM3229222_200598350005_R03C01&quot;
##  [3] &quot;GSM3229223_200598350005_R06C01&quot; &quot;GSM3229224_200598350005_R08C01&quot;
##  [5] &quot;GSM3229104_200598350077_R03C01&quot; &quot;GSM3229119_200598350079_R02C01&quot;
##  [7] &quot;GSM3229225_200598350031_R02C01&quot; &quot;GSM3229226_200598350031_R04C01&quot;
##  [9] &quot;GSM3229227_200598350031_R05C01&quot; &quot;GSM3229228_200598350031_R07C01&quot;
## [11] &quot;GSM3229230_200598350076_R03C01&quot; &quot;GSM3229231_200598350076_R06C01&quot;
## [13] &quot;GSM3229232_200598350080_R02C01&quot; &quot;GSM3229233_200598350080_R05C01&quot;
## [15] &quot;GSM3229234_200598350080_R07C01&quot; &quot;GSM3228624_200590490052_R03C01&quot;
## [17] &quot;GSM3228904_200598350005_R04C01&quot; &quot;GSM3228905_200598350005_R05C01&quot;
## [19] &quot;GSM3228906_200598350005_R07C01&quot; &quot;GSM3228922_200598350011_R08C01&quot;
## [21] &quot;GSM3228948_200598350018_R05C01&quot; &quot;GSM3228984_200598350031_R01C01&quot;
## [23] &quot;GSM3228985_200598350031_R03C01&quot; &quot;GSM3228986_200598350031_R06C01&quot;
## [25] &quot;GSM3228987_200598350031_R08C01&quot; &quot;GSM3229099_200598350076_R04C01&quot;
## [27] &quot;GSM3229100_200598350076_R05C01&quot; &quot;GSM3229126_200598350080_R01C01&quot;
## [29] &quot;GSM3229128_200598350080_R06C01&quot; &quot;GSM3229129_200598350080_R08C01&quot;</code></pre>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb82-1" data-line-number="1">targets &lt;-<span class="st"> </span>targets[<span class="op">!</span><span class="kw">rownames</span>(targets) <span class="op">%in%</span><span class="st"> </span>removed_ids, ]</a>
<a class="sourceLine" id="cb82-2" data-line-number="2"></a>
<a class="sourceLine" id="cb82-3" data-line-number="3">RGset &lt;-<span class="st"> </span>RGset[,<span class="kw">colnames</span>(RGset) <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(targets)]</a>
<a class="sourceLine" id="cb82-4" data-line-number="4">betas &lt;-<span class="st"> </span>betas[,<span class="kw">colnames</span>(betas) <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(targets)]</a>
<a class="sourceLine" id="cb82-5" data-line-number="5"></a>
<a class="sourceLine" id="cb82-6" data-line-number="6">RGset</a></code></pre></div>
<pre><code>## class: RGChannelSetExtended 
## dim: 1052641 496 
## metadata(0):
## assays(5): Green Red GreenSD RedSD NBeads
## rownames(1052641): 1600101 1600111 ... 99810990 99810992
## rowData names(0):
## colnames(496): GSM3228585_200550980034_R01C01
##   GSM3228586_200550980034_R02C01 ... GSM3229237_200598350015_R06C01
##   GSM3229239_200598350080_R03C01
## colData names(15): sample_ID geo_accession ... filenames beta_outlier
## Annotation
##   array: IlluminaHumanMethylationEPIC
##   annotation: 20a1.hg38</code></pre>
<p>In data with sample relationships, this would be shown in the above graph as green or black clusters<sup>12</sup>. It is important to carry out this type of visualization before probe-filtering as otherwise the genotyping will be based on very few SNPs.</p>
<hr />
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
