<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title></title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 52px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 57px;
  margin-top: -57px;
}

.section h2 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h3 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h4 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h5 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h6 {
  padding-top: 57px;
  margin-top: -57px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DNAmArray</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="02_SampleQualityControl.html">Quality control</a>
</li>
<li>
  <a href="04_Visualization.html">Visualizations</a>
</li>
<li>
  <a href="05_Normalization.html">Normalization</a>
</li>
<li>
  <a href="06_ProbeFiltering.html">Probe filtering</a>
</li>
<li>
  <a href="07_Miscellaneous.html">Miscellaneous</a>
</li>
<li>
  <a href="08_EWAS.html">Ewas</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">About</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">




</div>


<div id="running-methylaid" class="section level1">
<h1>Running MethylAid</h1>
<p>Sample quality control is performed using the <a href="http://bioconductor.org/MethylAid"><strong>MethylAid</strong></a> <span class="citation">(Iterson et al. 2014)</span>. <a href="http://bioconductor.org/MethylAid"><strong>MethylAid</strong></a> is specially designed for quality control of large sets of DNA methylation data e.g., epigenomewide association studies (EWAS). Extracting intensities from IDAT files can be done in batches and/or in parallel to reduce memory load and/or overcome long run-times. It requires two function calls in going from IDAT files to launch the interactive web application; <code>summarize</code> and <code>visualize</code>. For more information see <span class="citation">(Iterson et al. 2014)</span>.</p>
<p>First, prepare the targets file, similar to the one used by <a href="http://bioconductor.org/minfi"><strong>minfi</strong></a>.</p>
<pre class="r"><code>path450k &lt;- &quot;/exports/molepi/BIOS/LLS/450k/&quot;
targets &lt;- read.table(file.path(path450k, &quot;sample_sheet.txt&quot;), sep=&quot;\t&quot;, header=TRUE, as.is=TRUE)
head(targets)
targets$Basename &lt;- with(targets,
                              file.path(path450k, Sentrix_Barcode,
                                        paste(Sentrix_Barcode, Sentrix_Position, sep=&quot;_&quot;)))</code></pre>
<pre class="r"><code>targets[1:5, 1:5]</code></pre>
<p>Next, summarize and visualize the data, see the <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/MethylAid/inst/doc/MethylAid.pdf">MethylAid vignette</a> for more details.</p>
<pre class="r"><code>library(MethylAid)
library(BiocParallel)
library(BatchJobs)
conffile &lt;- system.file(&quot;scripts/config.R&quot;, package=&quot;MethylAid&quot;)
BPPARAM &lt;- BatchJobsParam(workers = 10, progressbar = FALSE, conffile = conffile)
sData &lt;- summarize(targets, batchSize = 50, BPPARAM = BPPARAM)</code></pre>
<pre class="r"><code>load(file.path(path450k, &quot;MethylAid.RData&quot;))
outliers &lt;- visualize(MethylAid)
write.table(outliers,
            file=file.path(path450k, &quot;outliers.txt&quot;),
            quote=FALSE, row.names=FALSE)</code></pre>
<p>Check the interactive visualization of the <a href="http://bios-vm.bbmrirp3-lumc.surf-hosted.nl:8008/BIOSRutils/DNAm/LLS">LLS 450k data</a> generated by <a href="http://bioconductor.org/MethylAid"><strong>MethylAid</strong></a>.</p>
<p>It turns out that 5 samples do not pass quality control.</p>
</div>
<div id="constructing-an-rgset-object-in-parallel" class="section level1">
<h1>Constructing an <code>RGset</code>-object in parallel</h1>
<p>Reading a large number of IDAT-files can be time-consuming. The <a href="https://github.com/molepi/DNAmArray"><strong>DNAmArray</strong></a> package offers the <code>read.metharray.exp.par</code>-function for reading IDAT-files in parallel using functionallity from <a href="http://bioconductor.org/BiocParallel"><strong>BiocParallel</strong></a>. The target-file is divided in a number of smaller targets files according to the number of workers (compute nodes) available. On each worknode a subset of IDAT-files are read using the regular <a href="http://bioconductor.org/minfi"><strong>minfi</strong></a> function-<code>read.metharray.expr</code>. Once all idat-files are read they are combined into on <code>RGChannelSet</code>-object and returned.</p>
<p>First, we remove those samples that did not pass quality control.</p>
<pre class="r"><code>outliers &lt;- read.table(file.path(path450k, &quot;outliers.txt&quot;), header=TRUE)    
targets &lt;- targets[!(targets$methylation_run_id %in% outliers$run_id), ]</code></pre>
<p>For example, running on a cluster with the Sun Grid Job scheduler.</p>
<pre class="r"><code>library(DNAmArray)
library(BiocParallel)
library(BatchJobs)
conffile &lt;- system.file(&quot;scripts/config.R&quot;, package=&quot;MethylAid&quot;)
BPPARAM &lt;- BatchJobsParam(workers = 10, progressbar = FALSE, conffile = conffile)
RGset &lt;- read.metharray.exp.par(targets, verbose=TRUE, extended=TRUE)</code></pre>
<pre class="r"><code>RGset</code></pre>
<p>Reading data in parallel can go wrong and often debugging is difficult. Recently, <a href="http://bioconductor.org/BiocParallel"><strong>BiocParallel</strong></a> has been extended with a comprehensive set of functions for debugging on various parallel architectures. We recommend running e.g. <code>BatchJobsParam</code> with option <code>log=TRUE</code>.</p>
<div class="references">
<h1>References</h1>
<p>Iterson, M. van, E. W. Tobi, R. C. Slieker, W. den Hollander, R. Luijk, P. E. Slagboom, and B. T. Heijmans. 2014. “MethylAid: visual and Interactive Quality Control of Large Illumina 450k Datasets.” <em>Bioinformatics</em> 30 (23): 3435–3437.</p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
