<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>02_SampleQualityControl.knit</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DNAmArray</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="01_GettingStarted.html">Getting Started</a>
</li>
<li>
  <a href="02_SampleQualityControl.html">Sample QC</a>
</li>
<li>
  <a href="03_Normalization.html">Normalize</a>
</li>
<li>
  <a href="04_Predict.html">Predict</a>
</li>
<li>
  <a href="05_ProbeQC.html">Probe QC</a>
</li>
<li>
  <a href="06_EWAS.html">Analyse</a>
</li>
<li>
  <a href="07_Interpret.html">Interpret</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">About</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">




</div>


<hr />
<div id="loading-dnam-data" class="section level1">
<h1>Loading DNAm data</h1>
<div id="summarizing-idats" class="section level2">
<h2>Summarizing IDATs</h2>
<p>The <a
href="http://bioconductor.org/packages/MethylAid"><strong>MethylAid</strong></a><sup>10</sup>
package, which will be used for quality control in this workflow,
requires the targets data frame to contain a <code>Basename</code>
variable that points to IDAT file names. Sometimes, data comes with a
sample sheet to facilitate this, but in this example the information is
extracted from the supplementary file column instead.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>targets<span class="sc">$</span>Basename <span class="ot">&lt;-</span> <span class="fu">substring</span>(targets<span class="sc">$</span>supplementary_file,<span class="dv">68</span>,<span class="dv">106</span>)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">head</span>(targets<span class="sc">$</span>Basename)</span></code></pre></div>
<pre><code>## [1] &quot;GSM3228562_200550980002_R01C01_Grn.idat&quot; &quot;GSM3228563_200550980002_R02C01_Grn.idat&quot; &quot;GSM3228564_200550980002_R03C01_Grn.idat&quot;
## [4] &quot;GSM3228565_200550980002_R04C01_Grn.idat&quot; &quot;GSM3228566_200550980002_R05C01_Grn.idat&quot; &quot;GSM3228567_200550980002_R06C01_Grn.idat&quot;</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">rownames</span>(targets) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;_Grn.idat&quot;</span>, <span class="st">&quot;&quot;</span>, targets<span class="sc">$</span>Basename)</span></code></pre></div>
<p>The following sample quality control steps will also require the <a
href="https://bioconductor.org/packages/release/bioc/html/BiocParallel.html"><strong>BiocParallel</strong></a>
package. Using parallel processing and/or batches will reduce both
memory load and run-times when extracting intensities from IDAT files.
Please see the <a
href="http://bioconductor.org/packages/release/bioc/vignettes/MethylAid/inst/doc/MethylAid.pdf">MethylAid
vignette</a> for more details.</p>
<p>The available number of cores is set usnig
<code>MulticoreParam</code>, and IDATs are summarized using the
<code>summarize</code> function.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>BPPARAM <span class="ot">&lt;-</span> <span class="fu">MulticoreParam</span>(<span class="dv">6</span>)</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>sData <span class="ot">&lt;-</span> MethylAid<span class="sc">::</span><span class="fu">summarize</span>(targets, <span class="at">batchSize=</span><span class="dv">50</span>, <span class="at">base=</span><span class="st">&quot;../GSE116339/IDATs&quot;</span>)</span></code></pre></div>
<hr />
</div>
</div>
<div id="creating-an-rgset" class="section level1">
<h1>Creating an <code>RGset</code></h1>
<p>In addition to the <code>sData</code> object, which is used by
MethylAid, it is useful to also create an
<code>RGChannelSetExtended</code> object, which contains more extensive
information on the data. Here, we use <code>read.metharray.exp()</code>
from <a
href="http://bioconductor.org/packages/minfi"><strong>minfi</strong></a><sup>9</sup>
but, since this has not been updated for the more recent arrays, we
advise monkey patching the internal <code>.guessArrayTypes</code>
function, with the appropriate array and annotation.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>.guessArrayTypes <span class="ot">&lt;-</span> <span class="cf">function</span>(nProbes) {</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>  <span class="cf">if</span> (nProbes <span class="sc">&gt;</span> <span class="dv">850000</span>) {</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>    <span class="co"># EPIC v2 array </span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>    arrayAnnotation <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">array =</span> <span class="st">&quot;IlluminaHumanMethylationEPIC&quot;</span>, <span class="at">annotation =</span> <span class="st">&quot;20a1.hg38&quot;</span>)</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (nProbes <span class="sc">&gt;</span> <span class="dv">622000</span>) {</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>    <span class="co"># EPIC (850k) array</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>    arrayAnnotation <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">array =</span> <span class="st">&quot;IlluminaHumanMethylationEPIC&quot;</span>, <span class="at">annotation =</span> <span class="st">&quot;ilm10b4.hg19&quot;</span>)</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (nProbes <span class="sc">&gt;</span> <span class="dv">480000</span>) {</span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>    <span class="co"># 450k array</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>    arrayAnnotation <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">array =</span> <span class="st">&quot;IlluminaHumanMethylation450k&quot;</span>, <span class="at">annotation =</span> <span class="st">&quot;ilmn12.hg19&quot;</span>)</span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a>    <span class="co"># 27k array</span></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a>    arrayAnnotation <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">array =</span> <span class="st">&quot;IlluminaHumanMethylation27k&quot;</span>, <span class="at">annotation =</span> <span class="st">&quot;ilmn12.hg19&quot;</span>)</span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a>  }</span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a></span>
<span id="cb5-16"><a href="#cb5-16" tabindex="-1"></a>  arrayAnnotation</span>
<span id="cb5-17"><a href="#cb5-17" tabindex="-1"></a>}</span>
<span id="cb5-18"><a href="#cb5-18" tabindex="-1"></a></span>
<span id="cb5-19"><a href="#cb5-19" tabindex="-1"></a><span class="fu">environment</span>(.guessArrayTypes) <span class="ot">&lt;-</span> <span class="fu">asNamespace</span>(<span class="st">&quot;minfi&quot;</span>)</span>
<span id="cb5-20"><a href="#cb5-20" tabindex="-1"></a><span class="fu">assignInNamespace</span>(<span class="st">&quot;.guessArrayTypes&quot;</span>, .guessArrayTypes, <span class="at">ns =</span> <span class="st">&quot;minfi&quot;</span>)</span></code></pre></div>
<p>Then, proceed using <code>BiocParallel</code> and <code>minfi</code>
to read in the IDAT files.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>RGset <span class="ot">&lt;-</span> <span class="fu">read.metharray.exp</span>(<span class="at">base=</span><span class="st">&quot;../GSE116339/IDATs/&quot;</span>, targets, <span class="at">verbose=</span><span class="cn">FALSE</span>, <span class="at">extended=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<p>Reading data in parallel is subject to errors and debugging is often
difficult. Recently, <a
href="https://bioconductor.org/packages/release/bioc/html/BiocParallel.html"><strong>BiocParallel</strong></a>
has been extended with a comprehensive set of functions for debugging on
various parallel architectures. If problems arise, we recommend using
<code>BatchJobsParam()</code> with the <code>log=TRUE</code> option in
order to facilitate resolution.</p>
<p>Our data is now an <code>RGset</code> object that can used for
visualization. You can see below that inside this object the
<code>colData</code> holds the same information as <code>targets</code>,
and that there are 5 <code>assay</code> layers. The annotation
information tells us that the methylation was measured using a EPIC
array and that hg19 is the reference genome.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>RGset</span></code></pre></div>
<pre><code>## class: RGChannelSetExtended 
## dim: 1052641 60 
## metadata(0):
## assays(5): Green Red GreenSD RedSD NBeads
## rownames(1052641): 1600101 1600111 ... 99810990 99810992
## rowData names(0):
## colnames(60): GSM3228562_200550980002_R01C01 GSM3228563_200550980002_R02C01 ... GSM3228620_200590490021_R07C01
##   GSM3228621_200590490021_R08C01
## colData names(14): sample_ID geo_accession ... Basename filenames
## Annotation
##   array: IlluminaHumanMethylationEPIC
##   annotation: 20a1.hg38</code></pre>
<hr />
</div>
<div id="beta-values" class="section level1">
<h1>Beta values</h1>
<p>In order to further visualize the data, we store the beta values
using the <code>getBeta()</code> function from <a
href="https://bioconductor.org/packages/release/bioc/html/minfi.html"><strong>minfi</strong></a>.
The <code>type="Illumina"</code> option adds 100 to the denominator of
the beta-value calculation, preventing NA values being recorded when the
methylated and unmethylated signal are both 0.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>betas <span class="ot">&lt;-</span> <span class="fu">getBeta</span>(RGset, <span class="at">type=</span><span class="st">&quot;Illumina&quot;</span>)</span></code></pre></div>
<p>On some environments, this function returns a list. In that case, the
beta-values can be extracted using <code>betas$completeObs</code>.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">is.list</span>(betas)){</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>  betas <span class="ot">&lt;-</span> betas<span class="sc">$</span>completeObs</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>}</span></code></pre></div>
<p>It can also be advisable to round the beta values to 4 significant
figures, which both reduces the data size and provides a realistic
measure of the available precision.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>betas <span class="ot">&lt;-</span> <span class="fu">round</span>(betas, <span class="dv">4</span>)</span></code></pre></div>
<p>Now, <code>betas</code> contains beta values for each measured probe
and sample.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="fu">dim</span>(betas)</span></code></pre></div>
<pre><code>## [1] 866836     60</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>betas[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##            GSM3228562_200550980002_R01C01 GSM3228563_200550980002_R02C01 GSM3228564_200550980002_R03C01 GSM3228565_200550980002_R04C01
## cg18478105                         0.0504                         0.0000                         0.0432                         0.0396
## cg09835024                         0.0745                         0.1027                         0.0548                         0.0499
## cg14361672                         0.6650                         0.6541                         0.6505                         0.6050
## cg01763666                         0.6506                         0.6182                         0.6523                         0.6663
## cg12950382                         0.8633                         0.8001                         0.8153                         0.8313
##            GSM3228566_200550980002_R05C01
## cg18478105                         0.0454
## cg09835024                         0.0750
## cg14361672                         0.7440
## cg01763666                         0.6309
## cg12950382                         0.7643</code></pre>
<hr />
</div>
<div id="methylaid" class="section level1">
<h1>MethylAid</h1>
<div id="control-probes" class="section level2">
<h2>Control probes</h2>
<p><code>MethylAid</code> plots to visualize sample quality are included
in the <code>DNAmArray</code> package and their use is outlined
below.</p>
<p>They rely on the following <code>qcProbes</code> object that
specifies the control probes.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>qcProbes <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>  <span class="at">BSI =</span> <span class="st">&quot;^BISULFITE CONVERSION I$&quot;</span>,</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>  <span class="at">BSII =</span> <span class="st">&quot;^BISULFITE CONVERSION II$&quot;</span>,</span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a>  <span class="at">EC =</span> <span class="st">&quot;^EXTENSION$&quot;</span>,</span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a>  <span class="at">SPI =</span> <span class="st">&quot;^SPECIFICITY I$&quot;</span>,</span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a>  <span class="at">HYB =</span> <span class="st">&quot;^HYBRIDIZATION$&quot;</span>,</span>
<span id="cb16-7"><a href="#cb16-7" tabindex="-1"></a>  <span class="at">NP =</span> <span class="st">&quot;^NON-POLYMORPHIC$&quot;</span>,</span>
<span id="cb16-8"><a href="#cb16-8" tabindex="-1"></a>  <span class="at">SPII =</span> <span class="st">&quot;^SPECIFICITY II$&quot;</span>,</span>
<span id="cb16-9"><a href="#cb16-9" tabindex="-1"></a>  <span class="at">TR =</span> <span class="st">&quot;^TARGET REMOVAL$&quot;</span>,</span>
<span id="cb16-10"><a href="#cb16-10" tabindex="-1"></a>  <span class="at">SC =</span> <span class="st">&quot;^STAINING$&quot;</span>,</span>
<span id="cb16-11"><a href="#cb16-11" tabindex="-1"></a>  <span class="at">NC =</span> <span class="st">&quot;^NEGATIVE$&quot;</span></span>
<span id="cb16-12"><a href="#cb16-12" tabindex="-1"></a>)</span></code></pre></div>
<p>They also rely on array specific thresholds, which we advise to set
on the basis of the following table:</p>
<table>
<thead>
<tr class="header">
<th>Array</th>
<th>MU</th>
<th>OP</th>
<th>BS</th>
<th>HC</th>
<th>DP</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>450k</td>
<td>10.5</td>
<td>11.75</td>
<td>12.75</td>
<td>13.25</td>
<td>0.95</td>
</tr>
<tr class="even">
<td>EPIC</td>
<td>10</td>
<td>12</td>
<td>11.75</td>
<td>12.75</td>
<td>0.95</td>
</tr>
<tr class="odd">
<td>EPIC v2</td>
<td>10.5</td>
<td>12.5</td>
<td>12.75</td>
<td>13.25</td>
<td>0.95</td>
</tr>
</tbody>
</table>
<p>The example data uses the EPIC array and the following thresholds</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>thresholds <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">10</span>, <span class="dv">12</span>, <span class="fl">11.75</span>, <span class="fl">12.75</span>, <span class="fl">0.95</span>)</span></code></pre></div>
<hr />
</div>
<div id="plots" class="section level2">
<h2>Plots</h2>
<p>All plots can be coloured by a variable in <code>targets</code>,
specified by the <code>col</code> argument.</p>
<p><code>plotMU</code> plots a rotated MU plot. This shows the median
methylated and unmethylated log2 intensities per sample, and flagging
low-intensity outliers.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="fu">plotMU</span>(<span class="at">object =</span> sData, <span class="at">threshold =</span> thresholds[<span class="dv">1</span>], <span class="at">col=</span><span class="st">&quot;plate&quot;</span>)</span></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/212plotMU-1.png" width="672" /></p>
<hr />
<p><code>plotOP</code> plots the sample-dependent overall quality
control on the basis of non-polymorphic (NP) control probes.</p>
<p>NP control probes query each of the four nucleotides in a
non-polymorphic region of the bisulfite genome. Signal intensity from
these probes in the red (A and T) and green (C and G) channels can then
be used to test overall performance of the assay, from amplification to
detection. Intensity should be high in the red channel for NP probes
querying A and T nucleotides, and high in the green channel for NP
probes querying G and C nucleotides. The intensities for the relevant
channel for NP probes are then combined and plotted per sample, where
they should be above the specific threshold.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="fu">plotOP</span>(<span class="at">object =</span> sData, <span class="at">threshold =</span> thresholds[<span class="dv">2</span>], <span class="at">col=</span><span class="st">&quot;plate&quot;</span>)</span></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/213plotOP-1.png" width="672" /></p>
<hr />
<p><code>plotBS</code> assesses bisulfite conversion efficiency using
control probes, identifying poorly converted samples. Type I BC control
probes monitor the efficiency of the bisulfite conversion. If the
conversion reaction was successful, the ‘C’ (converted) probes will be
extended, but if the sample has unconverted DNA, the ‘U’ (unconverted)
probes are extended instead.</p>
<p>Performance of the BC control probes C1 and C2 is monitored in the
green channel, and C3 and C4 are monitored in the red channel. The
intensities for the relevant channels are then combined for all BC
control probes and plotted per sample to ensure they are above the
specified threshold.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="fu">plotBS</span>(<span class="at">object =</span> sData, <span class="at">threshold =</span> thresholds[<span class="dv">3</span>], <span class="at">col=</span><span class="st">&quot;plate&quot;</span>)</span></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/214plotBS-1.png" width="672" /></p>
<hr />
<p><code>plotHC</code> uses hybridization controls to assess the
hybridization step using synthetic targets, which complement the array
perfectly. These synthetic targets are present in the hybridization
buffer (RA1) at three concentration levels, and their performance is
only monitored in the green channel. The difference in green intensity
between the high (H) and low (L) concentration is combined and can be
plotted to ensure it is above the specified threshold.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="fu">plotHC</span>(<span class="at">object =</span> sData, <span class="at">threshold =</span> thresholds[<span class="dv">4</span>], <span class="at">col=</span><span class="st">&quot;plate&quot;</span>)</span></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/215plotHC-1.png" width="672" /></p>
<hr />
<p><code>plotDP</code> is the last MethylAid plot we provide. This
function instead takes the <code>RGset</code> as input, to enable the
user to specify both how visually distinct from the background they
consider a detected signal (<code>detP</code> argument is the detection
p-value) and what proportion of probes they want to be meeting this
criterion (<code>threshold</code> is this proportion). The default we
suggest for all arrays is 0.01 (detP) and 0.95 (threshold), but this can
be adjusted.</p>
<p>This plot uses negative control probes, which are randomly permuted
sequences that should not hybridize to the DNA template. The mean signal
of these probes defines the background signal. This plot therefore show
the fraction of probe visually distinct from the background signal in
each sample, coloured by array number.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="fu">plotDP</span>(<span class="at">object =</span> RGset, <span class="at">threshold =</span> <span class="fl">0.95</span>, <span class="at">detP=</span><span class="fl">0.01</span>, <span class="at">col=</span><span class="st">&quot;plate&quot;</span>)</span></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/216plotDP-1.png" width="672" /></p>
<hr />
</div>
<div id="outliers" class="section level2">
<h2>Outliers</h2>
<p>On the basis of the above thresholds, outliers can be extracted using
the <code>get_outliers</code> function. The resulting table shows an
indicator of which quality control step was failed.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a>outliers_auto <span class="ot">&lt;-</span> <span class="fu">get_outliers</span>(<span class="at">object =</span> sData, <span class="at">thresholds =</span> thresholds)</span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a>outliers_auto</span></code></pre></div>
<pre><code>##                                   MU    OP    BS   HC    DP
## GSM3228562_200550980002_R01C01 FALSE FALSE FALSE TRUE FALSE
## GSM3228577_200550980025_R01C01 FALSE FALSE FALSE TRUE FALSE</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>outliers <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">rownames</span>(outliers_auto), <span class="dv">1</span>, <span class="dv">10</span>)</span></code></pre></div>
<p>Outliers can then be removed from <code>targets</code> before
proceding.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>targets <span class="ot">&lt;-</span> targets <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>geo_accession <span class="sc">%in%</span> outliers)</span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a><span class="fu">dim</span>(targets)</span></code></pre></div>
<pre><code>## [1] 58 13</code></pre>
<hr />
</div>
</div>
<div id="beta-density-plots" class="section level1">
<h1>Beta density plots</h1>
<p>Using <code>densityPlot()</code> from <a
href="https://bioconductor.org/packages/release/bioc/html/minfi.html"><strong>minfi</strong></a>,
we can visualize the per sample average beta-value distribution. This
gives us a global impression of the data and allows us to identify
possible anomalous samples. We expect this distribution to be bimodal
with the peaks representing methylated and unmethylated signals. Any
centre peaks should be further investigated for problems, such as
ambiguous mapping.</p>
<p>Centre peaks can be identified using a <code>beta_outlier</code>
object, which flags any samples with high beta values in the central
part of the distribution. Alternatively, a named boolean vector of the
samples can be used to flag a specific sample.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a>beta_outlier <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">colSums</span>(betas <span class="sc">&gt;</span> <span class="fl">0.3</span> <span class="sc">&amp;</span> betas <span class="sc">&lt;</span> <span class="fl">0.5</span>) <span class="sc">&gt;</span> <span class="dv">120000</span>, T, F)</span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a>RGset<span class="sc">$</span>beta_outlier <span class="ot">&lt;-</span> beta_outlier</span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a></span>
<span id="cb28-4"><a href="#cb28-4" tabindex="-1"></a>beta_outlier[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>## GSM3228562_200550980002_R01C01 GSM3228563_200550980002_R02C01 GSM3228564_200550980002_R03C01 GSM3228565_200550980002_R04C01 
##                          FALSE                          FALSE                          FALSE                          FALSE 
## GSM3228566_200550980002_R05C01 
##                          FALSE</code></pre>
<p>These are then coloured in the beta values plot.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a><span class="fu">densityPlot</span>(RGset, </span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a>            <span class="at">main=</span><span class="st">&quot;Beta density plot&quot;</span>, </span>
<span id="cb30-3"><a href="#cb30-3" tabindex="-1"></a>            <span class="at">xlab=</span><span class="st">&quot;Beta values&quot;</span>,</span>
<span id="cb30-4"><a href="#cb30-4" tabindex="-1"></a>            <span class="at">sampGroups =</span> beta_outlier) </span></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/220betaplot-1.png" width="672" /></p>
<p>Any flagged samples can then be removed using the sample name.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a>flagged_samples <span class="ot">&lt;-</span> <span class="fu">names</span>(beta_outlier[beta_outlier <span class="sc">==</span> T])</span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a>flagged_samples</span></code></pre></div>
<pre><code>## [1] &quot;GSM3228582_200550980025_R06C01&quot;</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a>targets <span class="ot">&lt;-</span> targets[<span class="sc">!</span><span class="fu">rownames</span>(targets) <span class="sc">%in%</span> flagged_samples, ]</span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a></span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a>RGset <span class="ot">&lt;-</span> RGset[,<span class="fu">colnames</span>(RGset) <span class="sc">%in%</span> <span class="fu">rownames</span>(targets)]</span>
<span id="cb33-4"><a href="#cb33-4" tabindex="-1"></a>betas <span class="ot">&lt;-</span> betas[,<span class="fu">colnames</span>(betas) <span class="sc">%in%</span> <span class="fu">rownames</span>(targets)]</span></code></pre></div>
<hr />
</div>
<div id="principal-components-analysis" class="section level1">
<h1>Principal components analysis</h1>
<div id="calculate-pcs" class="section level2">
<h2>Calculate PCs</h2>
<p>Principal components (PCs) of the beta values can be calculated using
<code>prcomp</code>.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a>tbetas <span class="ot">&lt;-</span> <span class="fu">t</span>(betas)</span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(tbetas, <span class="at">center =</span> <span class="cn">TRUE</span>, <span class="at">rank. =</span> <span class="dv">10</span>)</span>
<span id="cb34-3"><a href="#cb34-3" tabindex="-1"></a></span>
<span id="cb34-4"><a href="#cb34-4" tabindex="-1"></a><span class="fu">summary</span>(pca)</span></code></pre></div>
<pre><code>## Importance of first k=10 (out of 57) components:
##                            PC1      PC2      PC3     PC4     PC5     PC6     PC7     PC8     PC9   PC10
## Standard deviation     38.1497 14.16558 10.03321 8.36472 7.84595 7.49317 5.46166 5.28598 5.13590 4.9653
## Proportion of Variance  0.5373  0.07409  0.03717 0.02583 0.02273 0.02073 0.01101 0.01032 0.00974 0.0091
## Cumulative Proportion   0.5373  0.61144  0.64860 0.67444 0.69717 0.71790 0.72891 0.73923 0.74896 0.7581</code></pre>
<hr />
</div>
<div id="screeplot" class="section level2">
<h2>Screeplot</h2>
<p>Variance explained by each PC can be calculated.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a>var_explained <span class="ot">=</span></span>
<span id="cb36-2"><a href="#cb36-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">PC =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(betas),</span>
<span id="cb36-3"><a href="#cb36-3" tabindex="-1"></a>             <span class="at">var_explained =</span> pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span> <span class="sc">/</span> <span class="fu">sum</span>(pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>))[<span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(pca<span class="sc">$</span>x),]</span>
<span id="cb36-4"><a href="#cb36-4" tabindex="-1"></a></span>
<span id="cb36-5"><a href="#cb36-5" tabindex="-1"></a>var_explained</span></code></pre></div>
<pre><code>##    PC var_explained
## 1   1   0.537349884
## 2   2   0.074087375
## 3   3   0.037166859
## 4   4   0.025833196
## 5   5   0.022728271
## 6   6   0.020730371
## 7   7   0.011013480
## 8   8   0.010316377
## 9   9   0.009738856
## 10 10   0.009102599</code></pre>
<p>This can be plotted in a screeplot to visualize the structure of the
DNAm data.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a>var_explained <span class="sc">%&gt;%</span> </span>
<span id="cb38-2"><a href="#cb38-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>PC, <span class="at">y=</span>var_explained)) <span class="sc">+</span></span>
<span id="cb38-3"><a href="#cb38-3" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb38-4"><a href="#cb38-4" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color=</span><span class="st">&#39;grey5&#39;</span>, <span class="at">fill=</span><span class="st">&#39;#6DACBC&#39;</span>, <span class="at">shape=</span><span class="dv">21</span>, <span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span> </span>
<span id="cb38-5"><a href="#cb38-5" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks=</span><span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(pca<span class="sc">$</span>x)) <span class="sc">+</span></span>
<span id="cb38-6"><a href="#cb38-6" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Principal Component&quot;</span>) <span class="sc">+</span> </span>
<span id="cb38-7"><a href="#cb38-7" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Proportion of variance explained&quot;</span>) <span class="sc">+</span></span>
<span id="cb38-8"><a href="#cb38-8" tabindex="-1"></a>  <span class="fu">theme_bw</span>()</span></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/224screeplot-1.png" width="672" /></p>
<hr />
</div>
<div id="heatmap" class="section level2">
<h2>Heatmap</h2>
<p>Any constant variables are removed from the heatmap, as they will not
explain variance in the data.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a>plot_vars <span class="ot">&lt;-</span> <span class="fu">apply</span>(targets, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">sd</span>(<span class="fu">as.numeric</span>(<span class="fu">factor</span>(x)), <span class="at">na.rm=</span>T))</span>
<span id="cb39-2"><a href="#cb39-2" tabindex="-1"></a></span>
<span id="cb39-3"><a href="#cb39-3" tabindex="-1"></a>plot_vars <span class="ot">&lt;-</span> <span class="fu">names</span>(plot_vars[<span class="sc">!</span>plot_vars <span class="sc">%in%</span> <span class="fu">c</span>(<span class="cn">NA</span>, <span class="dv">0</span>)])</span>
<span id="cb39-4"><a href="#cb39-4" tabindex="-1"></a>plot_vars</span></code></pre></div>
<pre><code>##  [1] &quot;sample_ID&quot;          &quot;geo_accession&quot;      &quot;sex&quot;                &quot;age&quot;                &quot;log_total_pbb&quot;      &quot;pbb_153&quot;           
##  [7] &quot;pbb_77&quot;             &quot;pbb_101&quot;            &quot;pbb_180&quot;            &quot;supplementary_file&quot; &quot;plate&quot;              &quot;row&quot;               
## [13] &quot;Basename&quot;</code></pre>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a>heatmap_df <span class="ot">&lt;-</span> targets <span class="sc">%&gt;%</span> </span>
<span id="cb41-2"><a href="#cb41-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">any_of</span>(plot_vars))</span></code></pre></div>
<p>All variables are then converted to numeric and correlations between
them and the PCs are calculated.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a>heatmap_df <span class="ot">&lt;-</span> <span class="fu">apply</span>(heatmap_df, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">as.numeric</span>(<span class="fu">factor</span>(x)))</span>
<span id="cb42-2"><a href="#cb42-2" tabindex="-1"></a></span>
<span id="cb42-3"><a href="#cb42-3" tabindex="-1"></a>cxy <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">cor</span>(pca<span class="sc">$</span>x, <span class="fu">scale</span>(heatmap_df), <span class="at">use=</span><span class="st">&quot;pairwise.complete.obs&quot;</span>),<span class="dv">2</span>) </span></code></pre></div>
<p>A heatmap can then be used to visualize these correlations and
uncover measured variables that explain a large proportion of DNAm
variance.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a>col_fun <span class="ot">&lt;-</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">1</span>), <span class="fu">c</span>(<span class="st">&quot;#000042&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;#800000&quot;</span>))</span>
<span id="cb43-2"><a href="#cb43-2" tabindex="-1"></a></span>
<span id="cb43-3"><a href="#cb43-3" tabindex="-1"></a><span class="fu">Heatmap</span>(</span>
<span id="cb43-4"><a href="#cb43-4" tabindex="-1"></a>  <span class="fu">t</span>(cxy),                              </span>
<span id="cb43-5"><a href="#cb43-5" tabindex="-1"></a>  <span class="at">col =</span> col_fun,  </span>
<span id="cb43-6"><a href="#cb43-6" tabindex="-1"></a>  <span class="at">border =</span> <span class="st">&#39;grey5&#39;</span>,</span>
<span id="cb43-7"><a href="#cb43-7" tabindex="-1"></a>  <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,            </span>
<span id="cb43-8"><a href="#cb43-8" tabindex="-1"></a>  <span class="at">show_row_dend =</span> <span class="cn">TRUE</span>,             </span>
<span id="cb43-9"><a href="#cb43-9" tabindex="-1"></a>  <span class="at">show_column_dend =</span> <span class="cn">FALSE</span>,      </span>
<span id="cb43-10"><a href="#cb43-10" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&quot;Value&quot;</span>,                 </span>
<span id="cb43-11"><a href="#cb43-11" tabindex="-1"></a>  <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">8</span>), </span>
<span id="cb43-12"><a href="#cb43-12" tabindex="-1"></a>  <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">8</span>), </span>
<span id="cb43-13"><a href="#cb43-13" tabindex="-1"></a>  <span class="at">cell_fun =</span> <span class="cf">function</span>(j, i, x, y, width, height, fill) {</span>
<span id="cb43-14"><a href="#cb43-14" tabindex="-1"></a>    <span class="fu">grid.rect</span>(x, y, width, height, </span>
<span id="cb43-15"><a href="#cb43-15" tabindex="-1"></a>              <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">col =</span> <span class="st">&quot;white&quot;</span>, <span class="at">lwd =</span> <span class="dv">1</span>, <span class="at">fill =</span> <span class="cn">NA</span>))</span>
<span id="cb43-16"><a href="#cb43-16" tabindex="-1"></a>    <span class="fu">grid.text</span>(<span class="fu">ifelse</span>(<span class="fu">abs</span>(<span class="fu">t</span>(cxy)[i,j]) <span class="sc">&gt;</span> <span class="fl">0.4</span>,</span>
<span id="cb43-17"><a href="#cb43-17" tabindex="-1"></a>                     <span class="fu">sprintf</span>(<span class="st">&quot;%.2f&quot;</span>, <span class="fu">round</span>(<span class="fu">t</span>(cxy)[i, j], <span class="dv">2</span>)),</span>
<span id="cb43-18"><a href="#cb43-18" tabindex="-1"></a>                     <span class="st">&quot;&quot;</span>), </span>
<span id="cb43-19"><a href="#cb43-19" tabindex="-1"></a>              x, y, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">10</span>, <span class="at">col =</span> <span class="st">&quot;white&quot;</span>))</span>
<span id="cb43-20"><a href="#cb43-20" tabindex="-1"></a>  }</span>
<span id="cb43-21"><a href="#cb43-21" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/227heatmap-1.png" width="672" /></p>
<hr />
</div>
<div id="plot" class="section level2">
<h2>Plot</h2>
<p>In order to further investigate flagged variables, we can add the PCs
to the <code>targets</code> data frame.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" tabindex="-1"></a>pc_df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(targets, pca<span class="sc">$</span>x)</span></code></pre></div>
<p>PCs can then be plotted against each other, coloured by flagged
variables.</p>
<p>Strong batch effects can be seen, coloured by array number.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" tabindex="-1"></a>pc_df <span class="sc">%&gt;%</span> </span>
<span id="cb45-2"><a href="#cb45-2" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> plate)) <span class="sc">+</span></span>
<span id="cb45-3"><a href="#cb45-3" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb45-4"><a href="#cb45-4" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">paste0</span>(<span class="st">&quot;PC1 (&quot;</span>, <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span>var_explained[<span class="dv">1</span>,<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">&quot;%)&quot;</span>), </span>
<span id="cb45-5"><a href="#cb45-5" tabindex="-1"></a>         <span class="at">y =</span> <span class="fu">paste0</span>(<span class="st">&quot;PC2 (&quot;</span>, <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span>var_explained[<span class="dv">2</span>,<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">&quot;%)&quot;</span>), </span>
<span id="cb45-6"><a href="#cb45-6" tabindex="-1"></a>         <span class="at">color =</span> <span class="st">&quot;Array&quot;</span>) <span class="sc">+</span></span>
<span id="cb45-7"><a href="#cb45-7" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/229pcplot1-1.png" width="672" /></p>
<p>The second PC is mostly linked to sex differences.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" tabindex="-1"></a>pc_df <span class="sc">%&gt;%</span> </span>
<span id="cb46-2"><a href="#cb46-2" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> sex)) <span class="sc">+</span></span>
<span id="cb46-3"><a href="#cb46-3" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb46-4"><a href="#cb46-4" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">paste0</span>(<span class="st">&quot;PC1 (&quot;</span>, <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span>var_explained[<span class="dv">1</span>,<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">&quot;%)&quot;</span>), </span>
<span id="cb46-5"><a href="#cb46-5" tabindex="-1"></a>         <span class="at">y =</span> <span class="fu">paste0</span>(<span class="st">&quot;PC2 (&quot;</span>, <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span>var_explained[<span class="dv">2</span>,<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">&quot;%)&quot;</span>), </span>
<span id="cb46-6"><a href="#cb46-6" tabindex="-1"></a>         <span class="at">color =</span> <span class="st">&quot;Sex&quot;</span>) <span class="sc">+</span></span>
<span id="cb46-7"><a href="#cb46-7" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span></code></pre></div>
<p><img src="02_SampleQualityControl_files/figure-html/230pcplot2-1.png" width="672" /></p>
<hr />
</div>
</div>
<div id="checking-sample-relationships" class="section level1">
<h1>Checking Sample Relationships</h1>
<div id="annotations" class="section level2">
<h2>Annotations</h2>
<p><a
href="http://bioconductor.org/packages/release/bioc/html/omicsPrint.html"><strong>omicsPrint</strong></a><sup>12</sup>
is a package we developed to detect data linkage errors through
inspecting sample relations in multiple omics studies.</p>
<p>Manifests can be downloaded from the <a
href="https://zwdzwd.github.io/InfiniumAnnotation">Zhou lab GitHub
page</a><sup>15</sup> for all commonly used arrays and reference
genomes. For this example, we use SNP annotations downloaded for the
EPIC array and the hg19 genome.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" tabindex="-1"></a>snp_cpgs <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(</span>
<span id="cb47-2"><a href="#cb47-2" tabindex="-1"></a>  <span class="st">&quot;./data/EPIC.hg19.manifest.pop.tsv.gz&quot;</span>)</span></code></pre></div>
<pre><code>## Rows: 865918 Columns: 66
## ── Column specification ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────
## Delimiter: &quot;\t&quot;
## chr  (2): CpG_chrm, probeID
## dbl  (2): CpG_beg, CpG_end
## lgl (62): MASK_general_AFR, MASK_snp5_AFR, MASK_general_EAS, MASK_snp5_EAS, MASK_general_EUR, MASK_snp5_EUR, MASK_general_SAS, MASK_snp...
## 
## ℹ Use `spec()` to retrieve the full column specification for this data.
## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<p>Columns of interest include:</p>
<ul>
<li>cpg - ID of the probe for the CpG</li>
<li>chr - Chromosome where the CpG is located</li>
<li>start - Start position of the CpG</li>
<li>end - End position of the CpG</li>
</ul>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" tabindex="-1"></a>snp_cpgs <span class="ot">&lt;-</span> snp_cpgs <span class="sc">%&gt;%</span> </span>
<span id="cb49-2"><a href="#cb49-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb49-3"><a href="#cb49-3" tabindex="-1"></a>    <span class="at">cpg =</span> probeID,</span>
<span id="cb49-4"><a href="#cb49-4" tabindex="-1"></a>    <span class="at">chr =</span> CpG_chrm,</span>
<span id="cb49-5"><a href="#cb49-5" tabindex="-1"></a>    <span class="at">start =</span> CpG_beg,</span>
<span id="cb49-6"><a href="#cb49-6" tabindex="-1"></a>    <span class="at">end =</span> CpG_end,</span>
<span id="cb49-7"><a href="#cb49-7" tabindex="-1"></a>    MASK_snp5_EUR</span>
<span id="cb49-8"><a href="#cb49-8" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb49-9"><a href="#cb49-9" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb49-10"><a href="#cb49-10" tabindex="-1"></a>    <span class="at">chr =</span> <span class="fu">substr</span>(chr,<span class="dv">4</span>,<span class="dv">5</span>)</span>
<span id="cb49-11"><a href="#cb49-11" tabindex="-1"></a>  )</span>
<span id="cb49-12"><a href="#cb49-12" tabindex="-1"></a></span>
<span id="cb49-13"><a href="#cb49-13" tabindex="-1"></a>snp_cpgs <span class="ot">&lt;-</span> (snp_cpgs <span class="sc">%&gt;%</span> </span>
<span id="cb49-14"><a href="#cb49-14" tabindex="-1"></a>               dplyr<span class="sc">::</span><span class="fu">filter</span>(MASK_snp5_EUR <span class="sc">==</span> <span class="cn">TRUE</span>))<span class="sc">$</span>cpg</span>
<span id="cb49-15"><a href="#cb49-15" tabindex="-1"></a></span>
<span id="cb49-16"><a href="#cb49-16" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">&quot;There are &quot;</span>, <span class="fu">length</span>(snp_cpgs),</span>
<span id="cb49-17"><a href="#cb49-17" tabindex="-1"></a>             <span class="st">&quot; CpGs containing common European SNPs&quot;</span>))</span></code></pre></div>
<pre><code>## [1] &quot;There are 28152 CpGs containing common European SNPs&quot;</code></pre>
<p>Betas can then be subset to include only these CpG probes of
interest.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" tabindex="-1"></a>betas_snps <span class="ot">&lt;-</span> betas[snp_cpgs, ]</span>
<span id="cb51-2"><a href="#cb51-2" tabindex="-1"></a><span class="fu">dim</span>(betas_snps)</span></code></pre></div>
<pre><code>## [1] 28152    57</code></pre>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">rownames</span>(targets) <span class="sc">==</span> <span class="fu">colnames</span>(betas_snps))</span></code></pre></div>
<pre><code>## 
## TRUE 
##   57</code></pre>
<hr />
<div id="assumed-relations" class="section level3">
<h3>Assumed relations</h3>
<p>Create a data frame of all pairwise sample comparisons.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" tabindex="-1"></a>relations <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(</span>
<span id="cb55-2"><a href="#cb55-2" tabindex="-1"></a>  <span class="at">idx =</span> <span class="fu">colnames</span>(betas_snps), </span>
<span id="cb55-3"><a href="#cb55-3" tabindex="-1"></a>  <span class="at">idy =</span> <span class="fu">colnames</span>(betas_snps))</span>
<span id="cb55-4"><a href="#cb55-4" tabindex="-1"></a><span class="fu">head</span>(relations)</span></code></pre></div>
<pre><code>##                              idx                            idy
## 1 GSM3228563_200550980002_R02C01 GSM3228563_200550980002_R02C01
## 2 GSM3228564_200550980002_R03C01 GSM3228563_200550980002_R02C01
## 3 GSM3228565_200550980002_R04C01 GSM3228563_200550980002_R02C01
## 4 GSM3228566_200550980002_R05C01 GSM3228563_200550980002_R02C01
## 5 GSM3228567_200550980002_R06C01 GSM3228563_200550980002_R02C01
## 6 GSM3228568_200550980002_R07C01 GSM3228563_200550980002_R02C01</code></pre>
<p>Sample names are saved, as a reference for which pairs exist.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" tabindex="-1"></a>relations<span class="sc">$</span>sample_name_x <span class="ot">&lt;-</span> targets[relations<span class="sc">$</span>idx, <span class="st">&quot;sample_ID&quot;</span>] </span>
<span id="cb57-2"><a href="#cb57-2" tabindex="-1"></a>relations<span class="sc">$</span>sample_name_y <span class="ot">&lt;-</span> targets[relations<span class="sc">$</span>idy, <span class="st">&quot;sample_ID&quot;</span>]</span></code></pre></div>
<p>A relation_type variable is then set as <code>identical</code> if the
samples come from the same individual and <code>unrelated</code>
otherwise. We see that the 59 pairs comparing the same sample are set as
<code>identical</code>.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" tabindex="-1"></a>relations<span class="sc">$</span>relation_type <span class="ot">&lt;-</span> <span class="st">&quot;unrelated&quot;</span></span>
<span id="cb58-2"><a href="#cb58-2" tabindex="-1"></a>relations<span class="sc">$</span>relation_type[relations<span class="sc">$</span>sample_name_x <span class="sc">==</span> relations<span class="sc">$</span>sample_name_y] <span class="ot">&lt;-</span> <span class="st">&quot;identical&quot;</span></span>
<span id="cb58-3"><a href="#cb58-3" tabindex="-1"></a></span>
<span id="cb58-4"><a href="#cb58-4" tabindex="-1"></a><span class="fu">table</span>(relations<span class="sc">$</span>relation_type)</span></code></pre></div>
<pre><code>## 
## identical unrelated 
##        57      3192</code></pre>
<hr />
</div>
<div id="genotyping" class="section level3">
<h3>Genotyping</h3>
<p>The function <code>beta2genotype</code> from <code>omicsPrint</code>
genotypes the observations by measuring homozygous or heterozygous
alleles at these SNP probes. The arguments specified are:</p>
<ul>
<li><code>betas</code> - a beta matrix of probes possibly affected by
SNPs</li>
<li><code>na.rm</code> - whether to drop CpGs for which no clustering is
observed</li>
<li><code>minSep</code> - the minimal cluster separation</li>
<li><code>minSize</code> - the size of the smallest cluster as a
percentage</li>
<li><code>centers</code> - centers of clusters</li>
<li><code>assayName</code> - the name of the assay to be used</li>
</ul>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" tabindex="-1"></a>genotype <span class="ot">&lt;-</span> <span class="fu">beta2genotype</span>(betas_snps, </span>
<span id="cb60-2"><a href="#cb60-2" tabindex="-1"></a>                          <span class="at">na.rm=</span><span class="cn">TRUE</span>,</span>
<span id="cb60-3"><a href="#cb60-3" tabindex="-1"></a>                          <span class="at">minSep =</span> <span class="fl">0.25</span>,</span>
<span id="cb60-4"><a href="#cb60-4" tabindex="-1"></a>                          <span class="at">minSize =</span> <span class="dv">5</span>,</span>
<span id="cb60-5"><a href="#cb60-5" tabindex="-1"></a>                          <span class="at">centers =</span> <span class="fu">c</span>(<span class="fl">0.2</span>, <span class="fl">0.5</span>, <span class="fl">0.8</span>),</span>
<span id="cb60-6"><a href="#cb60-6" tabindex="-1"></a>                          <span class="at">assayName=</span><span class="cn">NULL</span>)</span>
<span id="cb60-7"><a href="#cb60-7" tabindex="-1"></a><span class="fu">str</span>(genotype)</span></code></pre></div>
<pre><code>##  int [1:1128, 1:57] 1 2 1 3 1 1 2 3 1 3 ...
##  - attr(*, &quot;dimnames&quot;)=List of 2
##   ..$ : chr [1:1128] &quot;cg12213037&quot; &quot;cg19405842&quot; &quot;cg24925741&quot; &quot;cg07584620&quot; ...
##   ..$ : chr [1:57] &quot;GSM3228563_200550980002_R02C01&quot; &quot;GSM3228564_200550980002_R03C01&quot; &quot;GSM3228565_200550980002_R04C01&quot; &quot;GSM3228566_200550980002_R05C01&quot; ...</code></pre>
<p><code>alleleSharing</code> can then assess the relationships between
different individuals, which can be unrelated, twins, or identical. This
uses identity by state (IBS), a genetic similarity measure that compares
at a single locus the genotypes between two individuals and counts the
number of alleles shared (0, 1 or 2).Essentially, if a sample is the
same then the mean IBS should be 2 (alleles) and the variance of this
should be low (close to 0).</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" tabindex="-1"></a>out <span class="ot">&lt;-</span> <span class="fu">alleleSharing</span>(genotype, <span class="at">relations =</span> relations)</span></code></pre></div>
<pre><code>## Hash relations</code></pre>
<pre><code>## Pruning 1128 SNPs ...</code></pre>
<pre><code>## 0 SNPs removed because of low call rate!</code></pre>
<pre><code>## 0 samples removed because too few SNPs called!</code></pre>
<pre><code>## Using 1128 polymorphic SNPs to determine allele sharing.</code></pre>
<pre><code>## Running `square` IBS algorithm!</code></pre>
<pre><code>## 58 of 1653 (3.51%) ...</code></pre>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" tabindex="-1"></a><span class="fu">head</span>(out)</span></code></pre></div>
<pre><code>##       mean       var                     colnames.x                     colnames.y  relation
## 1 2.000000 0.0000000 GSM3228563_200550980002_R02C01 GSM3228563_200550980002_R02C01 identical
## 2 1.333333 0.4229518 GSM3228564_200550980002_R03C01 GSM3228563_200550980002_R02C01 unrelated
## 3 1.333333 0.4158533 GSM3228565_200550980002_R04C01 GSM3228563_200550980002_R02C01 unrelated
## 4 1.359929 0.4204692 GSM3228566_200550980002_R05C01 GSM3228563_200550980002_R02C01 unrelated
## 5 1.335106 0.4306387 GSM3228567_200550980002_R06C01 GSM3228563_200550980002_R02C01 unrelated
## 6 1.322695 0.4139654 GSM3228568_200550980002_R07C01 GSM3228563_200550980002_R02C01 unrelated</code></pre>
</div>
<div id="identify-mismatches" class="section level3">
<h3>Identify mismatches</h3>
<p>The mean and variance of IBS can then be visualised using the
<code>inferRelations()</code> function. Genetically similar sample
comparisons will be plotted closer to the bottom, right of the graph,
and those that are genetically dissimilar are in the top, left. A table
of mismatches gives an overview of any identified.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" tabindex="-1"></a>mismatches <span class="ot">&lt;-</span> <span class="fu">inferRelations</span>(out, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>##                   Assumed relation
## Predicted relation identical unrelated
##          identical        57         .
##          unrelated         .      1596</code></pre>
<p><img src="02_SampleQualityControl_files/figure-html/239plot-1.png" width="672" /></p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" tabindex="-1"></a><span class="fu">table</span>(mismatches<span class="sc">$</span>relation, mismatches<span class="sc">$</span>predicted)</span></code></pre></div>
<pre><code>##            
##             identical unrelated
##   identical         0         0
##   unrelated         0         0</code></pre>
<p>Since there are no twins or relatives in our data, all observations
are shown as unrelated. In data with sample relationships, this would be
shown in the above graph as green or black clusters<sup>12</sup>. It is
important to carry out this type of visualization before probe-filtering
as otherwise the genotyping will be based on very few SNPs.</p>
<hr />
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
