<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title></title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 52px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h2 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h3 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h4 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h5 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h6 {
  padding-top: 57px;
  margin-top: -57px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DNAmArray</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="01_SampleQualityControl.html">Quality control</a>
</li>
<li>
  <a href="02_Visualization.html">Visualizations</a>
</li>
<li>
  <a href="03_Normalization.html">Normalization</a>
</li>
<li>
  <a href="04_ProbeFiltering.html">Probe filtering</a>
</li>
<li>
  <a href="05_Miscellaneous.html">Miscellaneous</a>
</li>
<li>
  <a href="06_EWAS.html">Ewas</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">About</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">




</div>


<hr />
<div id="beta-density-plots" class="section level1">
<h1>Beta density plots</h1>
<p>Using <code>densityPlot()</code> from <a href="https://bioconductor.org/packages/release/bioc/html/minfi.html"><strong>minfi</strong></a> <span class="citation">(Aryee et al. 2014)</span>, we can visualize the per sample average beta-value distribution. This gives us a global impression of the data and allows us to identify possible anomalous samples. We expect this distribution to be bimodal with the peaks representing methylated and unmethylated signals. Any centre peaks should be further investigated for problems, such as ambiguous mapping.</p>
<pre class="r"><code>library(minfi)
densityPlot(RGset, main=&quot;Beta density plot&quot;, xlab=&quot;Beta values&quot;)</code></pre>
<p><img src="02_Visualization_files/figure-html/302betaplot-1.png" width="672" /></p>
<p>For this data, the density plot is clearly bimodal with no obvious outliers. However, in order to further investigate the beta-values, we store them using the <code>getBeta()</code> function. Since some visualisations cannot handle NA values, we use the median to impute these. The <code>type=&quot;Illumina&quot;</code> option adds 100 to the denominator of the beta-value calculation, preventing NA values being recorded when the methylated and unmethylated signal are both 0.</p>
<pre class="r"><code>beta &lt;- getBeta(RGset, type=&quot;Illumina&quot;)
beta[is.na(beta)] &lt;- median(beta, na.rm=TRUE)</code></pre>
<hr />
</div>
<div id="principal-components-plot" class="section level1">
<h1>Principal components plot</h1>
<p>Using the <code>prcomp_irlba()</code> function from <a href="https://cran.r-project.org/web/packages/irlba/index.html"><strong>irlba</strong></a> we can calculate principal components. By assessing the amount of variance explained by these and visualising them, we can better interpret the data. For example, if there was evidence of clustering this might suggest grouping of samples, such as by sex if sex-chromosomes were not yet excluded.</p>
<pre class="r"><code>library(irlba)
pc &lt;- prcomp_irlba(t(beta), n=4)
summary(pc)</code></pre>
<pre><code>## Importance of components:
##                            PC1     PC2     PC3      PC4
## Standard deviation     37.6135 31.0639 20.9449 16.20783
## Proportion of Variance  0.3227  0.2201  0.1001  0.05992
## Cumulative Proportion   0.3227  0.5428  0.6429  0.70283</code></pre>
<pre class="r"><code>plot(pc$x, main=&quot;Principal components plot&quot;)</code></pre>
<p><img src="02_Visualization_files/figure-html/304pcplot-1.png" width="672" /></p>
<p>In this instance, our principal components explain nearly 65% of the variance in the data and there is no evidence of clustering.</p>
<hr />
</div>
<div id="heatmap" class="section level1">
<h1>Heatmap</h1>
<p>Each column in the targets dataframe should be considered as a potential covariate in our models. Both technical and biological factors should be investigated as these may introduce batch effects or be clinically relevant. In order to assess this, we convert each column to a categorical variable whose correlations can be visualised.</p>
<p>Homogenous columns can be excluded, as any variables that do not change by definition cannot explain variation in the data. In this instance, the <code>contact_institute</code> for all samples is Leiden, so there is no need to investigate a city effect. However, in some instances samples may processed in different cities, meaning that this could introduce a batch effect that needed to be corrected for.</p>
<pre class="r"><code>df &lt;- apply(targets, 2, function(x) as.numeric(factor(x)))
keep &lt;- apply(df, 2, sd) &gt; 0
summary(keep)</code></pre>
<pre><code>##    Mode   FALSE    TRUE 
## logical      24      10</code></pre>
<pre class="r"><code>df &lt;- df[ , keep]
colnames(df)</code></pre>
<pre><code>##  [1] &quot;title&quot;                 &quot;geo_accession&quot;        
##  [3] &quot;source_name_ch1&quot;       &quot;characteristics_ch1&quot;  
##  [5] &quot;characteristics_ch1.1&quot; &quot;supplementary_file&quot;   
##  [7] &quot;supplementary_file.1&quot;  &quot;dataset:ch1&quot;          
##  [9] &quot;tissue:ch1&quot;            &quot;Basename&quot;</code></pre>
<p>As you can see, 10 columns need to be examined. These include sample-level variables, such as <code>title</code>, <code>geo_accession</code>, and the <code>supplementary_file</code> variables, but also other higher-level groupings. Since this dataset contained multiple tissues collected from both cadavers and healthy individuals, we need to consider the influence of these using <code>source_name_ch1</code>, <code>characteristics_ch1.1</code>, and <code>tissue:ch1</code>. There also may be batch effects from the different data sets recorded in <code>characteristics_ch1</code> and <code>dataset:ch1</code>.</p>
<p>Correlations between these factors and the principal components is calculated, and then these are visualised using the <code>pheatmap()</code> function from <a href="https://cran.r-project.org/web/packages/pheatmap/index.html"><strong>pheatmap</strong></a>.</p>
<pre class="r"><code>library(pheatmap)
cxy &lt;- cor(pc$x, scale(df))   
pheatmap(cxy, cluster_rows=FALSE, legend=TRUE, main=&quot;Heatmap of correlations&quot;)</code></pre>
<p><img src="02_Visualization_files/figure-html/306heatmapplot-1.png" width="672" /></p>
<p>Here we see grouping of similar covariates together, with three main considerations: tissue type, sample-level variations, and batches, including dataset and source. Carrying on analyses separately for each tissue type will likely be sufficient adjust for these variables.</p>
<hr />
</div>
<div id="dendrogram" class="section level1">
<h1>Dendrogram</h1>
<p>Although more frequently used for moderately-sized data sets, a dendogram can be helpful to visualise clustering. We use <code>dist()</code> to compute the distance matrix for the beta values, before implementing hierarchical cluster analysis with <code>hclust()</code> from the <a href="https://cran.r-project.org/package=fastclust"><strong>fastcluster</strong></a>-package. We then plot this clustering as a horizontal dendogram.</p>
<pre class="r"><code>library(fastcluster)
sBeta &lt;- getBeta(RGset[,1:16], type=&quot;Illumina&quot;)
d &lt;- dist(t(sBeta), method=&quot;euclidean&quot;)
fit &lt;- hclust(d, method=&quot;average&quot;)
fitdd &lt;- as.dendrogram(fit)
nodePar &lt;- list(lab.cex=0.8, pch=c(NA, 15), cex=0.8, col=&quot;#41819a&quot;)
par(mar=c(3,1,2,11))
plot(fitdd, main=&quot;Cluster Dendogram&quot;, horiz=TRUE, nodePar=nodePar, xlab=&quot;Height&quot;)</code></pre>
<p><img src="02_Visualization_files/figure-html/307dendronorm-1.png" width="672" /></p>
<p>If an observation needs further investigation, we can identify it in the dendogram by highlighting it.</p>
<pre class="r"><code>interestObs &lt;- sample(colnames(sBeta), 1)
colorLeafs &lt;- function(x) {
  if (is.leaf(x) &amp;&amp; attr(x, &quot;label&quot;) %in% interestObs) {
    attr(x, &quot;nodePar&quot;) &lt;- list(lab.cex=0.8, lab.col=&quot;red&quot;, pch=c(NA,15))
  }
  return(x)
} 
dd &lt;- dendrapply(as.dendrogram(fit), colorLeafs)
par(mar=c(3,1,2,11))
plot(dd, main=&quot;Cluster Dendogram&quot;, horiz=TRUE, nodePar=nodePar, xlab=&quot;Height&quot;)</code></pre>
<p><img src="02_Visualization_files/figure-html/308dendrored-1.png" width="672" /></p>
<hr />
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references">
<div id="ref-Aryee2014">
<p>Aryee, M. J., A. E. Jaffe, H. Corrada-Bravo, C. Ladd-Acosta, A. P. Feinberg, K. D. Hansen, and R. A. Irizarry. 2014. “Minfi: a flexible and comprehensive Bioconductor package for the analysis of Infinium DNA methylation microarrays.” <em>Bioinformatics</em>.</p>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
