<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>03_ProbeQC.knit</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DNAmArray</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="01_GettingStarted.html">Getting Started</a>
</li>
<li>
  <a href="02_SampleQualityControl.html">Sample-level QC</a>
</li>
<li>
  <a href="03_Normalization.html">Normalization</a>
</li>
<li>
  <a href="04_ProbeFiltering.html">Probe-level QC</a>
</li>
<li>
  <a href="05_Predict.html">Predictors</a>
</li>
<li>
  <a href="06_Masking.html">Underperforming Probes</a>
</li>
<li>
  <a href="07_EWAS.html">Analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">About</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">




</div>


<hr />
<div id="annotation" class="section level1">
<h1>Annotation</h1>
<p>We load the EPIC annotation (hg19), which can be downloaded from <a
href="https://zwdzwd.github.io/InfiniumAnnotation"
class="uri">https://zwdzwd.github.io/InfiniumAnnotation</a></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>anno <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;./data/EPIC.hg19.manifest.tsv.gz&quot;</span>)</span></code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   .default = col_double(),
##   CpG_chrm = col_character(),
##   probe_strand = col_character(),
##   probeID = col_character(),
##   channel = col_character(),
##   designType = col_character(),
##   nextBase = col_character(),
##   nextBaseRef = col_character(),
##   probeType = col_character(),
##   orientation = col_character(),
##   ProbeSeq_A = col_character(),
##   ProbeSeq_B = col_character(),
##   gene = col_character(),
##   gene_HGNC = col_character(),
##   chrm_A = col_character(),
##   cigar_A = col_character(),
##   chrm_B = col_character(),
##   cigar_B = col_character(),
##   wDecoy_chrm_A = col_character(),
##   wDecoy_cigar_A = col_character(),
##   wDecoy_chrm_B = col_character()
##   # ... with 13 more columns
## )</code></pre>
<pre><code>## See spec(...) for full column specifications.</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>anno <span class="ot">&lt;-</span> anno <span class="sc">%&gt;%</span> </span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>    <span class="at">cpg =</span> probeID,</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>    <span class="at">chr =</span> CpG_chrm,</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>    <span class="at">start =</span> CpG_beg,</span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>    <span class="at">end =</span> CpG_end,</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>    <span class="at">strand =</span> probe_strand,</span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>    gene_HGNC,</span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a>    MASK_general,</span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>    probeType</span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a>    <span class="at">chr =</span> <span class="fu">substr</span>(chr,<span class="dv">4</span>,<span class="dv">5</span>)</span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a>  )</span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a></span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a><span class="fu">str</span>(anno)</span></code></pre></div>
<pre><code>## tibble [865,918 × 8] (S3: tbl_df/tbl/data.frame)
##  $ cpg         : chr [1:865918] &quot;cg14817997&quot; &quot;cg26928153&quot; &quot;cg16269199&quot; &quot;cg13869341&quot; ...
##  $ chr         : chr [1:865918] &quot;1&quot; &quot;1&quot; &quot;1&quot; &quot;1&quot; ...
##  $ start       : num [1:865918] 10524 10847 10849 15864 18826 ...
##  $ end         : num [1:865918] 10526 10849 10851 15866 18828 ...
##  $ strand      : chr [1:865918] &quot;-&quot; &quot;+&quot; &quot;+&quot; &quot;-&quot; ...
##  $ gene_HGNC   : chr [1:865918] NA &quot;DDX11L1&quot; &quot;DDX11L1&quot; &quot;WASH7P&quot; ...
##  $ MASK_general: logi [1:865918] TRUE TRUE TRUE TRUE TRUE TRUE ...
##  $ probeType   : chr [1:865918] &quot;cg&quot; &quot;cg&quot; &quot;cg&quot; &quot;cg&quot; ...</code></pre>
<hr />
</div>
<div id="filtering-on-intensities" class="section level1">
<h1>Filtering on Intensities</h1>
<p>We recommend performing probe-filtering by setting unreliably
measured probes to NA. Unfortunately, Functional Normalization does not
allow NA’s in the <code>RGChannelSet</code>. Therefore, it is best to
run normalization and probe-filtering separately and combine the results
afterwards using our <code>reduce()</code> function.</p>
<p>Some observations need to be removed from the dataset since they
provide little to no information that could skew analyses. Probes which
have zero intensity or are based on very few (default: &lt;3) beads are
removed, based on a specified cut off.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>RGset <span class="ot">&lt;-</span> <span class="fu">probeFiltering</span>(RGset)</span></code></pre></div>
<pre><code>## Filtering on number of beads... 
## On average 0.12 % of the probes ( 1052641 ) have number of beads below 3 
## Filtering on zero intensities... 
## On average 0.017 % of the Type II probes ( 724574 ) have Red and/or Green intensity below 1 
## On average 0.015 % of the Type I probes ( 99964 ), measured in Green channel, have intensity below 1 
## On average 0.015 % of the Type I probes ( 184560 ), measured in Red channel, have intensity below 1 
## Set filtered probes in Red and/or Green channels to NA...</code></pre>
<p>There are differences between probe types that are used in
<code>probeFiltering()</code>. Information on these can be extracted and
stored using functions from <a
href="https://bioconductor.org/packages/release/bioc/html/minfi.html"><strong>minfi</strong></a>
(Feinberg, 2014).</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>locusNames <span class="ot">&lt;-</span> <span class="fu">getManifestInfo</span>(RGset, <span class="st">&quot;locusNames&quot;</span>)</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>typeII <span class="ot">&lt;-</span> <span class="fu">getProbeInfo</span>(RGset, <span class="at">type=</span><span class="st">&quot;II&quot;</span>)</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>typeI <span class="ot">&lt;-</span> <span class="fu">getProbeInfo</span>(RGset, <span class="at">type=</span><span class="st">&quot;I&quot;</span>)</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>RGset</span></code></pre></div>
<pre><code>## class: RGChannelSet 
## dim: 1052641 61 
## metadata(0):
## assays(2): Green Red
## rownames(1052641): 1600101 1600111 ... 99810990 99810992
## rowData names(0):
## colnames(61): GSM3228563_200550980002_R02C01 GSM3228564_200550980002_R03C01 ... GSM3228624_200590490052_R03C01
##   GSM3228625_200590490052_R04C01
## colData names(15): sample_ID geo_accession ... filenames beta_outlier
## Annotation
##   array: IlluminaHumanMethylationEPIC
##   annotation: 20a1.hg38</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>betas <span class="ot">&lt;-</span> <span class="fu">getBeta</span>(RGset, <span class="at">type=</span><span class="st">&quot;Illumina&quot;</span>)</span></code></pre></div>
<hr />
</div>
<div id="encode-blacklist" class="section level1">
<h1>ENCODE Blacklist</h1>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;./data/ENCODEBlacklist_CpGomit-EPIC.RData&quot;</span>)</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>betas <span class="ot">&lt;-</span> betas[<span class="sc">!</span><span class="fu">rownames</span>(betas) <span class="sc">%in%</span> cpg_blacklist,]</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a><span class="fu">dim</span>(betas)</span></code></pre></div>
<pre><code>## [1] 857749     61</code></pre>
<hr />
</div>
<div id="zhou-probes" class="section level1">
<h1>Zhou probes</h1>
<p>Several studies (Zhou, 2017) have characterized cross-reactive and
polymorphic probes on both the 450k and EPIC methylation arrays and made
suggestions for removal. We chose to remove probes identified by Zhou et
al., since this set is actively maintained and supports both array
sizes. Additionally, different sets are specified depending on the
reference genome used.</p>
<p>In total, Zhou et al. suggest removal of probes for a variety of
reasons, from ambiguous mapping and cross-reactivity to polymorphisms
that interfere with extension. In general, around 60,000 probes are
suggested for removal in 450k arrays, and 100,000 for EPIC.</p>
<p>We developed the <code>probeMasking()</code> function, which removes
specified probes from either beta- or M- value matrices. In this
instance, 59,780 CpG rows are removed.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>maskProbes <span class="ot">&lt;-</span> anno[anno<span class="sc">$</span>MASK_general <span class="sc">==</span> <span class="cn">TRUE</span>,]<span class="sc">$</span>cpg</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a><span class="fu">length</span>(maskProbes)</span></code></pre></div>
<pre><code>## [1] 99360</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>betas <span class="ot">&lt;-</span> betas[<span class="sc">!</span><span class="fu">rownames</span>(betas) <span class="sc">%in%</span> maskProbes,]</span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a><span class="fu">dim</span>(betas)</span></code></pre></div>
<pre><code>## [1] 763466     61</code></pre>
<hr />
</div>
<div id="sex-chromosomes" class="section level1">
<h1>Sex chromosomes</h1>
<p>For most EWAS we filter out sex chromosomes.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>xy_cpgs <span class="ot">&lt;-</span> (anno <span class="sc">%&gt;%</span> </span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a>              dplyr<span class="sc">::</span><span class="fu">filter</span>(chr <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;X&quot;</span>, <span class="st">&quot;Y&quot;</span>)))<span class="sc">$</span>cpg</span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a>betas <span class="ot">&lt;-</span> betas[<span class="sc">!</span><span class="fu">rownames</span>(betas) <span class="sc">%in%</span> xy_cpgs,]</span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a><span class="fu">dim</span>(betas)</span></code></pre></div>
<pre><code>## [1] 746345     61</code></pre>
<hr />
</div>
<div id="outlier-removal" class="section level1">
<h1>Outlier removal</h1>
<p>Any DNAm values more than 3 IQR from the nearest quartile are
removed.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="fu">xtabs</span>(<span class="sc">~</span><span class="fu">is.na</span>(betas))</span></code></pre></div>
<pre><code>## is.na(betas)
##    FALSE     TRUE 
## 45470509    56536</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>iqr_dnam <span class="ot">&lt;-</span> <span class="fu">apply</span>(betas, <span class="dv">1</span>, <span class="cf">function</span>(x){</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>  iqr <span class="ot">&lt;-</span> <span class="fu">IQR</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a>  q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(x, <span class="at">na.rm=</span><span class="cn">TRUE</span>)[[<span class="dv">2</span>]]</span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a>  q3 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(x, <span class="at">na.rm=</span><span class="cn">TRUE</span>)[[<span class="dv">4</span>]]</span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a>  x <span class="ot">&lt;-</span> <span class="fu">ifelse</span>((x <span class="sc">&lt;=</span> q1 <span class="sc">-</span> (<span class="dv">3</span><span class="sc">*</span>iqr) <span class="sc">|</span> x <span class="sc">&gt;=</span> q3 <span class="sc">+</span> (<span class="dv">3</span><span class="sc">*</span>iqr)), <span class="cn">NA</span>, x)</span>
<span id="cb21-6"><a href="#cb21-6" tabindex="-1"></a>})</span>
<span id="cb21-7"><a href="#cb21-7" tabindex="-1"></a></span>
<span id="cb21-8"><a href="#cb21-8" tabindex="-1"></a><span class="fu">xtabs</span>(<span class="sc">~</span><span class="fu">is.na</span>(iqr_dnam))</span></code></pre></div>
<pre><code>## is.na(iqr_dnam)
##    FALSE     TRUE 
## 45415081   111964</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="fu">dim</span>(iqr_dnam)</span></code></pre></div>
<pre><code>## [1]     61 746345</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>betas <span class="ot">&lt;-</span> <span class="fu">t</span>(iqr_dnam)</span></code></pre></div>
<hr />
</div>
<div id="missingness" class="section level1">
<h1>Missingness</h1>
<p>Keep only the CpGs where less than 5% of sample values are
missing</p>
<p>This is all CpGs, representing high quality data</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>perc_na <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">is.na</span>(iqr_dnam))<span class="sc">*</span><span class="dv">100</span><span class="sc">/</span><span class="fu">ncol</span>(iqr_dnam)</span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a><span class="fu">summary</span>(perc_na)</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##  0.1018  0.1377  0.1558  0.2459  0.2191  1.4333</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a>betas <span class="ot">&lt;-</span> betas[,perc_na <span class="sc">&lt;=</span> <span class="dv">95</span>]</span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a><span class="fu">dim</span>(betas)</span></code></pre></div>
<pre><code>## [1] 746345     61</code></pre>
<p>Keep only the samples where less than 5% of CpGs are missing</p>
<p>This is all samples, representing high quality data</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a>perc_na <span class="ot">&lt;-</span> <span class="fu">colSums</span>(<span class="fu">is.na</span>(iqr_dnam))<span class="sc">*</span><span class="dv">100</span><span class="sc">/</span><span class="fu">nrow</span>(iqr_dnam)</span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a><span class="fu">summary</span>(perc_na)</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##  0.0000  0.0000  0.0000  0.2459  0.0000 27.8689</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a>betas <span class="ot">&lt;-</span> betas[perc_na <span class="sc">&lt;=</span> <span class="dv">95</span>,]</span>
<span id="cb32-2"><a href="#cb32-2" tabindex="-1"></a><span class="fu">dim</span>(betas)</span></code></pre></div>
<pre><code>## [1] 746345     61</code></pre>
<hr />
</div>
<div id="summarizedexperiment" class="section level1">
<h1>SummarizedExperiment</h1>
<p>Now that we have our annotations, we can combine them with our data
using the <a
href="https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html"><strong>SummarizedExperiment</strong></a>
package. After ensuring that any column subsetting is accounted for, we
extract the <code>colData()</code> from our <code>RGset</code>. Finally,
the beta values, probe annotations, and sample information are combined
into a <code>RangedSummarizedExperiment</code> object using the
<code>SummarizedExperiment()</code> function.</p>
<p>Annotation data</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a>anno <span class="ot">&lt;-</span> anno <span class="sc">%&gt;%</span> </span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(cpg <span class="sc">%in%</span> <span class="fu">rownames</span>(betas))</span>
<span id="cb34-3"><a href="#cb34-3" tabindex="-1"></a></span>
<span id="cb34-4"><a href="#cb34-4" tabindex="-1"></a><span class="fu">dim</span>(anno)</span></code></pre></div>
<pre><code>## [1] 745368      8</code></pre>
<p>Keep annotated probes</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a>betas <span class="ot">&lt;-</span> betas[anno<span class="sc">$</span>cpg, ]</span></code></pre></div>
<p>Sample information</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a>targets <span class="ot">&lt;-</span> targets[<span class="fu">colnames</span>(betas), ]</span>
<span id="cb37-2"><a href="#cb37-2" tabindex="-1"></a></span>
<span id="cb37-3"><a href="#cb37-3" tabindex="-1"></a><span class="fu">dim</span>(targets)</span></code></pre></div>
<pre><code>## [1] 61 13</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a><span class="fu">dim</span>(betas)</span></code></pre></div>
<pre><code>## [1] 745368     61</code></pre>
<p>Make summarized experiment object</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a>methData <span class="ot">&lt;-</span> <span class="fu">SummarizedExperiment</span>(</span>
<span id="cb41-2"><a href="#cb41-2" tabindex="-1"></a>  <span class="at">assays=</span><span class="fu">SimpleList</span>(<span class="at">beta=</span>betas), </span>
<span id="cb41-3"><a href="#cb41-3" tabindex="-1"></a>  <span class="at">rowData=</span>anno, </span>
<span id="cb41-4"><a href="#cb41-4" tabindex="-1"></a>  <span class="at">colData=</span>targets)</span>
<span id="cb41-5"><a href="#cb41-5" tabindex="-1"></a></span>
<span id="cb41-6"><a href="#cb41-6" tabindex="-1"></a>methData</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 745368 61 
## metadata(0):
## assays(1): beta
## rownames(745368): cg09499020 cg16535257 ... cg08423507 cg24129471
## rowData names(8): cpg chr ... MASK_general probeType
## colnames(61): GSM3228563_200550980002_R02C01 GSM3228564_200550980002_R03C01 ... GSM3228624_200590490052_R03C01
##   GSM3228625_200590490052_R04C01
## colData names(13): sample_ID geo_accession ... row Basename</code></pre>
You can read more about <a
href="http://bioconductor.org/packages/SummarizedExperiment"><strong>SummarizedExperiment</strong></a>
from the following resources:
<ul>
<li>
<a
href="http://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html">Package
vignette</a>
</li>
<li>
<a
href="http://bioconductor.org/help/course-materials/2012/BiocEurope2012/SummarizedExperiment.pdf">Course
material</a>
</li>
<li>
<a
href="http://www.nature.com/nmeth/journal/v12/n2/full/nmeth.3252.html">BioConductor
nature paper</a>
</li>
</ul>
<hr />
</div>
<div id="save" class="section level1">
<h1>Save</h1>
<p>Save RGset</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a>RGset <span class="ot">&lt;-</span> RGset[, <span class="fu">colnames</span>(RGset) <span class="sc">%in%</span> <span class="fu">colnames</span>(methData)]</span>
<span id="cb43-2"><a href="#cb43-2" tabindex="-1"></a>RGset</span></code></pre></div>
<pre><code>## class: RGChannelSet 
## dim: 1052641 61 
## metadata(0):
## assays(2): Green Red
## rownames(1052641): 1600101 1600111 ... 99810990 99810992
## rowData names(0):
## colnames(61): GSM3228563_200550980002_R02C01 GSM3228564_200550980002_R03C01 ... GSM3228624_200590490052_R03C01
##   GSM3228625_200590490052_R04C01
## colData names(15): sample_ID geo_accession ... filenames beta_outlier
## Annotation
##   array: IlluminaHumanMethylationEPIC
##   annotation: 20a1.hg38</code></pre>
<p>Save clean data</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" tabindex="-1"></a><span class="fu">save</span>(targets, <span class="at">file =</span> <span class="st">&quot;./data/targets.Rdata&quot;</span>)</span>
<span id="cb45-2"><a href="#cb45-2" tabindex="-1"></a><span class="fu">save</span>(methData, <span class="at">file=</span><span class="st">&quot;./data/methData.Rdata&quot;</span>)</span>
<span id="cb45-3"><a href="#cb45-3" tabindex="-1"></a><span class="fu">save</span>(RGset, <span class="at">file=</span><span class="st">&quot;./data/RGset.Rdata&quot;</span>)</span></code></pre></div>
<hr />
</div>
<div id="saving" class="section level1">
<h1>Saving</h1>
<p>We want to save our data as a <code>SummarizedExperiment</code>
object for the rest of the pipeline.</p>
<hr />
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
