<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<<<<<<< HEAD
<<<<<<<< HEAD:docs/03_Predict.html
<title>03_Predict.knit</title>

========
<title>03_Normalization.knit</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
>>>>>>>> a881b47ac32537db4f8636228d059d1464fdc2ea:docs/03_Normalization.html
=======
<title>03_Predict.knit</title>

>>>>>>> a881b47ac32537db4f8636228d059d1464fdc2ea
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
<<<<<<< HEAD
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
=======
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
>>>>>>> a881b47ac32537db4f8636228d059d1464fdc2ea
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
<<<<<<< HEAD
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
=======
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
>>>>>>> a881b47ac32537db4f8636228d059d1464fdc2ea
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
<<<<<<< HEAD
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
=======
a.sourceLine::before { text-decoration: underline; }
}
code span. { color: #1f1c1b; } /* Normal */
>>>>>>> a881b47ac32537db4f8636228d059d1464fdc2ea
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DNAmArray</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="01_GettingStarted.html">Getting Started</a>
</li>
<li>
  <a href="02_SampleQualityControl.html">Sample QC</a>
</li>
<li>
<<<<<<< HEAD
<<<<<<<< HEAD:docs/03_Predict.html
=======
>>>>>>> a881b47ac32537db4f8636228d059d1464fdc2ea
  <a href="03_Predict.html">Predict</a>
</li>
<li>
  <a href="04_Normalization.html">Normalize</a>
<<<<<<< HEAD
========
  <a href="03_Normalization.html">Normalize</a>
</li>
<li>
  <a href="04_Predict.html">Predict</a>
>>>>>>>> a881b47ac32537db4f8636228d059d1464fdc2ea:docs/03_Normalization.html
=======
>>>>>>> a881b47ac32537db4f8636228d059d1464fdc2ea
</li>
<li>
  <a href="05_ProbeQC.html">Probe QC</a>
</li>
<li>
  <a href="06_EWAS.html">Analyse</a>
</li>
<li>
  <a href="07_Interpret.html">Interpret</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">About</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">




</div>


<hr />
<<<<<<< HEAD
<<<<<<<< HEAD:docs/03_Predict.html
=======
>>>>>>> a881b47ac32537db4f8636228d059d1464fdc2ea
<div id="phenotype-prediction" class="section level1">
<h1>Phenotype Prediction</h1>
<div id="motivation" class="section level2">
<h2>Motivation</h2>
<p>Numerous phenotypic traits can be accurately inferred from DNA methylation (DNAm) data, including immune cell composition<sup>29</sup>, sex<sup>31</sup>, smoking status, and both chronological and biological age through the application of epigenetic clocks. These DNAm-derived predictions can be valuable for enriching or completing sample metadata, especially when certain variables were not directly measured during sample collection. Moreover, they serve as an important quality control tool by helping to identify potential sample mix-ups or data inconsistencies—for instance, mismatches between predicted and recorded sex may indicate errors in sample labeling or processing.</p>
<p>Here we outline methods to predict immune cell composition and sex from DNAm data as part of the DNAmArray pipeline.</p>
<hr />
</div>
<div id="cell-counts" class="section level2">
<h2>Cell counts</h2>
<p>The <code>EPIDish</code> package can be used to predict blood cell types. It is a R package that infers the proportions of a priori known cell-types present in a sample representing a mixture of such cell-types. Right now, the package can be used on DNAm data of blood-tissue of any age, from birth to old-age, generic epithelial tissue and breast tissue. The package also provides a function that allows the identification of differentially methylated cell-types and their directionality of change in Epigenome-Wide Association Studies.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">data</span>(cent12CT.m)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"></a>
<a class="sourceLine" id="cb1-3" data-line-number="3">BloodFrac.m &lt;-<span class="st"> </span><span class="kw">epidish</span>(<span class="dt">beta.m =</span> betas, </a>
<a class="sourceLine" id="cb1-4" data-line-number="4">                       <span class="dt">ref.m =</span> cent12CT.m, </a>
<a class="sourceLine" id="cb1-5" data-line-number="5">                       <span class="dt">method =</span> <span class="st">&quot;RPC&quot;</span>)<span class="op">$</span>estF</a></code></pre></div>
<p>After proportions of cell types have been estimated, they can be plotted and inspected.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">BloodFrac.m_long &lt;-<span class="st"> </span><span class="kw">pivot_longer</span>(<span class="kw">as.data.frame</span>(BloodFrac.m), <span class="dt">cols =</span> <span class="kw">colnames</span>(BloodFrac.m))</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"></a>
<a class="sourceLine" id="cb2-3" data-line-number="3">BloodFrac.m_long <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">y=</span>name, <span class="dt">x=</span>value)) <span class="op">+</span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="st">  </span><span class="kw">geom_boxplot</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="st">  </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&#39;&#39;</span>)</a></code></pre></div>
<p><img src="03_Predict_files/figure-html/302plot-1.png" width="672" /></p>
<p>Cell counts can be added to <code>targets</code> for use later when building EWAS models.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw">table</span>(<span class="kw">rownames</span>(<span class="kw">as.data.frame</span>(BloodFrac.m)) <span class="op">==</span><span class="st"> </span><span class="kw">rownames</span>(targets))</a></code></pre></div>
<pre><code>## 
## TRUE 
<<<<<<< HEAD
##  598</code></pre>
=======
##  496</code></pre>
>>>>>>> a881b47ac32537db4f8636228d059d1464fdc2ea
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">targets &lt;-<span class="st"> </span><span class="kw">cbind</span>(targets, BloodFrac.m)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw">colData</span>(RGset) &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(targets)</a></code></pre></div>
<p>Other extensions, including UniLIFE which predicts 19 immune cell-types applicable to blood tissue of any age, are available from within EpiDISH, for use in specific contexts<sup>29</sup>.</p>
<hr />
</div>
</div>
<div id="predict-sex" class="section level1">
<h1>Predict Sex</h1>
<p>Sex can also be predicted from CpGs on the X-chromosome. Here, we outline the use of <code>estimateSex</code> from <a href="https://www.bioconductor.org/packages/devel/bioc/html/wateRmelon.html"><strong>wateRmelon</strong></a><sup>31</sup>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">estimated_sex &lt;-<span class="st"> </span><span class="kw">estimateSex</span>(betas, <span class="dt">do_plot=</span><span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## Normalize beta values by Z score...</code></pre>
<pre><code>## Fishished Zscore normalization.</code></pre>
<p><img src="03_Predict_files/figure-html/304anno-1.png" width="672" /></p>
<p>Then a measure can be calculated, determining the sex of each sample and tabulated against recorded sex.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">table</span>(estimated_sex<span class="op">$</span>predicted_sex, targets<span class="op">$</span>sex)</a></code></pre></div>
<pre><code>##         
##          Female Male
##   47,XXY      0    1
<<<<<<< HEAD
##   Female    337    1
##   Male        2  257</code></pre>
<p>As you can see, there is an outlier in the data, which we can remove.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">targets &lt;-<span class="st"> </span>targets <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="st">  </span><span class="kw">filter</span>(Basename <span class="op">!=</span><span class="st"> &quot;GSM3228809_200594740080_R01C01&quot;</span>)</a></code></pre></div>
<p>This means that we can feel increased confidence that no incorrect labelling or mix-ups are remaining.</p>
========
<div id="functional-normalization" class="section level1">
<h1>Functional normalization</h1>
<div id="motivation" class="section level2">
<h2>Motivation</h2>
<p>Our workflow uses the functional normalization approach<sup>16</sup>,
which exploits internal control probes designed to detect technical
variations without assaying biological differences.</p>
<p>It has been shown to perform favourably when compared to other
approaches<sup>17</sup>. Using the internal control probes avoids the
problems associated with global normalization methods, where biological
variation can be mistaken for a technical effect and removed. This is
especially important in studies where groups are expected to have
differential methylation signatures, such as multiple tissue
studies<sup>18</sup>.</p>
<p>Conversations on the best approaches for normalization in DNAm data
pipelines are ongoing<sup>19</sup>.</p>
<hr />
</div>
</div>
<div id="principal-components" class="section level1">
<h1>Principal Components</h1>
<p>The default of selecting only two principal components is often too
low for this type of data. Often you will see a drop-off in proportion
of variance explained after a certain number of principal components,
and this can indicate an efficient selection.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>var_explained <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>PC, <span class="at">y=</span>var_explained)) <span class="sc">+</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color=</span><span class="st">&#39;grey5&#39;</span>, <span class="at">fill=</span><span class="st">&#39;#6DACBC&#39;</span>, <span class="at">shape=</span><span class="dv">21</span>, <span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span> </span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks=</span><span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(pca<span class="sc">$</span>x)) <span class="sc">+</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Principal Component&quot;</span>) <span class="sc">+</span> </span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Proportion of variance explained&quot;</span>) <span class="sc">+</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>()</span></code></pre></div>
<p><img src="03_Normalization_files/figure-html/301scree-1.png" width="672" /></p>
<hr />
</div>
<div id="running-normalization" class="section level1">
<h1>Running Normalization</h1>
<p>In order to run normalization the annotation of the
<code>RGset</code> must be updated for EPIC arrays.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>RGset<span class="sc">@</span>annotation <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">array =</span> <span class="st">&quot;IlluminaHumanMethylationEPIC&quot;</span>, <span class="at">annotation =</span> <span class="st">&quot;ilm10b4.hg19&quot;</span>)</span></code></pre></div>
<p>By default, functional normalization returns normalized copy number
data making the returned <code>GenomicRatioSet</code> twice the size
necessary when only beta-values or M-values are required. Therefore, we
developed <code>preprocessFunnorm.DNAmArray()</code>, which adds an
option not to return these to the <code>preprocessFunnorm()</code>
function from <a
href="https://bioconductor.org/packages/release/bioc/html/minfi.html"><strong>minfi</strong></a>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>RGset <span class="ot">&lt;-</span> <span class="fu">preprocessFunnorm.DNAmArray</span>(</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>  <span class="at">rgSet =</span> RGset,</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>  <span class="at">nPCs =</span> <span class="dv">4</span>,</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>  <span class="at">keepCN =</span> <span class="cn">FALSE</span>,</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">TRUE</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>)</span></code></pre></div>
<p>You can now extract the normalized beta values from this object. It
can also be advisable to round the beta values to 4 significant figures,
which both reduces the data size and provides a realistic measure of the
available precision.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>betas <span class="ot">&lt;-</span> <span class="fu">getBeta</span>(RGset, <span class="at">type=</span><span class="st">&quot;Illumina&quot;</span>)</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>betas <span class="ot">&lt;-</span> <span class="fu">round</span>(betas, <span class="dv">4</span>)</span></code></pre></div>
<p>If required, these can be transformed to M-values using <a
href="https://bioconductor.org/packages/release/bioc/html/minfi.html"><strong>minfi</strong></a>’s
<code>getM()</code> which applies a logit2-transformation to the
beta-values.</p>
>>>>>>>> a881b47ac32537db4f8636228d059d1464fdc2ea:docs/03_Normalization.html
=======
##   Female    270    0
##   Male        2  223</code></pre>
<p>As you can see, this is complete data, but the predicted and assumed sexes are identical. This means that we can feel increased confidence that no incorrect labelling or mix-ups are present.</p>
>>>>>>> a881b47ac32537db4f8636228d059d1464fdc2ea
<hr />
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
