<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>06_EWAS.knit</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DNAmArray</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="01_GettingStarted.html">Getting Started</a>
</li>
<li>
  <a href="02_SampleQualityControl.html">Sample QC</a>
</li>
<li>
  <a href="03_Normalization.html">Normalize</a>
</li>
<li>
  <a href="04_Predict.html">Predict</a>
</li>
<li>
  <a href="05_ProbeQC.html">Probe QC</a>
</li>
<li>
  <a href="06_EWAS.html">Analyse</a>
</li>
<li>
  <a href="07_Interpret.html">Interpret</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">About</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">




</div>


<hr />
<div id="covariates" class="section level1">
<h1>Covariates</h1>
<p>Each column in the <code>RGset</code> colData should be considered as
a potential covariate in EWAS models. Both technical and biological
factors should be investigated as these may introduce batch effects or
be clinically relevant. In order to assess this, you can visualize
correlations with PCs as done previously in this workflow.</p>
<p>Any constant variables are removed from the heatmap, as they will not
explain variance in the data.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>plot_vars <span class="ot">&lt;-</span> <span class="fu">apply</span>(targets, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">sd</span>(<span class="fu">as.numeric</span>(<span class="fu">factor</span>(x)), <span class="at">na.rm=</span>T))</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>plot_vars <span class="ot">&lt;-</span> <span class="fu">names</span>(plot_vars[<span class="sc">!</span>plot_vars <span class="sc">%in%</span> <span class="fu">c</span>(<span class="cn">NA</span>, <span class="dv">0</span>)])</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>plot_vars</span></code></pre></div>
<pre><code>##  [1] &quot;sample_ID&quot;          &quot;geo_accession&quot;      &quot;sex&quot;                &quot;age&quot;                &quot;log_total_pbb&quot;      &quot;pbb_153&quot;           
##  [7] &quot;pbb_77&quot;             &quot;pbb_101&quot;            &quot;pbb_180&quot;            &quot;supplementary_file&quot; &quot;plate&quot;              &quot;row&quot;               
## [13] &quot;Basename&quot;           &quot;CD4Tnv&quot;             &quot;Baso&quot;               &quot;CD4Tmem&quot;            &quot;Bmem&quot;               &quot;Bnv&quot;               
## [19] &quot;Treg&quot;               &quot;CD8Tmem&quot;            &quot;CD8Tnv&quot;             &quot;Eos&quot;                &quot;NK&quot;                 &quot;Neu&quot;               
## [25] &quot;Mono&quot;</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>heatmap_df <span class="ot">&lt;-</span> targets <span class="sc">%&gt;%</span> </span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">any_of</span>(plot_vars))</span></code></pre></div>
<p>All variables are then converted to numeric and correlations between
them and the PCs are calculated.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>heatmap_df <span class="ot">&lt;-</span> <span class="fu">apply</span>(heatmap_df, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">as.numeric</span>(<span class="fu">factor</span>(x)))</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>cxy <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">cor</span>(pca<span class="sc">$</span>x, <span class="fu">scale</span>(heatmap_df), <span class="at">use=</span><span class="st">&quot;pairwise.complete.obs&quot;</span>),<span class="dv">2</span>) </span></code></pre></div>
<p>A heatmap can then be used to visualize these correlations and
uncover measured variables that explain a large proportion of DNAm
variance.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>col_fun <span class="ot">&lt;-</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">1</span>), <span class="fu">c</span>(<span class="st">&quot;#000042&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;#800000&quot;</span>))</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="fu">Heatmap</span>(</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>  <span class="fu">t</span>(cxy),                              </span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>  <span class="at">col =</span> col_fun,  </span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>  <span class="at">border =</span> <span class="st">&#39;grey5&#39;</span>,</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>  <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,            </span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>  <span class="at">show_row_dend =</span> <span class="cn">TRUE</span>,             </span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>  <span class="at">show_column_dend =</span> <span class="cn">FALSE</span>,      </span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&quot;Value&quot;</span>,                 </span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>  <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">8</span>), </span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a>  <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">8</span>), </span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a>  <span class="at">cell_fun =</span> <span class="cf">function</span>(j, i, x, y, width, height, fill) {</span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a>    <span class="fu">grid.rect</span>(x, y, width, height, </span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a>              <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">col =</span> <span class="st">&quot;white&quot;</span>, <span class="at">lwd =</span> <span class="dv">1</span>, <span class="at">fill =</span> <span class="cn">NA</span>))</span>
<span id="cb5-16"><a href="#cb5-16" tabindex="-1"></a>    <span class="fu">grid.text</span>(<span class="fu">ifelse</span>(<span class="fu">abs</span>(<span class="fu">t</span>(cxy)[i,j]) <span class="sc">&gt;</span> <span class="fl">0.4</span>,</span>
<span id="cb5-17"><a href="#cb5-17" tabindex="-1"></a>                     <span class="fu">sprintf</span>(<span class="st">&quot;%.2f&quot;</span>, <span class="fu">round</span>(<span class="fu">t</span>(cxy)[i, j], <span class="dv">2</span>)),</span>
<span id="cb5-18"><a href="#cb5-18" tabindex="-1"></a>                     <span class="st">&quot;&quot;</span>), </span>
<span id="cb5-19"><a href="#cb5-19" tabindex="-1"></a>              x, y, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">8</span>, <span class="at">col =</span> <span class="st">&quot;white&quot;</span>))</span>
<span id="cb5-20"><a href="#cb5-20" tabindex="-1"></a>  }</span>
<span id="cb5-21"><a href="#cb5-21" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="06_EWAS_files/figure-html/603heatmap-1.png" width="672" /></p>
<p>By examining the correlations in the data, we can build our model in
a more informed manner. The second PC is highly correlated with sex and,
as is usual in EWAS, we intend to include this as a confounder.</p>
<p>Some predicted cell counts appear of substantial importance and we
advise also adjusting for these. Model specification should be informed
by a combination of prior knowledge and inspection of patterns in the
data.</p>
<hr />
</div>
<div id="sva" class="section level1">
<h1>SVA</h1>
<div id="motivation" class="section level2">
<h2>Motivation</h2>
<p>The <code>sva</code> package contains functions for removing batch
effects and other unwanted variation in high-throughput experiments.
Specifically, the sva package contains functions for the identifying and
building surrogate variables for high-dimensional data sets. Surrogate
variables are covariates constructed directly from high-dimensional data
(like DNAm data) that can be used in subsequent analyses to adjust for
unknown, unmodeled, or latent sources of noise.</p>
<p>The sva package can be used to remove artifacts in three ways:</p>
<ul>
<li>identifying and estimating surrogate variables for unknown sources
of variation in high-throughput experiments<sup>23</sup></li>
<li>directly removing known batch effects using ComBat<sup>24</sup>,
and</li>
<li>removing batch effects with known control probes<sup>25</sup></li>
</ul>
<p>Removing batch effects and using surrogate variables in differential
expression analysis have been shown to reduce dependence, stabilize
error rate estimates, and improve reproducibility<sup>23,26</sup></p>
<hr />
</div>
<div id="specification" class="section level2">
<h2>Specification</h2>
<p>The full model and null model are specified with and without the
variable of interest respectively. In this case, we adjust for
<code>sex</code>, <code>age</code>, <code>plate</code>,
<code>row</code>, <code>CD4Tnv</code>, <code>Baso</code>,
<code>CD4Tmem</code>, <code>Bmem</code>, <code>Bnv</code>,
<code>Treg</code>, <code>CD8Tmem</code>, <code>CD8Tnv</code>,
<code>Eos</code>, <code>NK</code>, and <code>Mono</code>. Neutrophils
are excluded to avoid collinearity as all predicted cell counts sum to
1.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>mod <span class="ot">=</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>log_total_pbb <span class="sc">+</span> age <span class="sc">+</span> sex <span class="sc">+</span> plate <span class="sc">+</span> row <span class="sc">+</span> CD4Tnv <span class="sc">+</span> Baso <span class="sc">+</span> CD4Tmem <span class="sc">+</span> Bmem <span class="sc">+</span> Bnv <span class="sc">+</span> Treg <span class="sc">+</span> CD8Tmem <span class="sc">+</span> CD8Tnv <span class="sc">+</span> Eos <span class="sc">+</span> NK <span class="sc">+</span> Mono, <span class="at">data=</span>targets)</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>mod0 <span class="ot">=</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>age <span class="sc">+</span> sex <span class="sc">+</span> plate <span class="sc">+</span> row <span class="sc">+</span> CD4Tnv <span class="sc">+</span> Baso <span class="sc">+</span> CD4Tmem <span class="sc">+</span> Bmem <span class="sc">+</span> Bnv <span class="sc">+</span> Treg <span class="sc">+</span> CD8Tmem <span class="sc">+</span> CD8Tnv <span class="sc">+</span> Eos <span class="sc">+</span> NK <span class="sc">+</span> Mono,<span class="at">data=</span>targets)</span></code></pre></div>
<hr />
</div>
<div id="imputation" class="section level2">
<h2>Imputation</h2>
<p>Since SVA can only be run on complete data, we use <a
href="https://bioconductor.org/packages/release/bioc/html/methyLImp2.html"><strong>methyLImp2</strong></a><sup>20</sup>
to create a complete matrix of beta values. The main function
<code>methyLImp2</code> can take a SummarizedExperiment or beta matrix
as input, and requires the array <code>type</code> to be specified (as
450k or EPIC). By default it will overwrite your SummarizedExperiment
object.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>methData_imputed <span class="ot">&lt;-</span> <span class="fu">methyLImp2</span>(</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>  <span class="at">input =</span> methData, </span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>  <span class="at">type =</span> <span class="st">&quot;EPIC&quot;</span>, </span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>  <span class="at">BPPARAM =</span> <span class="fu">SnowParam</span>(<span class="at">exportglobals =</span> <span class="cn">FALSE</span>),</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>  <span class="at">minibatch_frac =</span> <span class="fl">0.5</span>)</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>methData_imputed</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 745368 57 
## metadata(0):
## assays(1): beta
## rownames(745368): cg09499020 cg16535257 ... cg08423507
##   cg24129471
## rowData names(8): cpg chr ... MASK_general probeType
## colnames(57): GSM3228563_200550980002_R02C01
##   GSM3228564_200550980002_R03C01 ...
##   GSM3228620_200590490021_R07C01
##   GSM3228621_200590490021_R08C01
## colData names(25): sample_ID geo_accession ... Neu Mono</code></pre>
<hr />
</div>
<div id="latent-factor-calculation" class="section level2">
<h2>Latent factor calculation</h2>
<p>This can then be used to estimate the appropriate number of latent
factors to calculate.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>n.sv <span class="ot">=</span> <span class="fu">num.sv</span>(<span class="fu">assay</span>(methData_imputed),</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>              mod,</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>              <span class="at">method=</span><span class="st">&quot;leek&quot;</span>)</span></code></pre></div>
<p>The latent factors can then be calculated with the <code>sva</code>
function:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>svobj <span class="ot">=</span> <span class="fu">sva</span>(<span class="fu">assay</span>(methData_imputed),</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>            mod,</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>            mod0,</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>            <span class="at">n.sv=</span><span class="dv">4</span>)</span></code></pre></div>
<pre><code>## Number of significant surrogate variables is:  4 
## Iteration (out of 5 ):1  2  3  4  5</code></pre>
<p>These can then be added to <code>targets</code> to include in the
EWAS model.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>sv_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(svobj<span class="sc">$</span>sv)</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a><span class="fu">colnames</span>(sv_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;SV1&#39;</span>, <span class="st">&#39;SV2&#39;</span>, <span class="st">&#39;SV3&#39;</span>, <span class="st">&#39;SV4&#39;</span>)</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a>targets <span class="ot">&lt;-</span> <span class="fu">cbind</span>(targets, sv_df)</span></code></pre></div>
<hr />
</div>
</div>
<div id="ewas" class="section level1">
<h1>EWAS</h1>
<div id="packages" class="section level2">
<h2>Packages</h2>
<p><a
href="https://bioconductor.org/packages/release/bioc/html/limma.html"><strong>Limma</strong></a><sup>27</sup>
is a package with excellent documentation that can be useful for smaller
samples, due to the optional empirical Bayes step. Other packages that
may be of interest for running EWAS include [<strong>gee</strong>] and
<a
href="https://cran.r-project.org/web/packages/cate/index.html"><strong>cate</strong></a>.</p>
<hr />
</div>
<div id="formula" class="section level2">
<h2>Formula</h2>
<p>In order to run an EWAS a formula needs to be specified based on the
above exploratory analyses and SVA.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>formula <span class="ot">&lt;-</span> <span class="er">~</span>log_total_pbb <span class="sc">+</span> age <span class="sc">+</span> sex <span class="sc">+</span> plate <span class="sc">+</span> row <span class="sc">+</span> CD4Tnv <span class="sc">+</span> Baso <span class="sc">+</span> CD4Tmem <span class="sc">+</span> Bmem <span class="sc">+</span> Bnv <span class="sc">+</span> Treg <span class="sc">+</span> CD8Tmem <span class="sc">+</span> CD8Tnv <span class="sc">+</span> Eos <span class="sc">+</span> NK <span class="sc">+</span> Mono <span class="sc">+</span> SV1 <span class="sc">+</span> SV2 <span class="sc">+</span> SV3 <span class="sc">+</span> SV4</span></code></pre></div>
<p>You can then use this formula to create a design matrix.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(formula, </span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>                       <span class="at">data=</span>targets)</span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a><span class="fu">head</span>(design)</span></code></pre></div>
<pre><code>##                                (Intercept) log_total_pbb   age
## GSM3228563_200550980002_R02C01           1    -2.7535707 31.46
## GSM3228564_200550980002_R03C01           1    -0.3714986 46.72
## GSM3228565_200550980002_R04C01           1     0.1151128 46.36
## GSM3228566_200550980002_R05C01           1    -1.5478726 53.56
## GSM3228567_200550980002_R06C01           1    -0.9896682 56.36
## GSM3228568_200550980002_R07C01           1    -1.7620065 59.00
##                                sexMale plate200550980010
## GSM3228563_200550980002_R02C01       0                 0
## GSM3228564_200550980002_R03C01       1                 0
## GSM3228565_200550980002_R04C01       1                 0
## GSM3228566_200550980002_R05C01       0                 0
## GSM3228567_200550980002_R06C01       0                 0
## GSM3228568_200550980002_R07C01       0                 0
##                                plate200550980025
## GSM3228563_200550980002_R02C01                 0
## GSM3228564_200550980002_R03C01                 0
## GSM3228565_200550980002_R04C01                 0
## GSM3228566_200550980002_R05C01                 0
## GSM3228567_200550980002_R06C01                 0
## GSM3228568_200550980002_R07C01                 0
##                                plate200550980034
## GSM3228563_200550980002_R02C01                 0
## GSM3228564_200550980002_R03C01                 0
## GSM3228565_200550980002_R04C01                 0
## GSM3228566_200550980002_R05C01                 0
## GSM3228567_200550980002_R06C01                 0
## GSM3228568_200550980002_R07C01                 0
##                                plate200550980035
## GSM3228563_200550980002_R02C01                 0
## GSM3228564_200550980002_R03C01                 0
## GSM3228565_200550980002_R04C01                 0
## GSM3228566_200550980002_R05C01                 0
## GSM3228567_200550980002_R06C01                 0
## GSM3228568_200550980002_R07C01                 0
##                                plate200590490002
## GSM3228563_200550980002_R02C01                 0
## GSM3228564_200550980002_R03C01                 0
## GSM3228565_200550980002_R04C01                 0
## GSM3228566_200550980002_R05C01                 0
## GSM3228567_200550980002_R06C01                 0
## GSM3228568_200550980002_R07C01                 0
##                                plate200590490017
## GSM3228563_200550980002_R02C01                 0
## GSM3228564_200550980002_R03C01                 0
## GSM3228565_200550980002_R04C01                 0
## GSM3228566_200550980002_R05C01                 0
## GSM3228567_200550980002_R06C01                 0
## GSM3228568_200550980002_R07C01                 0
##                                plate200590490021 row     CD4Tnv
## GSM3228563_200550980002_R02C01                 0   2 0.11902209
## GSM3228564_200550980002_R03C01                 0   3 0.05644615
## GSM3228565_200550980002_R04C01                 0   4 0.06291340
## GSM3228566_200550980002_R05C01                 0   5 0.05244307
## GSM3228567_200550980002_R06C01                 0   6 0.02793638
## GSM3228568_200550980002_R07C01                 0   7 0.13030900
##                                       Baso    CD4Tmem       Bmem
## GSM3228563_200550980002_R02C01 0.000000000 0.06856902 0.04338180
## GSM3228564_200550980002_R03C01 0.003842171 0.06457686 0.03217819
## GSM3228565_200550980002_R04C01 0.000000000 0.09916235 0.03492393
## GSM3228566_200550980002_R05C01 0.000000000 0.08978173 0.02768882
## GSM3228567_200550980002_R06C01 0.000000000 0.10858777 0.04420792
## GSM3228568_200550980002_R07C01 0.004389993 0.10208036 0.02889159
##                                         Bnv       Treg    CD8Tmem
## GSM3228563_200550980002_R02C01 0.000000e+00 0.02987670 0.05170485
## GSM3228564_200550980002_R03C01 3.568496e-02 0.03253447 0.02588011
## GSM3228565_200550980002_R04C01 3.422224e-02 0.02091392 0.01348294
## GSM3228566_200550980002_R05C01 6.798882e-02 0.02574876 0.02891707
## GSM3228567_200550980002_R06C01 1.356557e-02 0.02396820 0.06586532
## GSM3228568_200550980002_R07C01 3.214358e-05 0.04655381 0.01622896
##                                     CD8Tnv         Eos         NK
## GSM3228563_200550980002_R02C01 0.064097832 0.001164834 0.03424776
## GSM3228564_200550980002_R03C01 0.000000000 0.012923973 0.06050254
## GSM3228565_200550980002_R04C01 0.000000000 0.002714143 0.02687347
## GSM3228566_200550980002_R05C01 0.005514883 0.013811095 0.11331037
## GSM3228567_200550980002_R06C01 0.000000000 0.007773618 0.02981967
## GSM3228568_200550980002_R07C01 0.003487802 0.000000000 0.03244102
##                                      Mono         SV1        SV2
## GSM3228563_200550980002_R02C01 0.07986611 -0.08479861 0.11245914
## GSM3228564_200550980002_R03C01 0.17395649 -0.10011645 0.04659800
## GSM3228565_200550980002_R04C01 0.09667953 -0.08706279 0.02731426
## GSM3228566_200550980002_R05C01 0.14498329 -0.07343356 0.17980275
## GSM3228567_200550980002_R06C01 0.07181471 -0.06994447 0.04691520
## GSM3228568_200550980002_R07C01 0.06782257 -0.08167816 0.04962268
##                                         SV3         SV4
## GSM3228563_200550980002_R02C01 -0.056605299  0.11744960
## GSM3228564_200550980002_R03C01  0.131833700 -0.05640994
## GSM3228565_200550980002_R04C01  0.124087644 -0.01504261
## GSM3228566_200550980002_R05C01  0.065412848  0.03247044
## GSM3228567_200550980002_R06C01  0.004589795  0.18848793
## GSM3228568_200550980002_R07C01  0.109370701 -0.34703374</code></pre>
<hr />
</div>
<div id="random-effects" class="section level2">
<h2>Random effects</h2>
<p>Random effects can be included using
<code>duplicateCorrelation</code> and specifying the <code>block</code>
argument. This is useful for paired experiments, not used in this
example, but with example code shown here.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>dupcor <span class="ot">&lt;-</span> <span class="fu">duplicateCorrelation</span>(<span class="fu">assay</span>(methData),</span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>                               design,</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>                               <span class="at">block =</span> <span class="fu">colData</span>(methData)<span class="sc">$</span>ID)</span></code></pre></div>
<hr />
</div>
<div id="fitting-models" class="section level2">
<h2>Fitting models</h2>
<p>Models are fit using <code>lmFit</code>.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(betas, design)</span></code></pre></div>
<pre><code>## Warning: Partial NA coefficients for 5 probe(s)</code></pre>
<p>Models can also be fit for random effects using the
<code>dupcor</code> object created above.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(betas, design,</span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a>             <span class="at">block =</span> <span class="fu">colData</span>(methData)<span class="sc">$</span>ID,</span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a>             <span class="at">correlation =</span> dupcor<span class="sc">$</span>consensus.correlation)</span></code></pre></div>
<hr />
</div>
<div id="extracting-results" class="section level2">
<h2>Extracting results</h2>
<p>Results can then be extracted from the <code>fit</code> object,
including the:</p>
<ul>
<li>coefficients <code>coef</code></li>
<li>standard errors <code>se</code></li>
<li>t-statistics <code>tstat</code></li>
<li>p-values <code>pval</code>, and</li>
<li>number of observations <code>n</code></li>
</ul>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a>coef <span class="ot">&lt;-</span> fit<span class="sc">$</span>coefficients[, <span class="dv">2</span>]</span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a>se <span class="ot">&lt;-</span> fit<span class="sc">$</span>stdev.unscaled[, <span class="dv">2</span>] <span class="sc">*</span> fit<span class="sc">$</span>sigma</span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a>tstat <span class="ot">&lt;-</span> coef <span class="sc">/</span> se</span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a>pval <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> <span class="fu">pt</span>(<span class="sc">-</span><span class="fu">abs</span>(tstat), fit<span class="sc">$</span>df.residual)</span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">ncol</span>(design) <span class="sc">+</span> fit<span class="sc">$</span>df.residual</span></code></pre></div>
<hr />
</div>
</div>
<div id="bacon" class="section level1">
<h1><code>bacon</code></h1>
<div id="motivation-1" class="section level2">
<h2>Motivation</h2>
<p>We developed a package called <a
href="http://bioconductor.org/packages/bacon"><strong>bacon</strong></a><sup>11</sup>
to estimate and correct for bias and inflation of test statistics in
EWAS. This maximizes power while properly controlling the false positive
rate, by estimating the empirical null distribution using Bayesian
statistics. The utility of the tool was illustrated through the
application of meta-analysis by performing epigenome- and
transcriptome-wide association studies (EWASs/TWAS) on age and smoking
which highlighted an overlap in differential methylation and expression
of associated genes.</p>
<hr />
</div>
<div id="adjusting-for-bias-and-inflation" class="section level2">
<h2>Adjusting for bias and inflation</h2>
<p>We advise using <code>bacon</code> to estimate inflation and bias
often observed in EWAS. This object can then be printed to report the
estimated bias and inflation.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>bc <span class="ot">&lt;-</span> <span class="fu">bacon</span>(<span class="at">teststatistics =</span> tstat,</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>            <span class="at">effectsizes =</span> coef,</span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a>            <span class="at">standarderrors =</span> se,</span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a>            <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Use multinomial weighted sampling...</code></pre>
<pre><code>## threshold = -5.3985
## Starting values:
## p0 = 1.0000, p1 = 0.0000, p2 = 0.0000
## mu0 = 0.0040, mu1 = 5.5135, mu2 = -5.5055
## sigma0 = 1.0206, sigma1 = 1.0206, sigma2 = 1.0206</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>bc</span></code></pre></div>
<pre><code>## Bacon-object containing 1 set(s) of 745368 test-statistics.
## ...estimated bias: 0.0047.
## ...estimated inflation: 1.
## 
## Empirical null estimates are based on 5000 iterations with a burnin-period of 2000.</code></pre>
<p>P-values and t-statistics can then be adjusted</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>pval <span class="ot">&lt;-</span> bacon<span class="sc">::</span><span class="fu">pval</span>(bc)</span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a>tstat <span class="ot">&lt;-</span> bacon<span class="sc">::</span><span class="fu">tstat</span>(bc)</span></code></pre></div>
<hr />
</div>
<div id="plots" class="section level2">
<h2>Plots</h2>
<p>Bacon also provides some visualizations of its performance, which can
come in handy.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">plot</span>(bc, <span class="at">type=</span><span class="st">&quot;hist&quot;</span>))</span></code></pre></div>
<pre><code>## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="06_EWAS_files/figure-html/617plot-1.png" width="672" /></p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">plot</span>(bc, <span class="at">type=</span><span class="st">&quot;qq&quot;</span>))</span></code></pre></div>
<p><img src="06_EWAS_files/figure-html/617plot-2.png" width="672" /></p>
<p>Additionally, rerunning bacon can provide an estimate of any residual
bias or inflation.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a>bc <span class="ot">&lt;-</span> bacon<span class="sc">::</span><span class="fu">bacon</span>(<span class="at">teststatistics =</span> tstat,</span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a>                   <span class="at">effectsizes =</span> coef,</span>
<span id="cb30-3"><a href="#cb30-3" tabindex="-1"></a>                   <span class="at">standarderrors =</span> se,</span>
<span id="cb30-4"><a href="#cb30-4" tabindex="-1"></a>                   <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Use multinomial weighted sampling...</code></pre>
<pre><code>## threshold = -5.3985
## Starting values:
## p0 = 1.0000, p1 = 0.0000, p2 = 0.0000
## mu0 = -0.0007, mu1 = 5.4171, mu2 = -5.4185
## sigma0 = 1.0036, sigma1 = 1.0036, sigma2 = 1.0036</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a>bc</span></code></pre></div>
<pre><code>## Bacon-object containing 1 set(s) of 745368 test-statistics.
## ...estimated bias: 0.00018.
## ...estimated inflation: 1.
## 
## Empirical null estimates are based on 5000 iterations with a burnin-period of 2000.</code></pre>
<hr />
</div>
</div>
<div id="saving-results" class="section level1">
<h1>Saving results</h1>
<p>Bacon-adjusted results can then be saved, for use in downstream
analysis.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a>limma_base <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">cpg =</span> <span class="fu">rownames</span>(fit<span class="sc">$</span>coefficients), </span>
<span id="cb35-2"><a href="#cb35-2" tabindex="-1"></a>                         <span class="at">beta =</span> coef, <span class="at">SE =</span> se, </span>
<span id="cb35-3"><a href="#cb35-3" tabindex="-1"></a>                         <span class="at">p =</span> pval, </span>
<span id="cb35-4"><a href="#cb35-4" tabindex="-1"></a>                         <span class="at">t =</span> tstat, <span class="at">N =</span> n)</span></code></pre></div>
<hr />
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
