<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title></title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 52px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 57px;
  margin-top: -57px;
}

.section h2 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h3 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h4 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h5 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h6 {
  padding-top: 57px;
  margin-top: -57px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DNAmArray</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="01_SampleQualityControl.html">Quality control</a>
</li>
<li>
  <a href="02_Visualization.html">Visualizations</a>
</li>
<li>
  <a href="03_Normalization.html">Normalization</a>
</li>
<li>
  <a href="04_ProbeFiltering.html">Probe filtering</a>
</li>
<li>
  <a href="05_Miscellaneous.html">Miscellaneous</a>
</li>
<li>
  <a href="06_EWAS.html">Ewas</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">About</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">




</div>


<div id="simple-ewas-using-limma" class="section level1">
<h1>Simple EWAS using <strong>limma</strong></h1>
<p>M-values valid to run EWAS using linear regression approach.</p>
<p>Here skipping the empirical Bayes step, because of the large sample size this should give very similar results?</p>
<pre class="r"><code>covariates &lt;- c(&quot;DNA_BloodSampling_Age&quot;, &quot;Sex&quot;, &quot;Sentrix_Position&quot;, &quot;Smoking&quot;, &quot;Mono_Perc&quot;, &quot;Baso_Perc&quot;, &quot;Eos_Perc&quot;, &quot;Neut_Perc&quot;) 
design &lt;- data.frame(Int=1, colData(mvalues)[, covariates])
complete &lt;- apply(design, 1, function(x) sum(is.na(x)) == 0 )
data &lt;- assays(mvalues)$data
design &lt;- design[complete,]
data &lt;- data[,complete]
dim(data)</code></pre>
<pre><code>## [1] 483098     22</code></pre>
<pre class="r"><code>design$Sentrix_Position &lt;- as.integer(as.factor(design$Sentrix_Position))</code></pre>
<pre class="r"><code>library(limma)
fit &lt;- lmFit(data, design)     
tstat &lt;- fit$coef/fit$stdev.unscaled/fit$sigma        
pval &lt;- 2*pnorm(-abs(tstat[,2]))
padj &lt;- p.adjust(sort(pval, decreasing=FALSE), method=&quot;bonf&quot;)
head(padj[padj &lt; 0.05])</code></pre>
<pre><code>##   cg26712943   cg00661620   cg08595806   cg15993871   cg26492480 
## 8.967774e-13 8.168777e-11 4.483074e-08 8.776593e-07 2.008576e-06 
##   cg00215551 
## 2.439025e-06</code></pre>
</div>
<div id="unobserved-covariates-using-cate" class="section level1">
<h1>Unobserved covariates using <strong>cate</strong></h1>
<p>Often unobserved technical or biological covariates can be present. A recent approach available from cran <a href="https://cran.r-project.org/package=cate"><strong>cate</strong></a> is able to estimate the unobserved covariates. There are other methods as well, …</p>
<pre class="r"><code>library(cate)
fa &lt;- cate.fit(design[,2,drop=FALSE], design[,-2], t(data), r = 3, calibrate=FALSE, adj.method=&quot;naive&quot;)</code></pre>
<p>Here we assumed the presence of three unobserved covariates but cate has two method implemented that can estimate the precise number of unobserved covariates.</p>
</div>
<div id="correct-for-bias-and-inflation" class="section level1">
<h1>Correct for bias and inflation</h1>
<p>As is seen in GWAS, inflation, also occurs in EWAS. And not only inflation but bias can be observed too. We developed a package called <a href="http://bioconductor.org/bacon"><strong>bacon</strong></a> to estimate and correct for bias and inflation for EWAS (and TWAS)<span class="citation">(Iterson, Zwet, and Heijmans 2017)</span>.</p>
<pre class="r"><code>library(bacon)
tstats &lt;- cbind(limma=tstat[,2], cate=as.vector(fa$beta.t))
bc &lt;- bacon(tstats)</code></pre>
<pre><code>## Detected 14 workers!
##  Running in parallel!</code></pre>
<pre class="r"><code>bc</code></pre>
<pre><code>## Bacon-object containing 2 set(s) of 483098 test-statistics.
## ...estimated bias: -0.099,-0.099.
## ...estimated inflation: 1,1.
## 
## Emprical null estimates are based on 5000 iterations with a burnin-period of 2000.</code></pre>
<pre class="r"><code>print(plot(bc, type=&quot;hist&quot;))</code></pre>
<pre><code>## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<pre><code>## Warning: Removed 19324 rows containing non-finite values (stat_bin).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_bar).</code></pre>
<pre><code>## Warning: Removed 9662 rows containing missing values (geom_path).

## Warning: Removed 9662 rows containing missing values (geom_path).</code></pre>
<p><img src="06_EWAS_files/figure-html/bacon-1.png" width="672" /></p>
<pre class="r"><code>print(plot(bc, type=&quot;qq&quot;))</code></pre>
<p><img src="06_EWAS_files/figure-html/bacon-2.png" width="672" /></p>
<pre class="r"><code>head(pval(bc))</code></pre>
<pre><code>##                 limma       cate
## cg01707559 0.80294660 0.80239960
## cg02004872 0.27920089 0.27842475
## cg02494853 0.34612971 0.34532481
## cg03244189 0.02563783 0.02542783
## cg03706273 0.60274926 0.60257113
## cg04023335 0.80514641 0.80521353</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references">
<div id="ref-Iterson2017">
<p>Iterson, M. van, E. W. van Zwet, and B. T. Heijmans. 2017. “Controlling bias and inflation in epigenome- and transcriptome-wide association studies using the empirical null distribution.” <em>Genome Biol.</em> 18 (1): 19.</p>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
