<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title></title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 52px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 57px;
  margin-top: -57px;
}

.section h2 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h3 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h4 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h5 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h6 {
  padding-top: 57px;
  margin-top: -57px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DNAmArray</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="02_SampleQualityControl.html">Quality control</a>
</li>
<li>
  <a href="03_ReadingIdats.html">Reading idats</a>
</li>
<li>
  <a href="04_Visualization.html">Visualizations</a>
</li>
<li>
  <a href="05_Normalization.html">Normalization</a>
</li>
<li>
  <a href="06_ProbeFiltering.html">Probe filtering</a>
</li>
<li>
  <a href="07_EWAS.html">Ewas</a>
</li>
<li>
  <a href="08_Miscellaneous.html">Miscellaneous</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">About</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">




</div>


<p><strong>TODO add more background Maarten</strong> <strong>beware 450K specific</strong></p>
<pre class="r"><code>RGset &lt;- probeFiltering(RGset)</code></pre>
<div id="number-of-beads" class="section level1">
<h1>Number of beads</h1>
<p>Occasionally some probes as measured with a too small number of beads leading to unreliable probe intensities. By default probes intensities are set to NA if the probe intensity is based on less then 3 beads.</p>
</div>
<div id="zero-intensities" class="section level1">
<h1>Zero intensities</h1>
<p>Ifthere are no beads at all for a certain probe the probe intensity will be exactly zero, these are set to NA, as well.</p>
<p>Samples which have more then 5% of their probes set to NA, based on the above filter steps, will be removed from the <code>RGChannelSet</code>.</p>
</div>
<div id="cpg-level-filtering" class="section level1">
<h1>CpG-level filtering</h1>
</div>
<div id="detection-p-value" class="section level1">
<h1>detection P-value</h1>
<p>The negative controls on the 450k array can be used to detect outlying samples e.g., samples that have a significant larger background then average, this is summarized in the so-called detection P-value.</p>
<p>Detection P-values are calculated using the <code>detectionP</code>-function from <a href="http:/bioconductor.org/minfi"><strong>minfi</strong></a>. By default detection P-values with a value &gt;0.01 are set to NA.</p>
</div>
<div id="cross-reactive-probescpgs" class="section level1">
<h1>Cross reactive probes/CpGs</h1>
<p>485,577 assays (482,421 CpG sites, 3091 non-CpG sites and 65 random SNPs)…<span class="citation">(Bibikova et al. 2011)</span></p>
<p>Furthermore, several probes do not map uniquely to the genome or contain common polymorphisms. Chen <span class="citation">(Chen et al. 2013)</span> analyzed in detail cross-reactive and polymorphic probes. We exclude the set of polymorphic CG and CH probes (42.403) discovered by Chen.</p>
<p>482,421 CpG sites are measured using two chemistries Type I (135,476) and Type II (350,036). the Type I chemistry uses two probes to measure a CpG site while the Type II measure CpG methylation using one probe. The Type II chemistry was specially designed to measure CpG methylation in low CpG dense regions. There are</p>
<p>Depending on the methylation status of the probe either of the green or red fluorescence signals is detected a strongest.</p>
<p>Information on Type I and II probes, address, etc. can be extracted using <a href="http:/bioconductor.org/minfi"><strong>minfi</strong></a>-functionality.</p>
<pre class="r"><code>library(minfi)
locusNames &lt;- getManifestInfo(RGset, &quot;locusNames&quot;)
typeII &lt;- getProbeInfo(RGset, type = &quot;II&quot;)
nrow(typeII)
typeI &lt;- getProbeInfo(RGset, type = &quot;I&quot;)
head(typeI)
nrow(typeI)</code></pre>
<p>482421 + 3091 (excluding 65 SNPs)</p>
<p>Methylated and Unmethylated Intensities</p>
</div>
<div id="success-rates" class="section level1">
<h1>Success Rates</h1>
<p>All these probe filter-steps are implemented in the function <code>filtering</code>.</p>
<p>RGset containing Red/Green signal intensities are converted to Methylated and Unmethylated signals (see also, <code>?preprocessRaw</code>, from minfi). These can in turn be converted to either beta-values or M-values.</p>
<pre class="r"><code>betas &lt;- reduce(GRset, RGset, what=&quot;beta&quot;)</code></pre>
<pre class="r"><code>mvalues &lt;- reduce(GRset, RGset, what=&quot;M&quot;)</code></pre>
</div>
<div id="array-design" class="section level1">
<h1>Array design</h1>
<p>Chips, plates, arrays, …</p>
<p>probe addressess A (Type I and Type II) B (Type I only) There are 620988 probes on the array</p>
<p>Number of beads</p>
<p>Each probe is measured using more then one bead, the average signal intensities is used as a measure for the probe intensity.</p>
<div class="references">
<p>Bibikova, M., B. Barnes, C. Tsan, V. Ho, B. Klotzle, J. M. Le, D. Delano, et al. 2011. “High Density DNA Methylation Array with Single CpG Site Resolution.” <em>Genomics</em> 98 (4) (Oct): 288–295.</p>
<p>Chen, Y. A., M. Lemire, S. Choufani, D. T. Butcher, D. Grafodatskaya, B. W. Zanke, S. Gallinger, T. J. Hudson, and R. Weksberg. 2013. “Discovery of Cross-Reactive Probes and Polymorphic CpGs in the Illumina Infinium HumanMethylation450 Microarray.” <em>Epigenetics</em> 8 (2) (Feb): 203–209.</p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
