<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Probe-level filtering</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; }
code > span.dt { color: #204a87; }
code > span.dv { color: #0000cf; }
code > span.bn { color: #0000cf; }
code > span.fl { color: #0000cf; }
code > span.ch { color: #4e9a06; }
code > span.st { color: #4e9a06; }
code > span.co { color: #8f5902; font-style: italic; }
code > span.ot { color: #8f5902; }
code > span.al { color: #ef2929; }
code > span.fu { color: #000000; }
code > span.er { font-weight: bold; }
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">My Website</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="01_Install.html">Install</a>
</li>
<li>
  <a href="02_SampleQualityControl.html">Sample quality control</a>
</li>
<li>
  <a href="03_ReadingIdats.html">Reading idats</a>
</li>
<li>
  <a href="04_Visualization.html">Visualization</a>
</li>
<li>
  <a href="05_Normalization.html">Normalization</a>
</li>
<li>
  <a href="06_ProbeFiltering.html">Probe filtering</a>
</li>
<li>
  <a href="07_Miscellaneous.html">Miscellaneous</a>
</li>
<li>
  <a href="08_SummarizedExperiment.html">SummarizedExperiment</a>
</li>
<li>
  <a href="09_EWAS.html">Ewas</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Probe-level filtering</h1>

</div>


<div id="sex-check" class="section level2">
<h2>Sex check</h2>
<p><strong>add explaination</strong></p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(BiocParallel)
<span class="kw">library</span>(DNAmArray)
<span class="kw">register</span>(<span class="kw">MulticoreParam</span>(<span class="dv">8</span>))
RGset &lt;-<span class="st"> </span><span class="kw">read.450k.exp.par</span>(targets)</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">pheno &lt;-<span class="st"> </span><span class="kw">getView</span>(<span class="st">&quot;allPhenotypes&quot;</span>)
id &lt;-<span class="st"> </span><span class="kw">match</span>(targets$ids, pheno$ids)
sex &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">2</span>, <span class="kw">nrow</span>(targets))
sex[!<span class="kw">is.na</span>(id)] &lt;-<span class="st"> </span>pheno$Sex[id[!<span class="kw">is.na</span>(id)]]</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">betas &lt;-<span class="st"> </span><span class="kw">getBeta</span>(RGset)
predicted &lt;-<span class="st"> </span><span class="kw">getSex.DNAmArray</span>(betas)
assumed &lt;-<span class="st"> </span>sex
assumed &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="kw">factor</span>(assumed, <span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;Male&quot;</span>, <span class="st">&quot;Female&quot;</span>, <span class="st">&quot;Unknown&quot;</span>)))
<span class="kw">str</span>(predicted)
<span class="kw">str</span>(assumed)
<span class="kw">print</span>(<span class="kw">table</span>(predicted, assumed))</code></pre>
</div>
<div id="genotype-check" class="section level2">
<h2>Genotype check</h2>
<p>Beta-values of the probes containing SNP can be extract from the RGset using <a href="http:/bioconductor.org/minfi"><strong>minfi</strong></a>â€™s <code>getSnpBeta</code>-function.</p>
<pre class="sourceCode r"><code class="sourceCode r">betas &lt;-<span class="st"> </span><span class="kw">getSnpBeta</span>(RGset)
<span class="kw">dim</span>(betas)</code></pre>
<p>Now we convert the beta-values to alleles using K-means clustering. A similar approach is taken by <a href="http:/bioconductor.org/wateRmelon"><strong>wateRmelon</strong></a>.</p>
<pre class="sourceCode r"><code class="sourceCode r">methAlleles &lt;-<span class="st"> </span><span class="kw">beta2allele</span>(betas)
methAlleles[<span class="dv">1</span>:<span class="dv">5</span>, <span class="dv">1</span>:<span class="dv">5</span>]
<span class="kw">dim</span>(methAlleles)</code></pre>
<p>If genotypes are available using the same or an overlapping set of SNPs these can be used to confirm the sample identity or to discover swaps. To ensure the the same alleles are called the function <code>relabelgenotypes</code> should be run first.</p>
<pre class="sourceCode r"><code class="sourceCode r">methAlleles &lt;-<span class="st"> </span><span class="kw">relabelgenotypes</span>(methAlleles, genoAlleles)
data &lt;-<span class="st"> </span><span class="kw">alleleSharing</span>(methAlleles, genoAlleles)</code></pre>
<p>Here we just correlated all DNA samples with each other to check for unreported relatedness. From the genotypes we calculated identity-by-state for each SNPs and summarize by mean and variance. Based on a idea from GRR-tool <span class="citation">[@Abecasis2001]</span> for the detection of sample relationship in genotype data.</p>
<pre class="sourceCode r"><code class="sourceCode r">data &lt;-<span class="st"> </span><span class="kw">alleleSharing</span>(methAlleles)
<span class="kw">dim</span>(data)
<span class="kw">head</span>(data)</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">km &lt;-<span class="st"> </span><span class="kw">kmeans</span>(data[,<span class="dv">1</span>:<span class="dv">2</span>], <span class="dt">centers=</span><span class="kw">matrix</span>(<span class="kw">c</span>(<span class="kw">tapply</span>(data[,<span class="dv">1</span>], data[,<span class="dv">5</span>], median),
                                          <span class="kw">tapply</span>(data[,<span class="dv">2</span>], data[,<span class="dv">5</span>], median)), <span class="dv">2</span>, <span class="dv">2</span>))
data$predicted &lt;-<span class="st"> </span><span class="kw">factor</span>(km$cluster)
<span class="kw">levels</span>(data$predicted) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;unrelated&quot;</span>, <span class="st">&quot;related&quot;</span>)
<span class="kw">table</span>(<span class="st">`</span><span class="dt">Predicted relationship</span><span class="st">`</span>=data$predicted, <span class="st">`</span><span class="dt">Assumed relationship</span><span class="st">`</span>=data$relationship)
<span class="kw">head</span>(<span class="kw">subset</span>(data[<span class="kw">order</span>(data$mean, <span class="dt">decreasing=</span><span class="ot">TRUE</span>),], predicted !=<span class="st"> </span>relationship), <span class="dt">n=</span><span class="dv">10</span>)</code></pre>
</div>
<div id="imputation" class="section level2">
<h2>Imputation</h2>
<p>Several algorithms do not allow NAâ€™s but beta-values and or M-values can contain NAâ€™s after preprocessing. Either because some probes/CpG were specifically set NA or because of transformation from Intensities to beta or M-values.</p>
<p>The <a href="https://cran.r-project.org/package=imputex"><strong>impute</strong></a> package can be used to impute these NAâ€™s, for example using the knn approach implemented in the <code>knn.impute</code>-function<span class="citation">[@Troyanskaya2001]</span>.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(impute)
mvalues &lt;-<span class="st"> </span><span class="kw">impute.knn</span>(<span class="kw">as.matrix</span>(mvalues))$data</code></pre>
</div>
<div id="annotation" class="section level2">
<h2>Annotation</h2>
<p>We have implemented a small function to map CpG-probes to their nearest genes.</p>
<pre class="sourceCode r"><code class="sourceCode r">cpgs &lt;-<span class="st"> </span><span class="kw">head</span>(<span class="kw">rownames</span>(mvalues))
<span class="kw">cpgInfo</span>(cpgs, <span class="dt">TxDb=</span><span class="st">&quot;TxDb.Hsapiens.UCSC.hg19.ensGene&quot;</span>)</code></pre>
</div>
<div id="white-blood-cell-counts" class="section level2">
<h2>White Blood Cell Counts</h2>
<p>White blood cell count can be predicted using Housemanâ€™s method<span class="citation">[@Houseman2012]</span> implemented in the <a href="http://bioconductor.org/minfi"><strong>minfi</strong></a><span class="citation">[@Jaffe2014]</span>.</p>
<p>See also our approach for the imputation of white blood cell counts:<a href="https://github.com/mvaniterson/wbccPredictor">wbccPredictor</a></p>
</div>
<div id="phenotyping" class="section level2">
<h2>Phenotyping</h2>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
