<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title></title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 52px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 57px;
  margin-top: -57px;
}

.section h2 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h3 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h4 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h5 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h6 {
  padding-top: 57px;
  margin-top: -57px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DNAmArray</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="SampleQualityControl.html">Quality control</a>
</li>
<li>
  <a href="Visualization.html">Visualizations</a>
</li>
<li>
  <a href="Normalization.html">Normalization</a>
</li>
<li>
  <a href="ProbeFiltering.html">Probe filtering</a>
</li>
<li>
  <a href="Miscellaneous.html">Miscellaneous</a>
</li>
<li>
  <a href="EWAS.html">Ewas</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">About</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">




</div>


<div id="acknowledgements" class="section level1">
<h1>Acknowledgements</h1>
<p>Contributors to the development of the package/workflow are:</p>
<p>Elmar W. Tobi, Roderick Slieker, Wouter den Hollander, Rene Luijk, Koen F.Dekkers and Bas Heijmans (PI) from Molecular Epidemiology Section, Leiden University Medical Center, The Netherlands.</p>
<p>Development of the workflow and generation of example data in LLS was funded by BBMRI-NL, a research infrastructure financed by the Dutch government (NWO 184.021.007).</p>
<p>Maarten van Iterson is the main author of the package/workflow and should be contacted for any questions: m.van_iterson at lumc.nl</p>
<p>Paul Hop and Jenny van Dongen were involved in testing the first versions and helped us improving the packages/workflow.</p>
<p>Please cite van Iterson at al. <span class="citation">(Iterson, Zwet, and Heijmans 2017)</span> when using the <a href="https://github.com/molepi/DNAmArray"><strong>DNAmArray</strong></a> package or workflow.</p>
<p>This web-site is generated with <a href="http://rmarkdown.rstudio.com/index.html"><strong>rmarkdown</strong></a> on 15 March, 2017 using R-version and packages:</p>
<pre><code>## R version 3.3.2 (2016-10-31)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 16.04.2 LTS
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    parallel  methods   stats     graphics  grDevices utils    
## [8] datasets  base     
## 
## other attached packages:
##  [1] IlluminaHumanMethylation450kanno.ilmn12.hg19_0.6.0
##  [2] IlluminaHumanMethylation450kmanifest_0.4.0        
##  [3] BBMRIomics_1.0.2                                  
##  [4] bacon_1.2.0                                       
##  [5] ellipse_0.3-8                                     
##  [6] ggplot2_2.1.0                                     
##  [7] impute_1.48.0                                     
##  [8] fastcluster_1.1.21                                
##  [9] pheatmap_1.0.8                                    
## [10] irlba_2.1.2                                       
## [11] Matrix_1.2-7.1                                    
## [12] limma_3.30.12                                     
## [13] cate_1.0.4                                        
## [14] BatchJobs_1.6                                     
## [15] BBmisc_1.10                                       
## [16] BiocParallel_1.8.1                                
## [17] MethylAid_1.8.0                                   
## [18] knitr_1.15.1                                      
## [19] DNAmArray_0.0.1                                   
## [20] minfi_1.20.2                                      
## [21] bumphunter_1.14.0                                 
## [22] locfit_1.5-9.1                                    
## [23] iterators_1.0.8                                   
## [24] foreach_1.4.3                                     
## [25] Biostrings_2.42.1                                 
## [26] XVector_0.14.0                                    
## [27] SummarizedExperiment_1.4.0                        
## [28] FDb.InfiniumMethylation.hg19_2.2.0                
## [29] org.Hs.eg.db_3.4.0                                
## [30] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2           
## [31] GenomicFeatures_1.26.3                            
## [32] AnnotationDbi_1.36.2                              
## [33] Biobase_2.34.0                                    
## [34] GenomicRanges_1.26.3                              
## [35] GenomeInfoDb_1.10.3                               
## [36] IRanges_2.8.1                                     
## [37] S4Vectors_0.12.1                                  
## [38] BiocGenerics_0.20.0                               
## [39] BiocInstaller_1.24.0                              
## 
## loaded via a namespace (and not attached):
##  [1] colorspace_1.2-7         siggenes_1.48.0         
##  [3] mclust_5.2               rprojroot_1.2           
##  [5] corpcor_1.6.8            base64enc_0.1-3         
##  [7] base64_2.0               hexbin_1.27.1           
##  [9] leapp_1.2                codetools_0.2-15        
## [11] splines_3.3.2            fail_1.3                
## [13] jsonlite_1.3             Rsamtools_1.26.1        
## [15] gridBase_0.4-7           annotate_1.52.1         
## [17] shiny_0.14.1             readr_1.0.0             
## [19] httr_1.2.1               backports_1.0.5         
## [21] assertthat_0.1           htmltools_0.3.5         
## [23] tools_3.3.2              gtable_0.2.0            
## [25] doRNG_1.6                Rcpp_0.12.7             
## [27] multtest_2.30.0          preprocessCore_1.36.0   
## [29] nlme_3.1-128             rtracklayer_1.34.2      
## [31] stringr_1.2.0            mime_0.5                
## [33] rngtools_1.2.4           RevoUtilsMath_10.0.0    
## [35] XML_3.98-1.4             beanplot_1.2            
## [37] zlibbioc_1.20.0          MASS_7.3-45             
## [39] scales_0.4.0             BSgenome_1.42.0         
## [41] VariantAnnotation_1.20.2 GEOquery_2.40.0         
## [43] RColorBrewer_1.1-2       yaml_2.1.14             
## [45] pkgmaker_0.22            biomaRt_2.30.0          
## [47] reshape_0.8.6            stringi_1.1.2           
## [49] RSQLite_1.0.0            genefilter_1.56.0       
## [51] checkmate_1.8.1          chron_2.3-47            
## [53] matrixStats_0.51.0       bitops_1.0-6            
## [55] nor1mix_1.2-2            evaluate_0.10           
## [57] lattice_0.20-34          ruv_0.9.6               
## [59] esaBcv_1.2.1             labeling_0.3            
## [61] GenomicAlignments_1.10.0 plyr_1.8.4              
## [63] magrittr_1.5             sendmailR_1.2-1         
## [65] R6_2.2.0                 DBI_0.5-1               
## [67] mgcv_1.8-15              survival_2.40-1         
## [69] RCurl_1.95-4.8           tibble_1.2              
## [71] rmarkdown_1.3.9004       grid_3.3.2              
## [73] sva_3.22.0               data.table_1.9.6        
## [75] digest_0.6.12            svd_0.4                 
## [77] xtable_1.8-2             httpuv_1.3.3            
## [79] brew_1.0-6               illuminaio_0.16.0       
## [81] openssl_0.9.5            munsell_0.4.3           
## [83] registry_0.3             quadprog_1.5-5</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references">
<div id="ref-Iterson2017">
<p>Iterson, M. van, E. W. van Zwet, and B. T. Heijmans. 2017. “Controlling bias and inflation in epigenome- and transcriptome-wide association studies using the empirical null distribution.” <em>Genome Biol.</em> 18 (1): 19.</p>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
