<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title></title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 52px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h2 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h3 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h4 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h5 {
  padding-top: 57px;
  margin-top: -57px;
}
.section h6 {
  padding-top: 57px;
  margin-top: -57px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DNAmArray</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="01_SampleQualityControl.html">Quality control</a>
</li>
<li>
  <a href="02_Visualization.html">Visualizations</a>
</li>
<li>
  <a href="03_Normalization.html">Normalization</a>
</li>
<li>
  <a href="04_ProbeFiltering.html">Probe filtering</a>
</li>
<li>
  <a href="05_Miscellaneous.html">Miscellaneous</a>
</li>
<li>
  <a href="06_EWAS.html">Ewas</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">About</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">




</div>


<div id="acknowledgements" class="section level1">
<h1>Acknowledgements</h1>
<p>Contributors to the development of the package/workflow are:</p>
<p>Elmar W. Tobi, Roderick Slieker, Wouter den Hollander, Rene Luijk, Koen F.Dekkers and Bas Heijmans (PI) from Molecular Epidemiology Section, Leiden University Medical Center, The Netherlands.</p>
<p>Development of the workflow and generation of previously used example data in LLS was funded by BBMRI-NL, a research infrastructure financed by the Dutch government (NWO 184.021.007).</p>
<p>Maarten van Iterson is the main author of the package/workflow. Lucy Sinke is the current maintainer and should be contacted with any questions: <em><a href="mailto:l.j.sinke@lumc.nl">l.j.sinke@lumc.nl</a></em></p>
<p>Paul Hop and Jenny van Dongen were involved in testing the first versions and helped us improving the packages/workflow.</p>
<p>Please cite van Iterson at al. <span class="citation">(Iterson, Zwet, and Heijmans 2017)</span> when using the <a href="https://github.com/molepi/DNAmArray"><strong>DNAmArray</strong></a> package or workflow.</p>
<p>This web-site is generated with <a href="http://rmarkdown.rstudio.com/index.html"><strong>rmarkdown</strong></a> on 12 April, 2019 using R-version and packages:</p>
<pre><code>## R version 3.5.3 (2019-03-11)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 17134)
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=English_United Kingdom.1252 
## [2] LC_CTYPE=English_United Kingdom.1252   
## [3] LC_MONETARY=English_United Kingdom.1252
## [4] LC_NUMERIC=C                           
## [5] LC_TIME=English_United Kingdom.1252    
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] bacon_1.10.1                                      
##  [2] ellipse_0.4.1                                     
##  [3] ggplot2_3.1.1                                     
##  [4] impute_1.56.0                                     
##  [5] IlluminaHumanMethylation450kanno.ilmn12.hg19_0.6.0
##  [6] IlluminaHumanMethylation450kmanifest_0.4.0        
##  [7] fastcluster_1.1.25                                
##  [8] pheatmap_1.0.12                                   
##  [9] irlba_2.3.3                                       
## [10] Matrix_1.2-15                                     
## [11] limma_3.38.3                                      
## [12] cate_1.0.4                                        
## [13] BatchJobs_1.7                                     
## [14] BBmisc_1.11                                       
## [15] MethylAid_1.16.0                                  
## [16] knitr_1.22                                        
## [17] DNAmArray_0.1.2                                   
## [18] minfi_1.28.4                                      
## [19] bumphunter_1.24.5                                 
## [20] locfit_1.5-9.1                                    
## [21] iterators_1.0.10                                  
## [22] foreach_1.4.4                                     
## [23] Biostrings_2.50.2                                 
## [24] XVector_0.22.0                                    
## [25] SummarizedExperiment_1.12.0                       
## [26] DelayedArray_0.8.0                                
## [27] BiocParallel_1.16.6                               
## [28] matrixStats_0.54.0                                
## [29] FDb.InfiniumMethylation.hg19_2.2.0                
## [30] org.Hs.eg.db_3.7.0                                
## [31] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2           
## [32] GenomicFeatures_1.34.8                            
## [33] AnnotationDbi_1.44.0                              
## [34] Biobase_2.42.0                                    
## [35] GenomicRanges_1.34.0                              
## [36] GenomeInfoDb_1.18.2                               
## [37] IRanges_2.16.0                                    
## [38] S4Vectors_0.20.1                                  
## [39] BiocGenerics_0.28.0                               
## [40] BiocInstaller_1.32.1                              
## 
## loaded via a namespace (and not attached):
##   [1] backports_1.1.4          plyr_1.8.4              
##   [3] lazyeval_0.2.2           splines_3.5.3           
##   [5] gridBase_0.4-7           sva_3.30.1              
##   [7] digest_0.6.18            htmltools_0.3.6         
##   [9] svd_0.4.3                magrittr_1.5            
##  [11] checkmate_1.9.1          memoise_1.1.0           
##  [13] readr_1.3.1              annotate_1.60.1         
##  [15] askpass_1.1              siggenes_1.56.0         
##  [17] prettyunits_1.0.2        colorspace_1.4-1        
##  [19] blob_1.1.1               xfun_0.6                
##  [21] dplyr_0.8.0.1            crayon_1.3.4            
##  [23] RCurl_1.95-4.12          hexbin_1.27.2           
##  [25] genefilter_1.64.0        GEOquery_2.50.5         
##  [27] brew_1.0-6               survival_2.43-3         
##  [29] sendmailR_1.2-1          glue_1.3.1              
##  [31] ruv_0.9.7                registry_0.5-1          
##  [33] gtable_0.3.0             zlibbioc_1.28.0         
##  [35] htm2txt_2.1.1            Rhdf5lib_1.4.3          
##  [37] HDF5Array_1.10.1         scales_1.0.0            
##  [39] DBI_1.0.0                rngtools_1.3.1          
##  [41] bibtex_0.4.2             Rcpp_1.0.1              
##  [43] xtable_1.8-3             progress_1.2.0          
##  [45] bit_1.1-14               mclust_5.4.3            
##  [47] preprocessCore_1.44.0    httr_1.4.0              
##  [49] RColorBrewer_1.1-2       pkgconfig_2.0.2         
##  [51] reshape_0.8.8            XML_3.98-1.19           
##  [53] tidyselect_0.2.5         rlang_0.3.4             
##  [55] later_0.8.0              munsell_0.5.0           
##  [57] tools_3.5.3              esaBcv_1.2.1            
##  [59] RSQLite_2.1.1            evaluate_0.13           
##  [61] stringr_1.4.0            yaml_2.2.0              
##  [63] bit64_0.9-7              beanplot_1.2            
##  [65] purrr_0.3.2              nlme_3.1-137            
##  [67] doRNG_1.7.1              mime_0.6                
##  [69] leapp_1.2                nor1mix_1.2-3           
##  [71] xml2_1.2.0               biomaRt_2.38.0          
##  [73] compiler_3.5.3           tibble_2.1.1            
##  [75] stringi_1.4.3            lattice_0.20-38         
##  [77] multtest_2.38.0          pillar_1.3.1            
##  [79] data.table_1.12.2        bitops_1.0-6            
##  [81] corpcor_1.6.9            httpuv_1.5.1            
##  [83] rtracklayer_1.42.2       R6_2.4.0                
##  [85] promises_1.0.1           gridExtra_2.3           
##  [87] codetools_0.2-16         MASS_7.3-51.1           
##  [89] assertthat_0.2.1         rhdf5_2.26.2            
##  [91] openssl_1.3              pkgmaker_0.27           
##  [93] withr_2.1.2              GenomicAlignments_1.18.1
##  [95] Rsamtools_1.34.1         GenomeInfoDbData_1.2.0  
##  [97] mgcv_1.8-27              hms_0.4.2               
##  [99] quadprog_1.5-5           grid_3.5.3              
## [101] tidyr_0.8.3              base64_2.0              
## [103] rmarkdown_1.12           DelayedMatrixStats_1.4.0
## [105] illuminaio_0.24.0        shiny_1.3.1             
## [107] base64enc_0.1-3</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references">
<div id="ref-Iterson2017">
<p>Iterson, M. van, E. W. van Zwet, and B. T. Heijmans. 2017. “Controlling bias and inflation in epigenome- and transcriptome-wide association studies using the empirical null distribution.” <em>Genome Biol.</em> 18 (1): 19.</p>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
