<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Maarten van Iterson, Elmar Tobi, Roderick Slieker, Wouter den Hollander, Koen Dekkers, Rene Luijk, Eline Slagboom and Bas Heijmans Department of Molecular Epidemiology, Leiden University Medical Center, Leiden, The Netherlands" />


<title>Title</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>


<style type="text/css">
  p.abstract{
    text-align: center;
    font-weight: bold;
  }
  div.abstract{
    margin: auto;
    width: 90%;
  }
</style>

<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">My Website</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="01_Install.html">Install</a>
</li>
<li>
  <a href="02_SampleQualityControl.html">Sample quality control</a>
</li>
<li>
  <a href="03_ReadingIdats.html">Reading idats</a>
</li>
<li>
  <a href="04_Visualization.html">Visualization</a>
</li>
<li>
  <a href="05_Normalization.html">Normalization</a>
</li>
<li>
  <a href="06_ProbeFiltering.html">Probe filtering</a>
</li>
<li>
  <a href="07_Miscellaneous.html">Miscellaneous</a>
</li>
<li>
  <a href="08_SummarizedExperiment.html">SummarizedExperiment</a>
</li>
<li>
  <a href="09_EWAS.html">Ewas</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Title</h1>
<h4 class="author"><em>Maarten van Iterson, Elmar Tobi, Roderick Slieker, Wouter den Hollander, Koen Dekkers, Rene Luijk, Eline Slagboom and Bas Heijmans Department of Molecular Epidemiology, Leiden University Medical Center, Leiden, The Netherlands</em></h4>
<div class="abstract">
<p class="abstract">Abstract</p>
<p>Streamlined workflow for the quality control, normalization and bias-free analysis of Illumina methylation array data - The Leiden approach</p>
</div>

</div>


<p><a href="https://doi.org/10.5281/zenodo.158908"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.158908.svg" alt="DOI" /></a></p>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Here we provide a complete workflow for the the preprocessing and analysis of DNA methylation array data. The workflow combines best practices in the field with in-house developed methodology and is geared towards large-scale studies, including Epigenomewide Association Studies (EWAS). A specific aim of the work flow is to support such studies within the BIOS consortium encompassing multiple omics data on 4000 individuals from 6 Dutch biobanks operating within the frame of <a href="https://www.bbmri.nl/on_offer/bios/">BBMRI-NL</a></p>
<p>The <strong>DNAmArray</strong>-package contains a series of convenient functions for the quality control, normalization and analysis of methylation array data. The workflow has been thoroughly tested on 450k array data but will similarly be applicable to the newer 850k EPIC array pending updates of Bioconductor packages from other groups.</p>
<p>The workflow and package extensively uses functionality of other BioConductor packages, e.g., for transforming idats to so-called <code>RGlist</code> we use <a href="http:/bioconductor.org/minfi"><strong>minfi</strong></a> but <strong>DNAmArray</strong> provides functionality to do this in parallel. Furthermore, we have developed two other packages <a href="http:/bioconductor.org/MethylAid"><strong>MethylAid</strong></a> <span class="citation">(M. van Iterson et al. 2014)</span> and <a href="http:/bioconductor.org/bacon"><strong>bacon</strong></a> <span class="citation">(M. M. van Iterson et al. 2016)</span>, reps., performing interactive sample quality control and bias- and inflation-correction in epigenomewide association studies, each have been deposited at BioConductor (of note: <a href="http:/bioconductor.org/bacon"><strong>bacon</strong></a> also applies to transcriptome- and any other ome-wide association studies) .</p>
</div>
<div id="leiden-longevity-study" class="section level1">
<h1>Leiden Longevity Study</h1>
<p>The example DNA methylation data that is used in this workflow is from the Leiden Longevity Study. The Leiden Longevity Study has been described in detail previously by Westendorp et al. <span class="citation">(Westendorp et al. 2009)</span>. It is a family based study consists of 1671 offspring of 421 nonagenarians sibling pairs of Dutch descent, and their 744 partners. In the current EWAS initiative 793 individuals with available DNA methylation data and phenotypic data (mean age = 60 years) were included.</p>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<p>library(DNAmArray)</p>
<div class="references">
<p>Iterson, M. van, E. W. Tobi, R. C. Slieker, W. den Hollander, R. Luijk, P. E. Slagboom, and B. T. Heijmans. 2014. “MethylAid: visual and Interactive Quality Control of Large Illumina 450k Datasets.” <em>Bioinformatics</em> 30 (23): 3435–3437.</p>
<p>Iterson, Maarten M. van, Erik W. van ZwetP. Eline Slagboom, and Bastiaan T. Heijmans. 2016. “Controlling Bias and Inflation in Epigenome- and Transcriptome-Wide Association Studies Using the Empirical Null Distribution.” <em>bioRxiv</em>. doi:<a href="http://dx.doi.org/10.1101/055772">10.1101/055772</a>. <a href="http://biorxiv.org/content/early/2016/05/27/055772">http://biorxiv.org/content/early/2016/05/27/055772</a>.</p>
<p>Westendorp, R. G., D. van Heemst, M. P. Rozing, M. Frolich, S. P. Mooijaart, G. J. Blauw, M. Beekman, B. T. Heijmans, A. J. de Craen, and P. E. Slagboom. 2009. “Nonagenarian Siblings and Their Offspring Display Lower Risk of Mortality and Morbidity Than Sporadic Nonagenarians: The Leiden Longevity Study.” <em>J Am Geriatr Soc</em> 57 (9) (Sep): 1634–1637.</p>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
