<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>index.knit</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>







<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DNAmArray</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="01_GettingStarted.html">Getting Started</a>
</li>
<li>
  <a href="02_SampleQualityControl.html">Sample-level QC</a>
</li>
<li>
  <a href="03_Normalization.html">Normalization</a>
</li>
<li>
  <a href="04_ProbeFiltering.html">Probe-level QC</a>
</li>
<li>
  <a href="05_Predict.html">Predictors</a>
</li>
<li>
  <a href="06_Masking.html">Underperforming Probes</a>
</li>
<li>
  <a href="07_EWAS.html">Analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">About</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore"><div class="line-block">Streamlined
workflow for the quality control, normalization, and analysis of
Illumina methylation array data<br />
</div></h1>
<h4 class="author"><div class="line-block">Lucy Sinke, Maarten van
Iterson, Davy Cats, BIOS Consortium, and Bas Heijmans<br />
<font size=2 color=#6b6b6b>Molecular Epidemiology, Department of
Biomedical Data Sciences, Leiden University Medical Center, Leiden, The
Netherlands</font></div></h4>

</div>


<div class="DOI">
<p><a href="http://doi.org/10.5281/zenodo.3355292"><img
src="../data/images/zenodo_3355292.svg" alt="DOI" /></a></p>
</div>
<hr />
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p><code>DNAmArray</code> is a comprehensive and modular workflow for
the pre-processing, quality control, and analysis of DNA methylation
(DNAm) array data, tailored for large-scale epigenome-wide association
studies (EWAS). It integrates best practices in the field with in-house
developed tools, and has been informed by almost a decade of research
using the <a
href="https://www.bbmri.nl/samples-images-data">Biobank-based
Integrative Omics Study (BIOS) consortium</a> data. The BIOS dataset
combines blood-based DNAm and gene expression from across six Dutch
biobanks and around 4,000 individuals, and has been used to extensively
advance our understanding of epigenetic regulation and
health<sup>1-8</sup>.</p>
<p>The workflow combines a series of convenient <a
href="https://github.com/molepi/DNAmArray"><strong>DNAmArray</strong></a>
functions with <a href="https://www.bioconductor.org">BioConductor</a>
packages, including:</p>
<ul>
<li><a
href="https://bioconductor.org/packages/release/bioc/html/minfi.html"><strong>minfi</strong></a>
for reading in IDAT files and normalization<sup>9</sup>,</li>
<li><a
href="https://www.bioconductor.org/packages/release/bioc/html/MethylAid.html"><strong>MethylAid</strong></a>
for sample-level quality control<sup>10</sup>,</li>
<li><a
href="https://bioconductor.org/packages/release/bioc/html/bacon.html"><strong>bacon</strong></a>
for controlling bias and inflation in EWAS test statistics<sup>11</sup>,
and</li>
<li><a
href="https://bioconductor.org/packages/release/bioc/html/omicsPrint.html"><strong>omicsPrint</strong></a>
for identifying and resolving sample mismatches<sup>12</sup>.</li>
</ul>
<p>While thoroughly validated on DNAm data profiled using the Illumina
Infinium HumanMethylation450 and EPIC arrays, notes within this
documentation also outline any changes needed to apply
<code>DNAmArray</code> to Infinium MethylationEPIC v2.0 BeadChip data.
In addition, any dependencies should be installed automatically, but
otherwise please refer to the relevant package’s documentation (and let
us know by opening a GitHub <a
href="https://github.com/molepi/DNAmArray/issues">issue</a>!)</p>
<p>In conclusion, <code>DNAmArray</code> provides a scalable,
reproducible, and EWAS-ready framework for DNA methylation data
analysis, compatible with evolving technologies and suitable for
integration with downstream follow-up analyses.</p>
<hr />
</div>
<div id="example-data" class="section level1">
<h1>Example Data</h1>
<p>The <a
href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE116339">example
data</a><sup>13</sup> used in this workflow is available from the <a
href="https://www.ncbi.nlm.nih.gov/geo/">NCBI Gene Expression
Omnibus</a> (GEO), a public repository of microarray data. It contains
genome-wide DNA methylation data from whole blood obtained using the
Illumina Infinium MethylationEPIC BeadChip microarray. The participants
consist of 679 children exposed to polybrominated biphenyl (PBB), an
endocrine-disrupting compound, which was accidentally added to the food
supply in Michigan in the 1970’s.</p>
<hr />
</div>
<div id="references" class="section level1">
<h1>References</h1>
<p><sup>1</sup>: Dekkers, K.F., van Iterson, M., Slieker, R.C., <i>et
al</i>. <a href="https://doi.org/10.1186/s13059-016-1000-6">Blood lipids
influence DNA methylation in circulating cells</a>. <i>Genome Biol.</i>
<b>17</b>, 138 (2016).</p>
<p><sup>2</sup>: Slieker, R.C., van Iterson, M., Luijk, R. <i>et al</i>.
<a href="https://doi.org/10.1186/s13059-016-1053-6">Age-related accrual
of methylomic variability is linked to fundamental ageing
mechanisms</a>. <i> Genome Biol.</i> <b>17</b>, 191 (2016).</p>
<p><sup>3</sup>: Bonder, M., Luijk, R., Zhernakova, D. <i>et al</i>. <a
href="https://doi.org/10.1038/ng.3721">Disease variants alter
transcription factor levels and methylation of their binding sites</a>.
<i>Nat Genet.</i> <b>49</b>, 131-138 (2017).</p>
<p><sup>4</sup>: Luijk, R., Wu, H., Ward-Caviness, C.K. <i>et al</i>. <a
href="https://doi.org/10.1038/s41467-018-05714-3">Autosomal genetic
variation is associated with DNA methylation in regions variably
escaping X-chromosome inactivation</a>. <i> Nat Commun.</i> <b>9</b>,
3738 (2018).</p>
<p><sup>5</sup>: van Rooij, J., Mandaviya, P.R., Claringbould, A., <i>et
al</i>. <a href="https://doi.org/10.1186/s13059-019-1878-x">Evaluation
of commonly used analysis strategies for epigenome- and
transcriptome-wide association studies through replication of
large-scale population studies</a>. <i>Genome Biol.</i> <b>20</b>, 1:235
(2019).</p>
<p><sup>6</sup>: Hop, P.J., Luijk, R., Daxinger, L., <i>et al</i>. <a
href="https://doi.org/10.1186/s13059-020-02114-z">Genome-wide
identification of genes regulating DNA methylation using genetic anchors
for causal inference</a>. <i>Genome Biol.</i> <b>21</b>, 220 (2020).</p>
<p><sup>7</sup>: Dekkers, K.F., Slieker, R.C., Ioan-Facsinay, A. <i>et
al</i>. <a
href="https://doi.org/10.1038/s41467-022-35663-x">Lipid-induced
transcriptomic changes in blood link to lipid metabolism and allergic
response</a>. <i>Nat Commun.</i> <b>14</b>, 544 (2023).</p>
<p><sup>8</sup>: Lui, Y., Sinke, L., Jonkman, T.H., <i>et al</i>. <a
href="https://doi.org/10.1186/s13148-023-01549-y">The inactive X
chromosome accumulates widespread epigenetic variability with age</a>.
<i>Clin Epigenetics.</i> <b>15</b>, 135 (2023).</p>
<p><sup>9</sup>: Aryee, M.J., Jaffe, A.E., Corrada-Bravo, H. <i>et
al</i>. <a href="https://doi.org/10.1093/bioinformatics/btu049">Minfi: a
flexible and comprehensive Bioconductor package for the analysis of
Infinium DNA methylation microarrays</a>. <i>Bioinformatics.</i>
<b>30</b>, 10 (2014).</p>
<p><sup>10</sup>: van Iterson, M., Tobi, E.W., Slieker, R.C., <i>et
al</i>. <a
href="https://doi.org/10.1093/bioinformatics/btu566">MethylAid: visual
and interactive quality control of large Illumina 450k datasets</a>.
<i>Bioinformatics</i> <b>30</b>, 23 (2014).</p>
<p><sup>11</sup>: van Iterson, M., van Zwet, E.W., BIOS consortium,
<i>et al</i>. <a
href="https://doi.org/10.1186/s13059-016-1131-9">Controlling bias and
inflation in epigenome- and transcriptome-wide association studies using
the empirical null distribution</a>. <i>Genome Biol.</i> <b>18</b>, 19
(2017).</p>
<p><sup>12</sup>: van Iterson, M., Cats, D., Hop, P. <i>et al</i>. <a
href="https://doi.org/10.1093/bioinformatics/bty062">omicsPrint:
detection of data linkage errors in multiple omics studies</a>.
<i>Bioinformatics.</i> <b>34</b>, 12 (2018).</p>
<p><sup>13</sup>: Curtis, S.W., Cobb, D.O., Kilaru, V. <i>et al</i>. <a
href="https://doi.org/10.1080/15592294.2019.1565590">Exposure to
polybrominated biphenyl (PBB) associates with genome-wide DNA
methylation differences in peripheral blood</a>. <i>Epigenetics.</i>
<b>14</b>, 1:52-66 (2019).</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
